<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>TropicalRepo: 5 Taxonomic Diversity</title>
  
  <meta property="description" itemprop="description" content="Reproducible workflow for taxonomic assessment of the 16S rRNA data."/>
  
  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>
  <link rel="icon" type="image/vnd.microsoft.icon" href="assets/favicon.ico"/>
  
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="TropicalRepo: 5 Taxonomic Diversity"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Reproducible workflow for taxonomic assessment of the 16S rRNA data."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="TropicalRepo"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="TropicalRepo: 5 Taxonomic Diversity"/>
  <meta property="twitter:description" content="Reproducible workflow for taxonomic assessment of the 16S rRNA data."/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["5 Taxonomic Diversity"]},{"type":"character","attributes":{},"value":["Reproducible workflow for taxonomic assessment of the 16S rRNA data.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["css","toc","toc_depth"]}},"value":[{"type":"character","attributes":{},"value":["assets/styles.css"]},{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[3]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }
  
  </style>
  
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>
  
  <script type="application/javascript">
  
  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }
  
  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }
  
  function createFuseIndex() {
  
    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);
  
    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });
  
        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }
  
  window.document.addEventListener("DOMContentLoaded", function (event) {
  
    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;
  
    createFuseIndex()
      .then(function(fuse) {
  
        // make search box visible
        searchEl.classList.remove('hidden');
  
        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });
  
  });
  
  </script>
  
  <style type="text/css">
  
  .nav-search {
    font-size: x-small;
  }
  
  /* Algolioa Autocomplete */
  
  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }
  
  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }
  
  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }
  
  </style>
  
  
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }
  
  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }
  
  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }
  
  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }
  
  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }
  
  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }
  
  d-article h3 {
    margin-top: 1.5rem;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  /* Tweak code blocks */
  
  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }
  
  d-article div.sourceCode {
    background-color: white;
  }
  
  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  d-article pre a {
    border-bottom: none;
  }
  
  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }
  
  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }
  
  @media(min-width: 768px) {
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }
  
  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }
  
  d-article pre {
    font-size: 14px;
  }
  
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for d-contents */
  
  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }
  
  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }
  
  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }
  
  .d-contents li {
    list-style-type: none
  }
  
  .d-contents nav > ul {
    padding-left: 0;
  }
  
  .d-contents ul {
    padding-left: 1em
  }
  
  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }
  
  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }
  
  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }
  
  .d-contents nav > ul > li > a {
    font-weight: 600;
  }
  
  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }
  
  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }
  
  
  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }
  
  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }
  
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  /* Citations */
  
  d-article .citation {
    color: inherit;
    cursor: inherit;
  }
  
  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }
  
  /* Citation hover box */
  
  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }
  
  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  /* Include appendix styles here so they can be overridden */
  
  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }
  
  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }
  
  d-appendix h3 + * {
    margin-top: 1em;
  }
  
  d-appendix ol {
    padding: 0 0 0 15px;
  }
  
  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }
  
  d-appendix li {
    margin-bottom: 1em;
  }
  
  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  d-appendix > * {
    grid-column: text;
  }
  
  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }
  
  /* Include footnote styles here so they can be overridden */
  
  d-footnote-list {
    contain: layout style;
  }
  
  d-footnote-list > * {
    grid-column: text;
  }
  
  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }
  
  
  
  /* Anchor.js */
  
  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }
  
  /* Styles for posts lists (not auto-injected) */
  
  
  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }
  
  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }
  
  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }
  
  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }
  
  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }
  
  .post-preview-last {
    border-bottom: none !important;
  }
  
  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }
  
  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }
  
  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }
  
  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }
  
  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }
  
  .posts-list .metadata > * {
    display: inline-block;
  }
  
  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }
  
  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }
  
  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }
  
  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  .posts-list img {
    opacity: 1;
  }
  
  .posts-list img[data-src] {
    opacity: 0;
  }
  
  .posts-more {
    clear: both;
  }
  
  
  .posts-sidebar {
    font-size: 16px;
  }
  
  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }
  
  .sidebar-section {
    margin-bottom: 30px;
  }
  
  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }
  
  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }
  
  .categories li>a {
    border-bottom: none;
  }
  
  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }
  
  .categories .active {
    font-weight: 600;
  }
  
  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }
  
  
  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  .downlevel .posts-list .post-preview {
    color: inherit;
  }
  
  
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }
  
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();
  
    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-contents').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500');
  /********************************/
  /* controls general properties of site*/
  /********************************/
  
  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }
  
  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }
  
  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }
  
  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }
  
  d-article h3 {
    margin-top: 1em;
  }
  
  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }
  
  img.orcid-id {
    margin: 0;
  }
  
  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }
  
  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }
  
  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }
  
  d-article li {
    margin: 0px 0px 4px;
  }
  
  /********************************/
  /* controls CODE*/
  /********************************/
  
  /** pre.sourceCode controls code block without lang **/
  
  /** Controls r code  **/
  
  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #323232;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }
  
   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #FC6464;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/
  
  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;
  
    }
  
  /** Controls inline code  **/
  
  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }
  
  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }
  
  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/
  
  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header  {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .headroom {
      z-index: 1000;
      position: fixed;
      top: 0;
      left: 0;
      right: 0;
  }
  .distill-site-nav {
      background-color: #00203F;
      font-size: 15px;
      font-weight: 300;
  }
  
  img {
    margin: 5px;
  }
  
  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }
  
  /** DETAILS fold down**/
  
  details {
    background-color: hsla(206, 95%, 20%, 0.2);
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }
  
  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;
  
    /* background-color: #acc2c2; */
  }
  
  
  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;
  }
  
  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;
  }
  
  summary::-webkit-details-marker {
    display: none;
    outline: none;
  }
  
  summary:focus {
    outline: none;
  }
  
  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }
  
  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }
  
  
  /********************************/
  /* dygraph*/
  /********************************/
  
  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"5 Taxonomic Diversity","description":"Reproducible workflow for taxonomic assessment of the 16S rRNA data.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title">TropicalRepo</a>
<input id="distill-search" class="nav-search hidden" type="text" placeholder="Search..."/>
</div>
<div class="nav-right">
<a href="trepo-intro.html">Introduction</a>
<a href="trepo-physical.html">Physical</a>
<a href="trepo-inverts.html">Inverts</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Microbial
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Diversity Workflows</span>
<hr/>
<a href="trepo-metadata.html">9. Metadata</a>
<a href="trepo-da.html">8. DA ASVs &amp; OTUs</a>
<a href="trepo-beta.html">7. Beta Diversity</a>
<a href="trepo-alpha.html">6. Alpha Diversity</a>
<a href="trepo-taxa.html">5. Taxonomic</a>
<hr/>
<span class="nav-dropdown-header">Processing Workflows</span>
<a href="trepo-pime.html">4. PIME</a>
<a href="trepo-otu.html">3. OTU Clustering</a>
<a href="trepo-data-prep.html">2. Data Set Prep</a>
<a href="trepo-dada2.html">1. DADA2</a>
<hr/>
</div>
</div>
<a href="trepo-data.html">Data</a>
<a href="https://github.com/tropical-repo/web/" aria-label="Link to source">
<i class="fab fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>5 Taxonomic Diversity</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Reproducible workflow for taxonomic assessment of the 16S rRNA data.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#synopsis">Synopsis</a></li>
<li><a href="#by-phyla">By Phyla</a></li>
<li><a href="#bar-charts-of-major-taxa">Bar charts of Major Taxa</a><ul>
<li><a href="#full-data-set">Full Data Set</a></li>
<li><a href="#merged-data-set">Merged Data Set</a></li>
</ul></li>
<li><a href="#major-taxa-by-family">Major Taxa by Family</a><ul>
<li><a href="#alphaproteobacteria">Alphaproteobacteria</a></li>
<li><a href="#gammaproteobacteria">Gammaproteobacteria</a></li>
<li><a href="#acidobacteriota">Acidobacteriota</a></li>
<li><a href="#chloroflexi">Chloroflexi</a></li>
<li><a href="#desulfobacterota">Desulfobacterota</a></li>
<li><a href="#gemmatimonadota">Gemmatimonadota</a></li>
<li><a href="#latescibacterota">Latescibacterota</a></li>
<li><a href="#nitrospirota">Nitrospirota</a></li>
<li><a href="#thermoplasmatota">Thermoplasmatota</a></li>
<li><a href="#planctomycetota">Planctomycetota</a></li>
<li><a href="#crenarchaeota">Crenarchaeota</a></li>
</ul></li>
<li><a href="#lineage-maps">Lineage Maps</a><ul>
<li><a href="#bacteria">Bacteria</a></li>
<li><a href="#archaea">Archaea</a></li>
<li><a href="#top-phyla">Top Phyla</a></li>
<li><a href="#source-code">Source Code</a></li>
</ul></li>
</ul>
</nav>
</div>
<blockquote>
<p>Hit the <em>Hide Code</em> button to hide the R code (shown by default).</p>
</blockquote>
<aside>
<div class="layout-chunk" data-layout="l-body">
<input onclick="codefolder(&#39;pre.sourceCode&#39;);" type="button" value="Hide Code" id="codefolder-button" style="position: absolute; right: 14%; z-index: 200"/>
<script>
  function codefolder(query) {

    var x = document.querySelectorAll(query);
    if (x.length === 0) return;

    function toggle_vis(o) {
      var d = o.style.display;
      o.style.display = (d === 'block' || d === '') ? 'none':'block';
    }

    for (i = 0; i < x.length; i++) {
      var y = x[i];
      toggle_vis(y);
    }

    var elem = document.getElementById("codefolder-button");
    if (elem.value === "Hide Code") elem.value = "Show Code";
    else elem.value = "Hide Code";
  }
</script>
</div>
</aside>
<h1 id="synopsis">Synopsis</h1>
<p>This workflow contains taxonomic diversity assessments. In order to run the workflow, you either need to first run the <a href="trepo-dada2.html">DADA2 Workflow</a> and then the <a href="trepo-data-prep.html">Data Preparation workflow</a> <strong>or</strong> begin with the output files from the Data Preparation and PIME workflows. See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
<p>In this workflow we will look at the </p>
<h1 id="by-phyla">By Phyla</h1>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co">#TEMP LOAD ONLY REMOVE WHEN WORKFLOW FINISHED</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">remove</span>(<span class="dt">list =</span> <span class="kw">ls</span>())</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw">load</span>(<span class="st">&quot;page_build/trepo/taxa_ssu_wf_part_1.rdata&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="bar-charts-of-major-taxa">Bar charts of Major Taxa</h1>
<p>For broad taxonomic comparisons, we use the the unfiltered data sets.</p>
<aside>
Dont forget to hit the <strong>Show Code</strong> button above if you want to see the R code
</aside>
<details>
<p><summary>Detailed Bar chart workflow </summary></p>
<p>To create a bar chart by Phylum and Proteobacteria classes, we perform the following steps:</p>
<ol type="1">
<li>Get all Class-level Proteobacteria names</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>ssu_data_sets &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ssu_ps_work_merge&quot;</span>,  <span class="st">&quot;ssu_ps_work_merge_otu&quot;</span>,</span>
<span id="cb2-2"><a href="#cb2-2"></a>                     <span class="st">&quot;ssu_ps_work&quot;</span>, <span class="st">&quot;ssu_ps_work_otu&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_data_sets) {</span>
<span id="cb2-5"><a href="#cb2-5"></a>     tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_proteo&quot;</span>))</span>
<span id="cb2-6"><a href="#cb2-6"></a>     tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb2-7"><a href="#cb2-7"></a>     tmp_df &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(tmp_get, Phylum <span class="op">==</span><span class="st"> &quot;Proteobacteria&quot;</span>)</span>
<span id="cb2-8"><a href="#cb2-8"></a>     <span class="kw">assign</span>(tmp_name, tmp_df)</span>
<span id="cb2-9"><a href="#cb2-9"></a>     <span class="kw">print</span>(tmp_name)</span>
<span id="cb2-10"><a href="#cb2-10"></a>     tmp_get_taxa &lt;-<span class="st"> </span><span class="kw">get_taxa_unique</span>(tmp_df,</span>
<span id="cb2-11"><a href="#cb2-11"></a>                                     <span class="dt">taxonomic.rank =</span> <span class="kw">rank_names</span>(tmp_df)[<span class="dv">3</span>],</span>
<span id="cb2-12"><a href="#cb2-12"></a>                                     <span class="dt">errorIfNULL=</span><span class="ot">TRUE</span>)</span>
<span id="cb2-13"><a href="#cb2-13"></a>     <span class="kw">print</span>(tmp_get_taxa)</span>
<span id="cb2-14"><a href="#cb2-14"></a>     <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb2-15"><a href="#cb2-15"></a>     <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;_proteo&quot;</span>))</span>
<span id="cb2-16"><a href="#cb2-16"></a>}</span></code></pre></div>
</div>
<ol start="2" type="1">
<li>Replace Phylum == Proteobacteria with the Class name.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="cf">for</span> (j <span class="cf">in</span> ssu_data_sets) {</span>
<span id="cb3-2"><a href="#cb3-2"></a>  tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(j, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_proteo_clean&quot;</span>))</span>
<span id="cb3-3"><a href="#cb3-3"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(j)</span>
<span id="cb3-4"><a href="#cb3-4"></a>  tmp_clean &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">tax_table</span>(tmp_get))</span>
<span id="cb3-5"><a href="#cb3-5"></a></span>
<span id="cb3-6"><a href="#cb3-6"></a>   <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(tmp_clean)){</span>
<span id="cb3-7"><a href="#cb3-7"></a>       <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="op">==</span><span class="st"> &quot;Proteobacteria&quot;</span> <span class="op">&amp;</span><span class="st"> </span>tmp_clean[i,<span class="dv">3</span>] <span class="op">==</span><span class="st"> &quot;Alphaproteobacteria&quot;</span>){</span>
<span id="cb3-8"><a href="#cb3-8"></a>           phylum &lt;-<span class="st"> </span>base<span class="op">::</span><span class="kw">paste</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>)</span>
<span id="cb3-9"><a href="#cb3-9"></a>           tmp_clean[i, <span class="dv">2</span>] &lt;-<span class="st"> </span>phylum</span>
<span id="cb3-10"><a href="#cb3-10"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="op">==</span><span class="st"> &quot;Proteobacteria&quot;</span> <span class="op">&amp;</span><span class="st"> </span>tmp_clean[i,<span class="dv">3</span>] <span class="op">==</span><span class="st"> &quot;Gammaproteobacteria&quot;</span>){</span>
<span id="cb3-11"><a href="#cb3-11"></a>           phylum &lt;-<span class="st"> </span>base<span class="op">::</span><span class="kw">paste</span>(<span class="st">&quot;Gammaproteobacteria&quot;</span>)</span>
<span id="cb3-12"><a href="#cb3-12"></a>           tmp_clean[i, <span class="dv">2</span>] &lt;-<span class="st"> </span>phylum</span>
<span id="cb3-13"><a href="#cb3-13"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="op">==</span><span class="st"> &quot;Proteobacteria&quot;</span> <span class="op">&amp;</span><span class="st"> </span>tmp_clean[i,<span class="dv">3</span>] <span class="op">==</span><span class="st"> &quot;Zetaproteobacteria&quot;</span>){</span>
<span id="cb3-14"><a href="#cb3-14"></a>              phylum &lt;-<span class="st"> </span>base<span class="op">::</span><span class="kw">paste</span>(<span class="st">&quot;Zetaproteobacteria&quot;</span>)</span>
<span id="cb3-15"><a href="#cb3-15"></a>           tmp_clean[i, <span class="dv">2</span>] &lt;-<span class="st"> </span>phylum</span>
<span id="cb3-16"><a href="#cb3-16"></a>   }   <span class="cf">else</span> <span class="cf">if</span> (tmp_clean[i,<span class="dv">2</span>] <span class="op">==</span><span class="st"> &quot;Proteobacteria&quot;</span> <span class="op">&amp;</span><span class="st"> </span>tmp_clean[i,<span class="dv">3</span>] <span class="op">==</span><span class="st"> &quot;p_Proteobacteria&quot;</span>){</span>
<span id="cb3-17"><a href="#cb3-17"></a>           phylum &lt;-<span class="st"> </span>base<span class="op">::</span><span class="kw">paste</span>(<span class="st">&quot;p_Proteobacteria&quot;</span>)</span>
<span id="cb3-18"><a href="#cb3-18"></a>           tmp_clean[i, <span class="dv">2</span>] &lt;-<span class="st"> </span>phylum</span>
<span id="cb3-19"><a href="#cb3-19"></a>       }</span>
<span id="cb3-20"><a href="#cb3-20"></a>     }</span>
<span id="cb3-21"><a href="#cb3-21"></a>  <span class="kw">tax_table</span>(tmp_get) &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(tmp_clean)</span>
<span id="cb3-22"><a href="#cb3-22"></a>  <span class="kw">rank_names</span>(tmp_get)</span>
<span id="cb3-23"><a href="#cb3-23"></a>  <span class="kw">assign</span>(tmp_name, tmp_get)</span>
<span id="cb3-24"><a href="#cb3-24"></a>  <span class="kw">print</span>(<span class="kw">c</span>(tmp_name, tmp_get))</span>
<span id="cb3-25"><a href="#cb3-25"></a>  <span class="kw">print</span>(<span class="kw">length</span>(<span class="kw">get_taxa_unique</span>(tmp_get,</span>
<span id="cb3-26"><a href="#cb3-26"></a>                               <span class="dt">taxonomic.rank =</span> <span class="kw">rank_names</span>(tmp_get)[<span class="dv">2</span>],</span>
<span id="cb3-27"><a href="#cb3-27"></a>                               <span class="dt">errorIfNULL=</span><span class="ot">TRUE</span>)))</span>
<span id="cb3-28"><a href="#cb3-28"></a>  tmp_path &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="st">&quot;files/trepo/taxa/rdata/&quot;</span>)</span>
<span id="cb3-29"><a href="#cb3-29"></a>  <span class="kw">saveRDS</span>(tmp_get, <span class="kw">paste</span>(tmp_path, j, <span class="st">&quot;_clean.rds&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb3-30"><a href="#cb3-30"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb3-31"><a href="#cb3-31"></a>}</span>
<span id="cb3-32"><a href="#cb3-32"></a><span class="kw">rm</span>(class, order, phylum)</span>
<span id="cb3-33"><a href="#cb3-33"></a><span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_proteo_clean&quot;</span>)</span></code></pre></div>
</div>
<ol start="3" type="1">
<li>Choose the <strong>number</strong> of taxa to display and the taxonomic <strong>level</strong>. Aggregate the rest into Other.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>top_hits &lt;-<span class="st"> </span><span class="dv">12</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>top_level &lt;-<span class="st"> &quot;Phylum&quot;</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_data_sets){</span>
<span id="cb4-4"><a href="#cb4-4"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb4-5"><a href="#cb4-5"></a>  tmp_otu &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">otu_table</span>(tmp_get)))</span>
<span id="cb4-6"><a href="#cb4-6"></a>  tmp_otu[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(tmp_otu, as.numeric)</span>
<span id="cb4-7"><a href="#cb4-7"></a>  tmp_otu &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(tmp_otu)</span>
<span id="cb4-8"><a href="#cb4-8"></a>  tmp_clean_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(., <span class="st">&quot;_proteo_clean&quot;</span>))</span>
<span id="cb4-9"><a href="#cb4-9"></a>  tmp_get_clean &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_clean_name)</span>
<span id="cb4-10"><a href="#cb4-10"></a>  tmp_tax &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">data.frame</span>(<span class="kw">tax_table</span>(tmp_get_clean)))</span>
<span id="cb4-11"><a href="#cb4-11"></a>  tmp_samples &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(tmp_get_clean))</span>
<span id="cb4-12"><a href="#cb4-12"></a>  tmp_clean_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_clean_&quot;</span>, top_level))</span>
<span id="cb4-13"><a href="#cb4-13"></a>  tmp_clean_df &lt;-<span class="st"> </span><span class="kw">merge_phyloseq</span>(<span class="kw">otu_table</span>(tmp_otu, <span class="dt">taxa_are_rows =</span> <span class="ot">TRUE</span>),</span>
<span id="cb4-14"><a href="#cb4-14"></a>                          <span class="kw">tax_table</span>(tmp_tax, tmp_tax),</span>
<span id="cb4-15"><a href="#cb4-15"></a>                          <span class="kw">sample_data</span>(tmp_samples))</span>
<span id="cb4-16"><a href="#cb4-16"></a>  <span class="kw">assign</span>(tmp_clean_name, tmp_clean_df)</span>
<span id="cb4-17"><a href="#cb4-17"></a>  tmp_agg_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_clean_&quot;</span>, top_level, <span class="st">&quot;_agg&quot;</span>))</span>
<span id="cb4-18"><a href="#cb4-18"></a>  tmp_agg_df &lt;-<span class="st"> </span>microbiome<span class="op">::</span><span class="kw">aggregate_top_taxa</span>(tmp_clean_df,</span>
<span id="cb4-19"><a href="#cb4-19"></a>                                                    <span class="dt">top =</span> top_hits,</span>
<span id="cb4-20"><a href="#cb4-20"></a>                                                    <span class="dt">level =</span> top_level)</span>
<span id="cb4-21"><a href="#cb4-21"></a>  <span class="kw">assign</span>(tmp_agg_name, tmp_agg_df)</span>
<span id="cb4-22"><a href="#cb4-22"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb4-23"><a href="#cb4-23"></a>}</span>
<span id="cb4-24"><a href="#cb4-24"></a><span class="kw">objects</span>(<span class="dt">pattern =</span> <span class="st">&quot;_agg&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_agg$&quot;</span>)) {</span>
<span id="cb5-2"><a href="#cb5-2"></a>  tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(., <span class="st">&quot;_order&quot;</span>))</span>
<span id="cb5-3"><a href="#cb5-3"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb5-4"><a href="#cb5-4"></a>  tmp_list &lt;-<span class="st"> </span><span class="kw">get_taxa_unique</span>(tmp_get, <span class="dt">taxonomic.rank =</span> <span class="kw">rank_names</span>(tmp_get)[<span class="dv">2</span>],</span>
<span id="cb5-5"><a href="#cb5-5"></a>                        <span class="dt">errorIfNULL =</span> <span class="ot">TRUE</span>)</span>
<span id="cb5-6"><a href="#cb5-6"></a>  <span class="kw">assign</span>(tmp_name, tmp_list)</span>
<span id="cb5-7"><a href="#cb5-7"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb5-8"><a href="#cb5-8"></a>}</span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_order&quot;</span>)</span></code></pre></div>
</div>
<p>Next, we need to set the order of the taxa to display in the plots. This must be done manually, probably.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>ssu_ps_work_merge_clean_Phylum_agg_order &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, <span class="st">&quot;Calditrichota&quot;</span>, <span class="st">&quot;Chloroflexi&quot;</span>, <span class="st">&quot;Desulfobacterota&quot;</span>, <span class="st">&quot;Gemmatimonadota&quot;</span>, <span class="st">&quot;Latescibacterota&quot;</span>, <span class="st">&quot;Nitrospirota&quot;</span>, <span class="st">&quot;Planctomycetota&quot;</span>, <span class="st">&quot;Crenarchaeota&quot;</span>, <span class="st">&quot;Thermoplasmatota&quot;</span>, <span class="st">&quot;Other&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a>ssu_ps_work_clean_Phylum_agg_order &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, <span class="st">&quot;Calditrichota&quot;</span>, <span class="st">&quot;Chloroflexi&quot;</span>, <span class="st">&quot;Desulfobacterota&quot;</span>, <span class="st">&quot;Gemmatimonadota&quot;</span>, <span class="st">&quot;Latescibacterota&quot;</span>, <span class="st">&quot;Nitrospirota&quot;</span>, <span class="st">&quot;Planctomycetota&quot;</span>, <span class="st">&quot;Crenarchaeota&quot;</span>, <span class="st">&quot;Thermoplasmatota&quot;</span>, <span class="st">&quot;Other&quot;</span>)</span></code></pre></div>
</div>
<ol start="4" type="1">
<li>Now, transform the data to relative abundance.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_agg$&quot;</span>)) {</span>
<span id="cb7-2"><a href="#cb7-2"></a>  tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_tax&quot;</span>))</span>
<span id="cb7-3"><a href="#cb7-3"></a>  tmp_agg &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_order&quot;</span>))</span>
<span id="cb7-4"><a href="#cb7-4"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb7-5"><a href="#cb7-5"></a>  tmp_get_agg &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_agg)</span>
<span id="cb7-6"><a href="#cb7-6"></a>  tmp_df &lt;-<span class="st"> </span>tmp_get <span class="op">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="st">            </span><span class="kw">transform_sample_counts</span>(<span class="cf">function</span>(x) {x<span class="op">/</span><span class="kw">sum</span>(x)} ) <span class="op">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="st">            </span><span class="kw">psmelt</span>()</span>
<span id="cb7-9"><a href="#cb7-9"></a>  tmp_df[[top_level]] &lt;-<span class="st"> </span>gdata<span class="op">::</span><span class="kw">reorder.factor</span>(tmp_df[[top_level]],</span>
<span id="cb7-10"><a href="#cb7-10"></a>                                                  <span class="dt">new.order =</span> <span class="kw">rev</span>(tmp_get_agg))</span>
<span id="cb7-11"><a href="#cb7-11"></a>  tmp_df &lt;-<span class="st"> </span>tmp_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">get</span>(top_level))</span>
<span id="cb7-12"><a href="#cb7-12"></a>  <span class="kw">assign</span>(tmp_name, tmp_df)</span>
<span id="cb7-13"><a href="#cb7-13"></a>  <span class="kw">print</span>(<span class="kw">c</span>(i, tmp_name, tmp_agg))</span>
<span id="cb7-14"><a href="#cb7-14"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb7-15"><a href="#cb7-15"></a>}</span>
<span id="cb7-16"><a href="#cb7-16"></a><span class="kw">objects</span>(<span class="dt">pattern =</span> <span class="st">&quot;_tax&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_tax&quot;</span>)) {</span>
<span id="cb8-2"><a href="#cb8-2"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb8-3"><a href="#cb8-3"></a>  tmp_levels &lt;-<span class="st"> </span><span class="kw">levels</span>(tmp_get[[top_level]])</span>
<span id="cb8-4"><a href="#cb8-4"></a>  <span class="kw">print</span>(<span class="kw">c</span>(i, tmp_levels))</span>
<span id="cb8-5"><a href="#cb8-5"></a>}</span></code></pre></div>
</div>
<ol start="5" type="1">
<li>Plot the data for a single phyloseq object. Here you use an aggragated tax file.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co"># SOME COLOR PALETTES</span></span>
<span id="cb9-2"><a href="#cb9-2"></a>fifteen &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#68023F&quot;</span>, <span class="st">&quot;#008169&quot;</span>, <span class="st">&quot;#EF0096&quot;</span>, <span class="st">&quot;#00DCB5&quot;</span>, <span class="st">&quot;#FFCFE2&quot;</span>, <span class="st">&quot;#003C86&quot;</span>, <span class="st">&quot;#9400E6&quot;</span>, <span class="st">&quot;#009FFA&quot;</span>, <span class="st">&quot;#FF71FD&quot;</span>, <span class="st">&quot;#7CFFFA&quot;</span>, <span class="st">&quot;#6A0213&quot;</span>, <span class="st">&quot;#008607&quot;</span>, <span class="st">&quot;#F60239&quot;</span>, <span class="st">&quot;#00E307&quot;</span>, <span class="st">&quot;#FFDC3D&quot;</span>)</span>
<span id="cb9-3"><a href="#cb9-3"></a>fifteen_alt &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#00463C&quot;</span>, <span class="st">&quot;#C00B6F&quot;</span>, <span class="st">&quot;#00A090&quot;</span>, <span class="st">&quot;#FF95BA&quot;</span>, <span class="st">&quot;#5FFFDE&quot;</span>, <span class="st">&quot;#590A87&quot;</span>, <span class="st">&quot;#0063E5&quot;</span>, <span class="st">&quot;#ED0DFD&quot;</span>, <span class="st">&quot;#00C7F9&quot;</span>, <span class="st">&quot;#FFD5FD&quot;</span>, <span class="st">&quot;#3D3C04&quot;</span>, <span class="st">&quot;#C80B2A&quot;</span>, <span class="st">&quot;#00A51C&quot;</span>, <span class="st">&quot;#FFA035&quot;</span>, <span class="st">&quot;#9BFF2D&quot;</span>)</span>
<span id="cb9-4"><a href="#cb9-4"></a>twelve_alt &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#006A5E&quot;</span>, <span class="st">&quot;#ED0D88&quot;</span>, <span class="st">&quot;#00BDA9&quot;</span>, <span class="st">&quot;#FFC4D4&quot;</span>, <span class="st">&quot;#0058CC&quot;</span>, <span class="st">&quot;#D208FB&quot;</span>, <span class="st">&quot;#FF66FD&quot;</span>, <span class="st">&quot;#00EFF9&quot;</span>, <span class="st">&quot;#156D03&quot;</span>, <span class="st">&quot;#009719&quot;</span>, <span class="st">&quot;#00C61B&quot;</span>, <span class="st">&quot;#00FB1D&quot;</span>)</span>
<span id="cb9-5"><a href="#cb9-5"></a>twelve &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#9F0162&quot;</span>, <span class="st">&quot;#009F81&quot;</span>, <span class="st">&quot;#FF5AAF&quot;</span>, <span class="st">&quot;#00FCCF&quot;</span>, <span class="st">&quot;#8400CD&quot;</span>, <span class="st">&quot;#008DF9&quot;</span>, <span class="st">&quot;#00C2F9&quot;</span>, <span class="st">&quot;#FFB2FD&quot;</span>, <span class="st">&quot;#A40122&quot;</span>, <span class="st">&quot;#E20134&quot;</span>, <span class="st">&quot;#FF6E3A&quot;</span>, <span class="st">&quot;#FFC33B&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>tmp_select &lt;-<span class="st"> </span>ssu_ps_work_merge_clean_Phylum_agg_tax</span>
<span id="cb10-2"><a href="#cb10-2"></a></span>
<span id="cb10-3"><a href="#cb10-3"></a>tmp_select<span class="op">$</span>SITE &lt;-<span class="st"> </span><span class="kw">factor</span>(tmp_select<span class="op">$</span>SITE, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;ALMR&quot;</span>, <span class="st">&quot;PAST&quot;</span>, <span class="st">&quot;CRIS&quot;</span>, <span class="st">&quot;PUCL&quot;</span>))</span>
<span id="cb10-4"><a href="#cb10-4"></a></span>
<span id="cb10-5"><a href="#cb10-5"></a>ssu_ps_work_merge_Phylum_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tmp_select,</span>
<span id="cb10-6"><a href="#cb10-6"></a>                <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(SITE),</span>
<span id="cb10-7"><a href="#cb10-7"></a>                    <span class="dt">y =</span> Abundance, <span class="dt">fill =</span> <span class="kw">get</span>(top_level))) <span class="op">+</span></span>
<span id="cb10-8"><a href="#cb10-8"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb10-9"><a href="#cb10-9"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> ssu_colvec.tax) <span class="op">+</span></span>
<span id="cb10-10"><a href="#cb10-10"></a><span class="st">  </span><span class="co">#scale_x_discrete(&quot;Temperature&quot;, expand = waiver(), position = &quot;bottom&quot;, drop = FALSE) +</span></span>
<span id="cb10-11"><a href="#cb10-11"></a><span class="st">  </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></span>
<span id="cb10-12"><a href="#cb10-12"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> top_level)) <span class="op">+</span></span>
<span id="cb10-13"><a href="#cb10-13"></a><span class="st">  </span><span class="co">#guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +</span></span>
<span id="cb10-14"><a href="#cb10-14"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Relative Abundance (% total reads)&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Site&quot;</span>) <span class="op">+</span></span>
<span id="cb10-15"><a href="#cb10-15"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb10-16"><a href="#cb10-16"></a>        <span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb10-17"><a href="#cb10-17"></a>        <span class="dt">plot.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb10-18"><a href="#cb10-18"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb10-19"><a href="#cb10-19"></a>        <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb10-20"><a href="#cb10-20"></a>        <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">6</span>),</span>
<span id="cb10-21"><a href="#cb10-21"></a>        <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>))</span>
<span id="cb10-22"><a href="#cb10-22"></a>ssu_ps_work_merge_Phylum_plot</span></code></pre></div>
</div>
<ol start="6" type="1">
<li>Plot the data for a multiple phyloseq objects. Here again you use an aggregated tax file. This code can be used to generate plots for multiple data frames by adding the base phyloseq names to the <code>ssu_split_df</code> variable. This code will also facet the plots by a metadata variable. If you do not want to facet remove the line beginning with <code>facet_grid</code>.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co">#ssu_split_df &lt;- c(&quot;ssu_ps_work&quot;, &quot;ssu_ps_work_otu&quot;, &quot;ssu_ps_pime&quot;, &quot;ssu_ps_pime_otu&quot;)</span></span>
<span id="cb11-2"><a href="#cb11-2"></a>ssu_split_df &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ssu_ps_work_merge&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_split_df) {</span>
<span id="cb11-4"><a href="#cb11-4"></a>  tmp_level_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb11-5"><a href="#cb11-5"></a>  tmp_level &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(tmp_level_get))</span>
<span id="cb11-6"><a href="#cb11-6"></a>  tmp_level &lt;-<span class="st"> </span>tmp_level[<span class="kw">order</span>(tmp_level<span class="op">$</span>SITE), ]</span>
<span id="cb11-7"><a href="#cb11-7"></a>  tmp_level &lt;-<span class="st"> </span><span class="kw">as.vector</span>(tmp_level<span class="op">$</span>SamName)</span>
<span id="cb11-8"><a href="#cb11-8"></a></span>
<span id="cb11-9"><a href="#cb11-9"></a>  tmp_agg_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_clean_&quot;</span>, top_level, <span class="st">&quot;_agg_tax&quot;</span>))</span>
<span id="cb11-10"><a href="#cb11-10"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_agg_name)</span>
<span id="cb11-11"><a href="#cb11-11"></a>  tmp_df &lt;-<span class="st"> </span>reshape<span class="op">::</span><span class="kw">melt</span>(tmp_get, <span class="dt">id.vars =</span> <span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;SITE&quot;</span>, <span class="st">&quot;Abundance&quot;</span>, <span class="st">&quot;Phylum&quot;</span>))</span>
<span id="cb11-12"><a href="#cb11-12"></a>  <span class="co">## THIS next line reorders the facet</span></span>
<span id="cb11-13"><a href="#cb11-13"></a>  tmp_df<span class="op">$</span>SITE_r &lt;-<span class="st">    </span><span class="kw">factor</span>(tmp_df<span class="op">$</span>SITE, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;ALMR&quot;</span>, <span class="st">&quot;PAST&quot;</span>, <span class="st">&quot;CRIS&quot;</span>, <span class="st">&quot;PUCL&quot;</span>))</span>
<span id="cb11-14"><a href="#cb11-14"></a>  tmp_plot_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_&quot;</span>, top_level, <span class="st">&quot;_melt_plot&quot;</span>))</span>
<span id="cb11-15"><a href="#cb11-15"></a></span>
<span id="cb11-16"><a href="#cb11-16"></a>  </span>
<span id="cb11-17"><a href="#cb11-17"></a>  tmp_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tmp_df,</span>
<span id="cb11-18"><a href="#cb11-18"></a>                <span class="kw">aes</span>(<span class="dt">x =</span> Sample,</span>
<span id="cb11-19"><a href="#cb11-19"></a>                    <span class="dt">y =</span> Abundance, <span class="dt">fill =</span> <span class="kw">get</span>(top_level))) <span class="op">+</span></span>
<span id="cb11-20"><a href="#cb11-20"></a><span class="st">    </span><span class="kw">facet_grid</span>(. <span class="op">~</span>SITE_r, <span class="dt">scale =</span> <span class="st">&quot;free_x&quot;</span>, <span class="dt">space =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span></span>
<span id="cb11-21"><a href="#cb11-21"></a><span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>)  <span class="op">+</span></span>
<span id="cb11-22"><a href="#cb11-22"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> ssu_colvec.tax) <span class="op">+</span></span>
<span id="cb11-23"><a href="#cb11-23"></a><span class="st">    </span><span class="co">#scale_x_discrete(&quot;Treatment&quot;, expand = waiver(),</span></span>
<span id="cb11-24"><a href="#cb11-24"></a><span class="st">    </span><span class="co">#                 position = &quot;bottom&quot;, drop = FALSE, limits = tmp_level) +</span></span>
<span id="cb11-25"><a href="#cb11-25"></a><span class="st">    </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></span>
<span id="cb11-26"><a href="#cb11-26"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> top_level, <span class="dt">reverse =</span> <span class="ot">FALSE</span>,</span>
<span id="cb11-27"><a href="#cb11-27"></a>                               <span class="dt">keywidth =</span> <span class="fl">0.9</span>, <span class="dt">keyheight =</span> <span class="fl">0.9</span>)) <span class="op">+</span></span>
<span id="cb11-28"><a href="#cb11-28"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb11-29"><a href="#cb11-29"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb11-30"><a href="#cb11-30"></a>        <span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb11-31"><a href="#cb11-31"></a>        <span class="dt">plot.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb11-32"><a href="#cb11-32"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb11-33"><a href="#cb11-33"></a>        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>),</span>
<span id="cb11-34"><a href="#cb11-34"></a>        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>),</span>
<span id="cb11-35"><a href="#cb11-35"></a>        <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb11-36"><a href="#cb11-36"></a>        <span class="co">#legend.key.size = unit(1.5, &quot;cm&quot;),</span></span>
<span id="cb11-37"><a href="#cb11-37"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>),</span>
<span id="cb11-38"><a href="#cb11-38"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">0</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb11-39"><a href="#cb11-39"></a>        <span class="dt">strip.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>, <span class="dt">angle =</span> <span class="dv">0</span>),</span>
<span id="cb11-40"><a href="#cb11-40"></a>        <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="ot">NULL</span>)</span>
<span id="cb11-41"><a href="#cb11-41"></a>  <span class="kw">assign</span>(tmp_plot_name, tmp_plot)</span>
<span id="cb11-42"><a href="#cb11-42"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb11-43"><a href="#cb11-43"></a>}</span>
<span id="cb11-44"><a href="#cb11-44"></a>ssu_ps_work_merge_Phylum_melt_plot</span></code></pre></div>
</div>
<ol start="7" type="1">
<li>Finally we use the <code>patchwork</code> package to combine the two plots and customize the look.</li>
</ol>
<div class="layout-chunk" data-layout="l-page">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Repeat for full data set.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>tmp_select &lt;-<span class="st"> </span>ssu_ps_work_clean_Phylum_agg_tax</span>
<span id="cb12-2"><a href="#cb12-2"></a></span>
<span id="cb12-3"><a href="#cb12-3"></a>tmp_select<span class="op">$</span>SITE &lt;-<span class="st"> </span><span class="kw">factor</span>(tmp_select<span class="op">$</span>SITE, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;ALMR&quot;</span>, <span class="st">&quot;PAST&quot;</span>, <span class="st">&quot;CRIS&quot;</span>, <span class="st">&quot;PUCL&quot;</span>))</span>
<span id="cb12-4"><a href="#cb12-4"></a></span>
<span id="cb12-5"><a href="#cb12-5"></a>ssu_ps_work_Phylum_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tmp_select,</span>
<span id="cb12-6"><a href="#cb12-6"></a>                <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(SITE),</span>
<span id="cb12-7"><a href="#cb12-7"></a>                    <span class="dt">y =</span> Abundance, <span class="dt">fill =</span> <span class="kw">get</span>(top_level))) <span class="op">+</span></span>
<span id="cb12-8"><a href="#cb12-8"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb12-9"><a href="#cb12-9"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> ssu_colvec.tax) <span class="op">+</span></span>
<span id="cb12-10"><a href="#cb12-10"></a><span class="st">  </span><span class="co">#scale_x_discrete(&quot;Temperature&quot;, expand = waiver(), position = &quot;bottom&quot;, drop = FALSE) +</span></span>
<span id="cb12-11"><a href="#cb12-11"></a><span class="st">  </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></span>
<span id="cb12-12"><a href="#cb12-12"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> top_level)) <span class="op">+</span></span>
<span id="cb12-13"><a href="#cb12-13"></a><span class="st">  </span><span class="co">#guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +</span></span>
<span id="cb12-14"><a href="#cb12-14"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Relative Abundance (% total reads)&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Site&quot;</span>) <span class="op">+</span></span>
<span id="cb12-15"><a href="#cb12-15"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb12-16"><a href="#cb12-16"></a>        <span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb12-17"><a href="#cb12-17"></a>        <span class="dt">plot.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb12-18"><a href="#cb12-18"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb12-19"><a href="#cb12-19"></a>        <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb12-20"><a href="#cb12-20"></a>        <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">6</span>),</span>
<span id="cb12-21"><a href="#cb12-21"></a>        <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>))</span>
<span id="cb12-22"><a href="#cb12-22"></a>ssu_ps_work_Phylum_plot</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co">#ssu_split_df &lt;- c(&quot;ssu_ps_work&quot;, &quot;ssu_ps_work_otu&quot;, &quot;ssu_ps_pime&quot;, &quot;ssu_ps_pime_otu&quot;)</span></span>
<span id="cb13-2"><a href="#cb13-2"></a>ssu_split_df &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ssu_ps_work&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_split_df) {</span>
<span id="cb13-4"><a href="#cb13-4"></a>  tmp_level_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb13-5"><a href="#cb13-5"></a>  tmp_level &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(tmp_level_get))</span>
<span id="cb13-6"><a href="#cb13-6"></a>  tmp_level &lt;-<span class="st"> </span>tmp_level[<span class="kw">order</span>(tmp_level<span class="op">$</span>SITE), ]</span>
<span id="cb13-7"><a href="#cb13-7"></a>  tmp_level &lt;-<span class="st"> </span><span class="kw">as.vector</span>(tmp_level<span class="op">$</span>SamName)</span>
<span id="cb13-8"><a href="#cb13-8"></a></span>
<span id="cb13-9"><a href="#cb13-9"></a>  tmp_agg_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_clean_&quot;</span>, top_level, <span class="st">&quot;_agg_tax&quot;</span>))</span>
<span id="cb13-10"><a href="#cb13-10"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_agg_name)</span>
<span id="cb13-11"><a href="#cb13-11"></a>  tmp_df &lt;-<span class="st"> </span>reshape<span class="op">::</span><span class="kw">melt</span>(tmp_get, <span class="dt">id.vars =</span> <span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;SITE&quot;</span>, <span class="st">&quot;Abundance&quot;</span>, <span class="st">&quot;Phylum&quot;</span>))</span>
<span id="cb13-12"><a href="#cb13-12"></a>  <span class="co">## THIS next line reorders the facet</span></span>
<span id="cb13-13"><a href="#cb13-13"></a>  tmp_df<span class="op">$</span>SITE_r &lt;-<span class="st">    </span><span class="kw">factor</span>(tmp_df<span class="op">$</span>SITE, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;ALMR&quot;</span>, <span class="st">&quot;PAST&quot;</span>, <span class="st">&quot;CRIS&quot;</span>, <span class="st">&quot;PUCL&quot;</span>))</span>
<span id="cb13-14"><a href="#cb13-14"></a>  tmp_plot_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_&quot;</span>, top_level, <span class="st">&quot;_melt_plot&quot;</span>))</span>
<span id="cb13-15"><a href="#cb13-15"></a></span>
<span id="cb13-16"><a href="#cb13-16"></a>  </span>
<span id="cb13-17"><a href="#cb13-17"></a>  tmp_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tmp_df,</span>
<span id="cb13-18"><a href="#cb13-18"></a>                <span class="kw">aes</span>(<span class="dt">x =</span> Sample,</span>
<span id="cb13-19"><a href="#cb13-19"></a>                    <span class="dt">y =</span> Abundance, <span class="dt">fill =</span> <span class="kw">get</span>(top_level))) <span class="op">+</span></span>
<span id="cb13-20"><a href="#cb13-20"></a><span class="st">    </span><span class="kw">facet_grid</span>(. <span class="op">~</span>SITE_r, <span class="dt">scale =</span> <span class="st">&quot;free_x&quot;</span>, <span class="dt">space =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span></span>
<span id="cb13-21"><a href="#cb13-21"></a><span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>)  <span class="op">+</span></span>
<span id="cb13-22"><a href="#cb13-22"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> ssu_colvec.tax) <span class="op">+</span></span>
<span id="cb13-23"><a href="#cb13-23"></a><span class="st">    </span><span class="co">#scale_x_discrete(&quot;Treatment&quot;, expand = waiver(),</span></span>
<span id="cb13-24"><a href="#cb13-24"></a><span class="st">    </span><span class="co">#                 position = &quot;bottom&quot;, drop = FALSE, limits = tmp_level) +</span></span>
<span id="cb13-25"><a href="#cb13-25"></a><span class="st">    </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></span>
<span id="cb13-26"><a href="#cb13-26"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> top_level, <span class="dt">reverse =</span> <span class="ot">FALSE</span>,</span>
<span id="cb13-27"><a href="#cb13-27"></a>                               <span class="dt">keywidth =</span> <span class="fl">0.9</span>, <span class="dt">keyheight =</span> <span class="fl">0.9</span>)) <span class="op">+</span></span>
<span id="cb13-28"><a href="#cb13-28"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb13-29"><a href="#cb13-29"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb13-30"><a href="#cb13-30"></a>        <span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb13-31"><a href="#cb13-31"></a>        <span class="dt">plot.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb13-32"><a href="#cb13-32"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb13-33"><a href="#cb13-33"></a>        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>),</span>
<span id="cb13-34"><a href="#cb13-34"></a>        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>),</span>
<span id="cb13-35"><a href="#cb13-35"></a>        <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb13-36"><a href="#cb13-36"></a>        <span class="co">#legend.key.size = unit(1.5, &quot;cm&quot;),</span></span>
<span id="cb13-37"><a href="#cb13-37"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>),</span>
<span id="cb13-38"><a href="#cb13-38"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">0</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb13-39"><a href="#cb13-39"></a>        <span class="dt">strip.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>, <span class="dt">angle =</span> <span class="dv">0</span>),</span>
<span id="cb13-40"><a href="#cb13-40"></a>        <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="ot">NULL</span>)</span>
<span id="cb13-41"><a href="#cb13-41"></a>  <span class="kw">assign</span>(tmp_plot_name, tmp_plot)</span>
<span id="cb13-42"><a href="#cb13-42"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb13-43"><a href="#cb13-43"></a>}</span>
<span id="cb13-44"><a href="#cb13-44"></a>ssu_ps_work_Phylum_melt_plot</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-page">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<h2 id="full-data-set">Full Data Set</h2>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/trepo/taxa/ssu_ps_work_tax_div_bar_plots.png" width="2480" /></p>
</div>
<h2 id="merged-data-set">Merged Data Set</h2>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/trepo/taxa/ssu_ps_work_merge_tax_div_bar_plots.png" width="2480" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="kw">save.image</span>(<span class="st">&quot;page_build/trepo/taxa_ssu_wf_part_1.rdata&quot;</span>)</span></code></pre></div>
</div>
<h1 id="major-taxa-by-family">Major Taxa by Family</h1>
<p>We can also look at the relative abundance of groups within dominant Phyla. This code <strong><em>only</em></strong> works for multiple taxa from a <em>single</em> phyloseq object. For this section, we only present the data for the merged data set.</p>
<details>
<p><summary>Detailed Bar chart workflow </summary></p>
<ol type="1">
<li>Subset Phyla of interest. This can be done for multiple data sets by adding to the <code>ssu_data_sets</code> list</li>
</ol>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>ssu_data_sets &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ssu_ps_work_merge&quot;</span>)</span>
<span id="cb15-2"><a href="#cb15-2"></a>tax_group &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, </span>
<span id="cb15-3"><a href="#cb15-3"></a>               <span class="st">&quot;Calditrichota&quot;</span>, <span class="st">&quot;Chloroflexi&quot;</span>, <span class="st">&quot;Desulfobacterota&quot;</span>, <span class="st">&quot;Gemmatimonadota&quot;</span>, </span>
<span id="cb15-4"><a href="#cb15-4"></a>               <span class="st">&quot;Latescibacterota&quot;</span>, <span class="st">&quot;Nitrospirota&quot;</span>, <span class="st">&quot;Planctomycetota&quot;</span>, <span class="st">&quot;Crenarchaeota&quot;</span>, </span>
<span id="cb15-5"><a href="#cb15-5"></a>               <span class="st">&quot;Thermoplasmatota&quot;</span>)</span>
<span id="cb15-6"><a href="#cb15-6"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_data_sets) {</span>
<span id="cb15-7"><a href="#cb15-7"></a>  <span class="cf">for</span> (j <span class="cf">in</span> tax_group) {</span>
<span id="cb15-8"><a href="#cb15-8"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(., <span class="st">&quot;_proteo_clean&quot;</span>)))</span>
<span id="cb15-9"><a href="#cb15-9"></a>  tmp_sub &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(tmp_get, Phylum <span class="op">==</span><span class="st"> </span>j)</span>
<span id="cb15-10"><a href="#cb15-10"></a>  tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(., <span class="st">&quot;_&quot;</span>, j))</span>
<span id="cb15-11"><a href="#cb15-11"></a>  <span class="kw">assign</span>(tmp_name, tmp_sub)</span>
<span id="cb15-12"><a href="#cb15-12"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb15-13"><a href="#cb15-13"></a>  }</span>
<span id="cb15-14"><a href="#cb15-14"></a>}</span>
<span id="cb15-15"><a href="#cb15-15"></a><span class="kw">objects</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb16-2"><a href="#cb16-2"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge&quot;</span>, <span class="st">&quot;_&quot;</span>, i)))</span>
<span id="cb16-3"><a href="#cb16-3"></a>  tmp_list &lt;-<span class="st">  </span><span class="kw">get_taxa_unique</span>(tmp_get, <span class="dt">taxonomic.rank =</span> <span class="kw">rank_names</span>(tmp_get)[<span class="dv">5</span>], <span class="dt">errorIfNULL =</span> <span class="ot">TRUE</span>)</span>
<span id="cb16-4"><a href="#cb16-4"></a>  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb16-5"><a href="#cb16-5"></a>  <span class="kw">cat</span>(<span class="st">&quot;####################################################&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb16-6"><a href="#cb16-6"></a>  tmp_print &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Unique taxa:&quot;</span>, i)</span>
<span id="cb16-7"><a href="#cb16-7"></a>  <span class="kw">cat</span>(tmp_print, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb16-8"><a href="#cb16-8"></a>  <span class="kw">cat</span>(<span class="st">&quot;####################################################&quot;</span>)</span>
<span id="cb16-9"><a href="#cb16-9"></a>  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb16-10"><a href="#cb16-10"></a>  <span class="kw">print</span>(tmp_list)</span>
<span id="cb16-11"><a href="#cb16-11"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb16-12"><a href="#cb16-12"></a>}</span></code></pre></div>
</div>
<ol start="3" type="1">
<li>Choose the <strong>number</strong> of taxa to display and the taxonomic <strong>level</strong>. Aggregate the rest into Other.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>top_hits &lt;-<span class="st"> </span><span class="dv">12</span></span>
<span id="cb17-2"><a href="#cb17-2"></a>top_level &lt;-<span class="st"> &quot;Family&quot;</span></span>
<span id="cb17-3"><a href="#cb17-3"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group){</span>
<span id="cb17-4"><a href="#cb17-4"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge&quot;</span>, <span class="st">&quot;_&quot;</span>, i)))</span>
<span id="cb17-5"><a href="#cb17-5"></a>  tmp_otu &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">otu_table</span>(tmp_get)))</span>
<span id="cb17-6"><a href="#cb17-6"></a>  tmp_otu[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(tmp_otu, as.numeric)</span>
<span id="cb17-7"><a href="#cb17-7"></a>  tmp_otu &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(tmp_otu)</span>
<span id="cb17-8"><a href="#cb17-8"></a>  tmp_tax &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">data.frame</span>(<span class="kw">tax_table</span>(tmp_get)))</span>
<span id="cb17-9"><a href="#cb17-9"></a>  tmp_samples &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(tmp_get))</span>
<span id="cb17-10"><a href="#cb17-10"></a>  tmp_clean_df &lt;-<span class="st"> </span><span class="kw">merge_phyloseq</span>(<span class="kw">otu_table</span>(tmp_otu, <span class="dt">taxa_are_rows =</span> <span class="ot">TRUE</span>),</span>
<span id="cb17-11"><a href="#cb17-11"></a>                          <span class="kw">tax_table</span>(tmp_tax),</span>
<span id="cb17-12"><a href="#cb17-12"></a>                          <span class="kw">sample_data</span>(tmp_samples))</span>
<span id="cb17-13"><a href="#cb17-13"></a>  tmp_agg_df &lt;-<span class="st"> </span>microbiome<span class="op">::</span><span class="kw">aggregate_top_taxa</span>(tmp_clean_df,</span>
<span id="cb17-14"><a href="#cb17-14"></a>                                                    <span class="dt">top =</span> top_hits,</span>
<span id="cb17-15"><a href="#cb17-15"></a>                                                    <span class="dt">level =</span> top_level)</span>
<span id="cb17-16"><a href="#cb17-16"></a>  tmp_agg_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge&quot;</span>, <span class="st">&quot;_&quot;</span>, i, <span class="st">&quot;_agg&quot;</span>))</span>
<span id="cb17-17"><a href="#cb17-17"></a>  <span class="kw">assign</span>(tmp_agg_name, tmp_agg_df)</span>
<span id="cb17-18"><a href="#cb17-18"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;_sep_agg&quot;</span>))</span>
<span id="cb17-19"><a href="#cb17-19"></a>}</span>
<span id="cb17-20"><a href="#cb17-20"></a><span class="kw">objects</span>(<span class="dt">pattern =</span> <span class="st">&quot;_agg$&quot;</span>)</span>
<span id="cb17-21"><a href="#cb17-21"></a><span class="kw">objects</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group){</span>
<span id="cb18-2"><a href="#cb18-2"></a>  tmp_data &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, i, <span class="st">&quot;_agg&quot;</span>))</span>
<span id="cb18-3"><a href="#cb18-3"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_data)</span>
<span id="cb18-4"><a href="#cb18-4"></a>  tmp_list &lt;-<span class="st"> </span><span class="kw">get_taxa_unique</span>(tmp_get, <span class="dt">taxonomic.rank =</span> <span class="kw">rank_names</span>(tmp_get)[<span class="dv">1</span>],</span>
<span id="cb18-5"><a href="#cb18-5"></a>                        <span class="dt">errorIfNULL =</span> <span class="ot">TRUE</span>)</span>
<span id="cb18-6"><a href="#cb18-6"></a>  tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(tmp_data, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(., <span class="st">&quot;_order&quot;</span>))</span>
<span id="cb18-7"><a href="#cb18-7"></a>  <span class="kw">assign</span>(tmp_name, tmp_list)</span>
<span id="cb18-8"><a href="#cb18-8"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb18-9"><a href="#cb18-9"></a>}</span>
<span id="cb18-10"><a href="#cb18-10"></a><span class="kw">objects</span>(<span class="dt">pattern =</span> <span class="st">&quot;_agg_order$&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb19-2"><a href="#cb19-2"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge&quot;</span>, <span class="st">&quot;_&quot;</span>, i, <span class="st">&quot;_agg_order&quot;</span>)))</span>
<span id="cb19-3"><a href="#cb19-3"></a>  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb19-4"><a href="#cb19-4"></a>  <span class="kw">cat</span>(<span class="st">&quot;#########&quot;</span>, i, <span class="st">&quot;########&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb19-5"><a href="#cb19-5"></a>  tmp_print &lt;-<span class="st"> </span><span class="kw">c</span>(tmp_get)</span>
<span id="cb19-6"><a href="#cb19-6"></a>  <span class="kw">cat</span>(tmp_print, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb19-7"><a href="#cb19-7"></a>  <span class="kw">cat</span>(<span class="st">&quot;####################################################&quot;</span>)</span>
<span id="cb19-8"><a href="#cb19-8"></a>  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb19-9"><a href="#cb19-9"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb19-10"><a href="#cb19-10"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>ssu_ps_work_merge_Alphaproteobacteria_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;c_Alphaproteobacteria&quot;</span>, <span class="st">&quot;o_AT-s3-44&quot;</span>, <span class="st">&quot;o_Defluviicoccales&quot;</span>, <span class="st">&quot;Rhizobiales_Incertae_Sedis&quot;</span>, <span class="st">&quot;PS1_clade&quot;</span>, <span class="st">&quot;Hyphomicrobiaceae&quot;</span>, <span class="st">&quot;Magnetospiraceae&quot;</span>, <span class="st">&quot;Kiloniellaceae&quot;</span>, <span class="st">&quot;Methyloligellaceae&quot;</span>, <span class="st">&quot;Rhizobiaceae&quot;</span>, <span class="st">&quot;Rhodobacteraceae&quot;</span>, <span class="st">&quot;Xanthobacteraceae&quot;</span>))</span>
<span id="cb20-2"><a href="#cb20-2"></a>ssu_ps_work_merge_Gammaproteobacteria_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;c_Gammaproteobacteria&quot;</span>, <span class="st">&quot;o_AT-s2-59&quot;</span>, <span class="st">&quot;o_B2M28&quot;</span>, <span class="st">&quot;o_BD7-8&quot;</span>, <span class="st">&quot;o_EPR3968-O8a-Bc78&quot;</span>, <span class="st">&quot;o_SS1-B-09-64&quot;</span>, <span class="st">&quot;o_UBA10353_marine_group&quot;</span>, <span class="st">&quot;Unknown_Family&quot;</span>, <span class="st">&quot;Ectothiorhodospiraceae&quot;</span>, <span class="st">&quot;Nitrosococcaceae&quot;</span>, <span class="st">&quot;Thioalkalispiraceae&quot;</span>, <span class="st">&quot;Woeseiaceae&quot;</span>))</span>
<span id="cb20-3"><a href="#cb20-3"></a>ssu_ps_work_merge_Acidobacteriota_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;p_Acidobacteriota&quot;</span>, <span class="st">&quot;Other&quot;</span>, <span class="st">&quot;c_AT-s3-28&quot;</span>, <span class="st">&quot;c_Subgroup_18&quot;</span>, <span class="st">&quot;c_Subgroup_21&quot;</span>, <span class="st">&quot;c_Subgroup_22&quot;</span>, <span class="st">&quot;c_Subgroup_26&quot;</span>, <span class="st">&quot;o_Aminicenantales&quot;</span>, <span class="st">&quot;o_PAUC26f&quot;</span>, <span class="st">&quot;o_Subgroup_17&quot;</span>, <span class="st">&quot;o_Subgroup_9&quot;</span>, <span class="st">&quot;o_Vicinamibacterales&quot;</span>, <span class="st">&quot;Thermoanaerobaculaceae&quot;</span>))</span>
<span id="cb20-4"><a href="#cb20-4"></a>ssu_ps_work_merge_Calditrichota_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;Calditrichaceae&quot;</span>))</span>
<span id="cb20-5"><a href="#cb20-5"></a>ssu_ps_work_merge_Chloroflexi_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;AB-539-J10&quot;</span>, <span class="st">&quot;c_Anaerolineae&quot;</span>, <span class="st">&quot;c_Dehalococcoidia&quot;</span>, <span class="st">&quot;o_ADurb.Bin180&quot;</span>, <span class="st">&quot;o_Ardenticatenales&quot;</span>, <span class="st">&quot;o_FS117-23B-02&quot;</span>, <span class="st">&quot;o_FW22&quot;</span>, <span class="st">&quot;o_GIF3&quot;</span>, <span class="st">&quot;o_MSBL5&quot;</span>, <span class="st">&quot;o_Napoli-4B-65&quot;</span>, <span class="st">&quot;o_SBR1031&quot;</span>, <span class="st">&quot;Anaerolineaceae&quot;</span>))</span>
<span id="cb20-6"><a href="#cb20-6"></a>ssu_ps_work_merge_Desulfobacterota_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;p_Desulfobacterota&quot;</span>, <span class="st">&quot;o_Bradymonadales&quot;</span>, <span class="st">&quot;o_Desulfobacterales&quot;</span>, <span class="st">&quot;o_Syntrophobacterales&quot;</span>, <span class="st">&quot;Desulfobaccaceae&quot;</span>, <span class="st">&quot;Desulfatiglandaceae&quot;</span>, <span class="st">&quot;Desulfobulbaceae&quot;</span>, <span class="st">&quot;Desulfocapsaceae&quot;</span>, <span class="st">&quot;Desulfomonilaceae&quot;</span>, <span class="st">&quot;Desulfosarcinaceae&quot;</span>, <span class="st">&quot;Desulfurivibrionaceae&quot;</span>, <span class="st">&quot;Syntrophobacteraceae&quot;</span>))</span>
<span id="cb20-7"><a href="#cb20-7"></a>ssu_ps_work_merge_Gemmatimonadota_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;p_Gemmatimonadota&quot;</span>, <span class="st">&quot;c_AKAU4049&quot;</span>, <span class="st">&quot;c_BD2-11_terrestrial_group&quot;</span>, <span class="st">&quot;c_MD2902-B12&quot;</span>, <span class="st">&quot;c_S0134_terrestrial_group&quot;</span>, <span class="st">&quot;c_PAUC43f_marine_benthic_group&quot;</span>, <span class="st">&quot;Gemmatimonadaceae&quot;</span>))</span>
<span id="cb20-8"><a href="#cb20-8"></a>ssu_ps_work_merge_Latescibacterota_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;p_Latescibacterota&quot;</span>, <span class="st">&quot;Latescibacteraceae&quot;</span>))</span>
<span id="cb20-9"><a href="#cb20-9"></a>ssu_ps_work_merge_Nitrospirota_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;p_Nitrospirota&quot;</span>, <span class="st">&quot;c_4-29-1&quot;</span>, <span class="st">&quot;c_BMS9AB35&quot;</span>, <span class="st">&quot;c_Thermodesulfovibrionia&quot;</span>, <span class="st">&quot;Thermodesulfovibrionaceae&quot;</span>, <span class="st">&quot;Nitrospiraceae&quot;</span>))</span>
<span id="cb20-10"><a href="#cb20-10"></a>ssu_ps_work_merge_Planctomycetota_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;p_Planctomycetota&quot;</span>, <span class="st">&quot;c_OM190&quot;</span>, <span class="st">&quot;c_Pla3_lineage&quot;</span>, <span class="st">&quot;c_Pla4_lineage&quot;</span>, <span class="st">&quot;c_vadinHA49&quot;</span>, <span class="st">&quot;o_CCM11a&quot;</span>, <span class="st">&quot;AKAU3564_sediment_group&quot;</span>, <span class="st">&quot;SG8-4&quot;</span>, <span class="st">&quot;Phycisphaeraceae&quot;</span>, <span class="st">&quot;Pirellulaceae&quot;</span>, <span class="st">&quot;Rubinisphaeraceae&quot;</span>, <span class="st">&quot;Gimesiaceae&quot;</span>)) </span>
<span id="cb20-11"><a href="#cb20-11"></a>ssu_ps_work_merge_Crenarchaeota_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;Other&quot;</span>, <span class="st">&quot;p_Crenarchaeota&quot;</span>, <span class="st">&quot;c_Bathyarchaeia&quot;</span>, <span class="st">&quot;c_Nitrososphaeria&quot;</span>, <span class="st">&quot;c_Thermoprotei&quot;</span>, <span class="st">&quot;o_Caldiarchaeales&quot;</span>, <span class="st">&quot;o_Marine_Benthic_Group_A&quot;</span>, <span class="st">&quot;o_SCGC_AB-179-E04&quot;</span>, <span class="st">&quot;Acidilobaceae&quot;</span>, <span class="st">&quot;Geothermarchaeaceae&quot;</span>, <span class="st">&quot;Nitrososphaeraceae&quot;</span>, <span class="st">&quot;Nitrosopumilaceae&quot;</span>, <span class="st">&quot;Thermofilaceae&quot;</span>)) </span>
<span id="cb20-12"><a href="#cb20-12"></a>ssu_ps_work_merge_Thermoplasmatota_agg_order &lt;-<span class="st"> </span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;p_Thermoplasmatota&quot;</span>, <span class="st">&quot;c_Thermoplasmata&quot;</span>, <span class="st">&quot;o_JdFR-43&quot;</span>, <span class="st">&quot;o_Methanomassiliicoccales&quot;</span>, <span class="st">&quot;o_SG8-5&quot;</span>, <span class="st">&quot;o_Marine_Benthic_Group_D_and_DHVEG-1&quot;</span>, <span class="st">&quot;Aciduliprofundaceae&quot;</span>, <span class="st">&quot;Thermoplasmata_fa&quot;</span>, <span class="st">&quot;Thermoplasmatota_fa&quot;</span>)) </span></code></pre></div>
</div>
<ol start="4" type="1">
<li>Now, transform the data to relative abundance.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb21-2"><a href="#cb21-2"></a>  tmp_agg &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, i, <span class="st">&quot;_agg&quot;</span>))</span>
<span id="cb21-3"><a href="#cb21-3"></a>  tmp_order &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(tmp_agg, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_order&quot;</span>))</span>
<span id="cb21-4"><a href="#cb21-4"></a>  tmp_get_agg &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_agg)</span>
<span id="cb21-5"><a href="#cb21-5"></a>  tmp_get_order &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_order)</span>
<span id="cb21-6"><a href="#cb21-6"></a>  tmp_df &lt;-<span class="st"> </span>tmp_get_agg <span class="op">%&gt;%</span></span>
<span id="cb21-7"><a href="#cb21-7"></a><span class="st">            </span><span class="kw">transform_sample_counts</span>(<span class="cf">function</span>(x) {x<span class="op">/</span><span class="kw">sum</span>(x)} ) <span class="op">%&gt;%</span></span>
<span id="cb21-8"><a href="#cb21-8"></a><span class="st">            </span><span class="kw">psmelt</span>()</span>
<span id="cb21-9"><a href="#cb21-9"></a>  tmp_df[[top_level]] &lt;-<span class="st"> </span>gdata<span class="op">::</span><span class="kw">reorder.factor</span>(tmp_df[[top_level]],</span>
<span id="cb21-10"><a href="#cb21-10"></a>                                                  <span class="dt">new.order =</span> <span class="kw">rev</span>(tmp_get_order))</span>
<span id="cb21-11"><a href="#cb21-11"></a>  tmp_df &lt;-<span class="st"> </span>tmp_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">get</span>(top_level))</span>
<span id="cb21-12"><a href="#cb21-12"></a>  tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(tmp_agg, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_tax&quot;</span>))</span>
<span id="cb21-13"><a href="#cb21-13"></a>  <span class="kw">assign</span>(tmp_name, tmp_df)</span>
<span id="cb21-14"><a href="#cb21-14"></a>  <span class="co">#print(c(i, tmp_name, tmp_agg))</span></span>
<span id="cb21-15"><a href="#cb21-15"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb21-16"><a href="#cb21-16"></a>}</span>
<span id="cb21-17"><a href="#cb21-17"></a><span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_tax&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb22-2"><a href="#cb22-2"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, i, <span class="st">&quot;_agg_tax&quot;</span>)))</span>
<span id="cb22-3"><a href="#cb22-3"></a>  tmp_levels &lt;-<span class="st"> </span><span class="kw">levels</span>(tmp_get[[top_level]])</span>
<span id="cb22-4"><a href="#cb22-4"></a>  <span class="kw">print</span>(<span class="kw">c</span>(i, tmp_levels))</span>
<span id="cb22-5"><a href="#cb22-5"></a>}</span></code></pre></div>
</div>
<ol start="5" type="1">
<li>Plot the data for a single phyloseq object. Here you use an aggragated tax file.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb23-2"><a href="#cb23-2"></a>      tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, i, <span class="st">&quot;_agg_tax&quot;</span>)))</span>
<span id="cb23-3"><a href="#cb23-3"></a>      tmp_get<span class="op">$</span>SITE &lt;-<span class="st"> </span><span class="kw">factor</span>(tmp_get<span class="op">$</span>SITE, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;ALMR&quot;</span>, <span class="st">&quot;PAST&quot;</span>, <span class="st">&quot;CRIS&quot;</span>, <span class="st">&quot;PUCL&quot;</span>))</span>
<span id="cb23-4"><a href="#cb23-4"></a></span>
<span id="cb23-5"><a href="#cb23-5"></a>      tmp_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tmp_get, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(SITE), </span>
<span id="cb23-6"><a href="#cb23-6"></a>                                        <span class="dt">y =</span> Abundance, <span class="dt">fill =</span> <span class="kw">get</span>(top_level))) <span class="op">+</span></span>
<span id="cb23-7"><a href="#cb23-7"></a><span class="st">          </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb23-8"><a href="#cb23-8"></a><span class="st">          </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> ssu_colvec.tax) <span class="op">+</span></span>
<span id="cb23-9"><a href="#cb23-9"></a><span class="st">          </span><span class="co">#scale_x_discrete(&quot;Temperature&quot;, expand = waiver(), position = &quot;bottom&quot;, drop = FALSE) +</span></span>
<span id="cb23-10"><a href="#cb23-10"></a><span class="st">          </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></span>
<span id="cb23-11"><a href="#cb23-11"></a><span class="st">          </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> top_level)) <span class="op">+</span></span>
<span id="cb23-12"><a href="#cb23-12"></a><span class="st">          </span><span class="co">#guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +</span></span>
<span id="cb23-13"><a href="#cb23-13"></a><span class="st">          </span><span class="kw">ylab</span>(<span class="st">&quot;Relative Abundance (% total reads)&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Site&quot;</span>) <span class="op">+</span></span>
<span id="cb23-14"><a href="#cb23-14"></a><span class="st">          </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb23-15"><a href="#cb23-15"></a>                <span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb23-16"><a href="#cb23-16"></a>                <span class="dt">plot.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb23-17"><a href="#cb23-17"></a>                <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb23-18"><a href="#cb23-18"></a>                <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">0</span>),</span>
<span id="cb23-19"><a href="#cb23-19"></a>                <span class="co">#axis.title = element_text(size = 8),</span></span>
<span id="cb23-20"><a href="#cb23-20"></a>                <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb23-21"><a href="#cb23-21"></a>      tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, ., <span class="st">&quot;_plot&quot;</span>))</span>
<span id="cb23-22"><a href="#cb23-22"></a>      <span class="kw">assign</span>(tmp_name, tmp_plot)</span>
<span id="cb23-23"><a href="#cb23-23"></a>      <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb23-24"><a href="#cb23-24"></a>}</span>
<span id="cb23-25"><a href="#cb23-25"></a></span>
<span id="cb23-26"><a href="#cb23-26"></a><span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_plot&quot;</span>)</span>
<span id="cb23-27"><a href="#cb23-27"></a><span class="kw">objects</span>()</span></code></pre></div>
</div>
<ol start="6" type="1">
<li>Plot the data for multiple taxa. Here again you use an aggregated tax file. This code can be used to generate plots for multiple data frames by adding the base phyloseq names to the <code>ssu_split_df</code> variable. This code will also facet the plots by a metadata variable. If you do not want to facet remove the line beginning with <code>facet_grid</code>.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb24-2"><a href="#cb24-2"></a>  tmp_level_get &lt;-<span class="st"> </span><span class="kw">get</span>(purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, .)))</span>
<span id="cb24-3"><a href="#cb24-3"></a>  tmp_level &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(tmp_level_get))</span>
<span id="cb24-4"><a href="#cb24-4"></a>  tmp_level &lt;-<span class="st"> </span>tmp_level[<span class="kw">order</span>(tmp_level<span class="op">$</span>SITE), ]</span>
<span id="cb24-5"><a href="#cb24-5"></a>  tmp_level &lt;-<span class="st"> </span><span class="kw">as.vector</span>(tmp_level<span class="op">$</span>SamName)</span>
<span id="cb24-6"><a href="#cb24-6"></a></span>
<span id="cb24-7"><a href="#cb24-7"></a>  tmp_agg_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, .,  <span class="st">&quot;_agg_tax&quot;</span>))</span>
<span id="cb24-8"><a href="#cb24-8"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_agg_name)</span>
<span id="cb24-9"><a href="#cb24-9"></a>  tmp_df &lt;-<span class="st"> </span>reshape<span class="op">::</span><span class="kw">melt</span>(tmp_get, <span class="dt">id.vars =</span> <span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;SITE&quot;</span>, <span class="st">&quot;Abundance&quot;</span>, <span class="st">&quot;Family&quot;</span>))</span>
<span id="cb24-10"><a href="#cb24-10"></a>  tmp_df<span class="op">$</span>SITE_r &lt;-<span class="st">    </span><span class="kw">factor</span>(tmp_df<span class="op">$</span>SITE, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;ALMR&quot;</span>, <span class="st">&quot;PAST&quot;</span>, <span class="st">&quot;CRIS&quot;</span>, <span class="st">&quot;PUCL&quot;</span>))</span>
<span id="cb24-11"><a href="#cb24-11"></a></span>
<span id="cb24-12"><a href="#cb24-12"></a>  tmp_plot_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, ., <span class="st">&quot;_plot_melt&quot;</span>))</span>
<span id="cb24-13"><a href="#cb24-13"></a></span>
<span id="cb24-14"><a href="#cb24-14"></a>  tmp_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tmp_df,</span>
<span id="cb24-15"><a href="#cb24-15"></a>                <span class="kw">aes</span>(<span class="dt">x =</span> Sample,</span>
<span id="cb24-16"><a href="#cb24-16"></a>                    <span class="dt">y =</span> Abundance, <span class="dt">fill =</span> <span class="kw">get</span>(top_level))) <span class="op">+</span></span>
<span id="cb24-17"><a href="#cb24-17"></a><span class="st">    </span><span class="kw">facet_grid</span>(. <span class="op">~</span>SITE_r, <span class="dt">scale =</span> <span class="st">&quot;free_x&quot;</span>, <span class="dt">space =</span> <span class="st">&quot;free_x&quot;</span>)<span class="op">+</span></span>
<span id="cb24-18"><a href="#cb24-18"></a><span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>)  <span class="op">+</span></span>
<span id="cb24-19"><a href="#cb24-19"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> ssu_colvec.tax) <span class="op">+</span></span>
<span id="cb24-20"><a href="#cb24-20"></a><span class="st">    </span><span class="co">#scale_x_discrete(&quot;Treatment&quot;, expand = waiver(),</span></span>
<span id="cb24-21"><a href="#cb24-21"></a><span class="st">    </span><span class="co">#                 position = &quot;bottom&quot;, drop = FALSE, limits = tmp_level) +</span></span>
<span id="cb24-22"><a href="#cb24-22"></a><span class="st">    </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></span>
<span id="cb24-23"><a href="#cb24-23"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> top_level, <span class="dt">reverse =</span> <span class="ot">FALSE</span>,</span>
<span id="cb24-24"><a href="#cb24-24"></a>                               <span class="dt">keywidth =</span> <span class="fl">0.7</span>, <span class="dt">keyheight =</span> <span class="fl">0.7</span>)) <span class="op">+</span></span>
<span id="cb24-25"><a href="#cb24-25"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb24-26"><a href="#cb24-26"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb24-27"><a href="#cb24-27"></a>        <span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb24-28"><a href="#cb24-28"></a>        <span class="dt">plot.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb24-29"><a href="#cb24-29"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb24-30"><a href="#cb24-30"></a>        <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb24-31"><a href="#cb24-31"></a>        <span class="co">#legend.text = element_text(size = 6),</span></span>
<span id="cb24-32"><a href="#cb24-32"></a>        <span class="co">#legend.title = element_text(size = 8),</span></span>
<span id="cb24-33"><a href="#cb24-33"></a>        <span class="co">#legend.key.size = unit(1.5, &quot;cm&quot;),</span></span>
<span id="cb24-34"><a href="#cb24-34"></a>        <span class="co">#axis.text.y = element_text(size = 6),</span></span>
<span id="cb24-35"><a href="#cb24-35"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">size =</span> <span class="dv">0</span>)</span>
<span id="cb24-36"><a href="#cb24-36"></a>        <span class="co">#strip.text = element_text(size = 8, angle = 0),</span></span>
<span id="cb24-37"><a href="#cb24-37"></a>        <span class="co">#axis.title = element_text(size = 8)</span></span>
<span id="cb24-38"><a href="#cb24-38"></a>        ) <span class="op">+</span><span class="st"> </span></span>
<span id="cb24-39"><a href="#cb24-39"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>)</span>
<span id="cb24-40"><a href="#cb24-40"></a>  </span>
<span id="cb24-41"><a href="#cb24-41"></a>  <span class="kw">assign</span>(tmp_plot_name, tmp_plot)</span>
<span id="cb24-42"><a href="#cb24-42"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb24-43"><a href="#cb24-43"></a>}</span>
<span id="cb24-44"><a href="#cb24-44"></a><span class="kw">objects</span>()</span></code></pre></div>
</div>
<ol start="7" type="1">
<li>Finally we use the <code>patchwork</code> package to combine the two plots and customize the look.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tax_group) {</span>
<span id="cb25-2"><a href="#cb25-2"></a>  tmp_plot_main &lt;-<span class="st"> </span><span class="kw">get</span>(purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, ., <span class="st">&quot;_plot&quot;</span>)))</span>
<span id="cb25-3"><a href="#cb25-3"></a>  tmp_plot_melt &lt;-<span class="st"> </span><span class="kw">get</span>(purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, ., <span class="st">&quot;_plot_melt&quot;</span>)))</span>
<span id="cb25-4"><a href="#cb25-4"></a>  tmp_final &lt;-<span class="st"> </span>tmp_plot_main <span class="op">+</span><span class="st"> </span>tmp_plot_melt</span>
<span id="cb25-5"><a href="#cb25-5"></a>  tmp_final &lt;-<span class="st"> </span>tmp_final <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-6"><a href="#cb25-6"></a><span class="st">    </span><span class="kw">plot_annotation</span>(<span class="dt">tag_levels =</span> <span class="st">&#39;A&#39;</span>, </span>
<span id="cb25-7"><a href="#cb25-7"></a>                    <span class="dt">title =</span> i) <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-8"><a href="#cb25-8"></a><span class="st">                    </span><span class="co">#subtitle = &#39;Top taxa of non-filtered data&#39;,</span></span>
<span id="cb25-9"><a href="#cb25-9"></a><span class="st">                    </span><span class="co">#caption = &#39;A) grouped by temperature.,</span></span>
<span id="cb25-10"><a href="#cb25-10"></a><span class="st">                    </span><span class="co">#B) All samples, faceted by temperature.&#39;) + </span></span>
<span id="cb25-11"><a href="#cb25-11"></a><span class="st">    </span><span class="kw">plot_layout</span>(<span class="dt">widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)) <span class="op">&amp;</span></span>
<span id="cb25-12"><a href="#cb25-12"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>),</span>
<span id="cb25-13"><a href="#cb25-13"></a>        <span class="dt">plot.subtitle =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">1</span>),</span>
<span id="cb25-14"><a href="#cb25-14"></a>        <span class="dt">plot.tag =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">6</span>),</span>
<span id="cb25-15"><a href="#cb25-15"></a>        <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>),</span>
<span id="cb25-16"><a href="#cb25-16"></a>        <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">6</span>),</span>
<span id="cb25-17"><a href="#cb25-17"></a>        <span class="dt">strip.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">angle =</span> <span class="dv">0</span>),</span>
<span id="cb25-18"><a href="#cb25-18"></a>        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>),</span>
<span id="cb25-19"><a href="#cb25-19"></a>        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>)</span>
<span id="cb25-20"><a href="#cb25-20"></a>        )</span>
<span id="cb25-21"><a href="#cb25-21"></a>        <span class="co">#legend.position = &quot;right&quot;)</span></span>
<span id="cb25-22"><a href="#cb25-22"></a>        <span class="co">#legend.position = &quot;right&quot;,</span></span>
<span id="cb25-23"><a href="#cb25-23"></a>        <span class="co">#legend.title = element_text(size = rel(1)),</span></span>
<span id="cb25-24"><a href="#cb25-24"></a>        <span class="co">#legend.text = element_text(size = rel(1)))</span></span>
<span id="cb25-25"><a href="#cb25-25"></a>  tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(<span class="st">&quot;ssu_ps_work_merge_&quot;</span>, ., <span class="st">&quot;_final_plot&quot;</span>))</span>
<span id="cb25-26"><a href="#cb25-26"></a>  <span class="kw">assign</span>(tmp_name, tmp_final)</span>
<span id="cb25-27"><a href="#cb25-27"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb25-28"><a href="#cb25-28"></a></span>
<span id="cb25-29"><a href="#cb25-29"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="kw">save.image</span>(<span class="st">&quot;page_build/trepo/taxa_ssu_wf_part_2.rdata&quot;</span>)</span></code></pre></div>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="alphaproteobacteria">Alphaproteobacteria</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-235-1.png" width="624" /></p>
</div>
<h2 id="gammaproteobacteria">Gammaproteobacteria</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-236-1.png" width="624" /></p>
</div>
<h2 id="acidobacteriota">Acidobacteriota</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-237-1.png" width="624" /></p>
</div>
<h2 id="chloroflexi">Chloroflexi</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-238-1.png" width="624" /></p>
</div>
<h2 id="desulfobacterota">Desulfobacterota</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-239-1.png" width="624" /></p>
</div>
<h2 id="gemmatimonadota">Gemmatimonadota</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-240-1.png" width="624" /></p>
</div>
<h2 id="latescibacterota">Latescibacterota</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-241-1.png" width="624" /></p>
</div>
<h2 id="nitrospirota">Nitrospirota</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-242-1.png" width="624" /></p>
</div>
<h2 id="thermoplasmatota">Thermoplasmatota</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-243-1.png" width="624" /></p>
</div>
<h2 id="planctomycetota">Planctomycetota</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-244-1.png" width="624" /></p>
</div>
<h2 id="crenarchaeota">Crenarchaeota</h2>
<div class="layout-chunk" data-layout="l-body-outset">
<p><img src="trepo-taxa_files/figure-html5/unnamed-chunk-245-1.png" width="624" /></p>
</div>
<h1 id="lineage-maps">Lineage Maps</h1>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a><span class="co">#TEMP LOAD ONLY REMOVE WHEN WORKFLOW FINISHED</span></span>
<span id="cb27-2"><a href="#cb27-2"></a><span class="kw">remove</span>(<span class="dt">list =</span> <span class="kw">ls</span>())</span>
<span id="cb27-3"><a href="#cb27-3"></a><span class="kw">load</span>(<span class="st">&quot;page_build/trepo/taxa_ssu_wf_part_3.rdata&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>ps_work_merge &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;files/trepo/data-prep/rdata/ssu_ps_work_merge.rds&quot;</span>)</span>
<span id="cb28-2"><a href="#cb28-2"></a>ps_water_mc_<span class="dv">100</span> &lt;-<span class="st"> </span><span class="kw">prune_taxa</span>(<span class="kw">taxa_sums</span>(ps_work_merge) <span class="op">&gt;</span><span class="st"> </span><span class="dv">500</span>, ps_work_merge)</span>
<span id="cb28-3"><a href="#cb28-3"></a>total_reads_ps_water &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">readcount</span>(ps_work_merge))</span>
<span id="cb28-4"><a href="#cb28-4"></a>total_reads_ps_water_<span class="dv">100</span> &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">readcount</span>(ps_water_mc_<span class="dv">100</span>))</span>
<span id="cb28-5"><a href="#cb28-5"></a>ps_obj &lt;-<span class="st"> </span>ps_water_mc_<span class="dv">100</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="co">#summarize_phyloseq(ps_water_o)</span></span>
<span id="cb29-2"><a href="#cb29-2"></a><span class="co">#summarize_phyloseq(ps_water_mc)</span></span>
<span id="cb29-3"><a href="#cb29-3"></a>tax_tab1 &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">tax_table</span>(ps_obj))</span>
<span id="cb29-4"><a href="#cb29-4"></a>tax_tab1 &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(tax_tab1, <span class="st">&quot;otu_id&quot;</span>)</span>
<span id="cb29-5"><a href="#cb29-5"></a>asv_tab1 &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">t</span>(<span class="kw">otu_table</span>(ps_obj)))</span>
<span id="cb29-6"><a href="#cb29-6"></a>asv_tab1 &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(asv_tab1, <span class="st">&quot;otu_id&quot;</span>)</span>
<span id="cb29-7"><a href="#cb29-7"></a>sam_tab1 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(ps_obj))</span>
<span id="cb29-8"><a href="#cb29-8"></a>sam_tab1[<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb29-9"><a href="#cb29-9"></a>da_samp &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(sam_tab1, <span class="st">&quot;sample_id&quot;</span>)</span>
<span id="cb29-10"><a href="#cb29-10"></a>water_mc &lt;-<span class="st"> </span><span class="kw">merge</span>(asv_tab1, tax_tab1, <span class="dt">by=</span><span class="st">&quot;otu_id&quot;</span>)</span>
<span id="cb29-11"><a href="#cb29-11"></a>water_mc<span class="op">$</span>ASV_SEQ &lt;-<span class="st"> </span><span class="ot">NULL</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>obj &lt;-<span class="st"> </span><span class="kw">parse_tax_data</span>(water_mc, <span class="dt">class_cols =</span> <span class="kw">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb30-2"><a href="#cb30-2"></a>                                             <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb30-3"><a href="#cb30-3"></a>                                             <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;ASV_ID&quot;</span> ))</span>
<span id="cb30-4"><a href="#cb30-4"></a>obj<span class="op">$</span>data<span class="op">$</span>tax_abund &lt;-<span class="st"> </span><span class="kw">calc_taxon_abund</span>(obj, <span class="st">&quot;tax_data&quot;</span>,</span>
<span id="cb30-5"><a href="#cb30-5"></a>                                       <span class="dt">cols =</span> da_samp<span class="op">$</span>sample_id)</span>
<span id="cb30-6"><a href="#cb30-6"></a>obj<span class="op">$</span>data<span class="op">$</span>tax_occ &lt;-<span class="st"> </span><span class="kw">calc_n_samples</span>(obj, <span class="st">&quot;tax_abund&quot;</span>,</span>
<span id="cb30-7"><a href="#cb30-7"></a>                                   <span class="dt">groups =</span> da_samp<span class="op">$</span>REGION,</span>
<span id="cb30-8"><a href="#cb30-8"></a>                                   <span class="dt">cols =</span> da_samp<span class="op">$</span>sample_id)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>obj<span class="op">$</span>data<span class="op">$</span>diff_table &lt;-<span class="st"> </span><span class="kw">compare_groups</span>(obj,</span>
<span id="cb31-2"><a href="#cb31-2"></a>                                      <span class="dt">data =</span> <span class="st">&quot;tax_abund&quot;</span>,</span>
<span id="cb31-3"><a href="#cb31-3"></a>                                      <span class="dt">cols =</span> da_samp<span class="op">$</span>sample_id,</span>
<span id="cb31-4"><a href="#cb31-4"></a>                                      <span class="dt">groups =</span> da_samp<span class="op">$</span>REGION)</span></code></pre></div>
</div>
<h2 id="bacteria">Bacteria</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a><span class="kw">range.default</span>(obj<span class="op">$</span>data<span class="op">$</span>diff_table<span class="op">$</span>log2_median_ratio, <span class="dt">finite=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>[1] -7.957102  8.071462</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="co">#set.seed(1999)</span></span>
<span id="cb34-2"><a href="#cb34-2"></a>obj <span class="op">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3"></a><span class="st">  </span><span class="kw">filter_taxa</span>(taxon_names <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Bacteria&quot;</span>), <span class="dt">subtaxa =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb34-4"><a href="#cb34-4"></a><span class="co">#  filter_taxa(taxon_ranks == &quot;f&quot;, supertaxa = TRUE) %&gt;% # subset to the order rank</span></span>
<span id="cb34-5"><a href="#cb34-5"></a><span class="co">#  filter_taxa(taxon_names %in% c(&quot;Proteobacteria&quot;, &quot;Bacteroidia&quot;, &quot;Archaea&quot;), subtaxa = TRUE, invert = TRUE) # to remove taxa</span></span>
<span id="cb34-6"><a href="#cb34-6"></a><span class="st">  </span><span class="kw">heat_tree</span>(</span>
<span id="cb34-7"><a href="#cb34-7"></a>          <span class="dt">node_label =</span> taxon_names,</span>
<span id="cb34-8"><a href="#cb34-8"></a>          <span class="dt">node_size =</span> n_obs,</span>
<span id="cb34-9"><a href="#cb34-9"></a>         <span class="co"># node_size_range = c(0.01, 0.05),</span></span>
<span id="cb34-10"><a href="#cb34-10"></a>          <span class="dt">node_label_size_range =</span> <span class="kw">c</span>(<span class="fl">0.008</span>, <span class="fl">0.04</span>),</span>
<span id="cb34-11"><a href="#cb34-11"></a>          <span class="dt">node_color =</span> log2_median_ratio,</span>
<span id="cb34-12"><a href="#cb34-12"></a>          <span class="dt">node_color_interval =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">8</span>, <span class="dv">8</span>),</span>
<span id="cb34-13"><a href="#cb34-13"></a>          <span class="dt">edge_color_interval =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">8</span>, <span class="dv">8</span>),</span>
<span id="cb34-14"><a href="#cb34-14"></a>          <span class="dt">node_color_trans =</span> <span class="st">&quot;area&quot;</span>,</span>
<span id="cb34-15"><a href="#cb34-15"></a>          <span class="dt">node_color_range =</span> <span class="kw">c</span>(<span class="st">&quot;#0072B2&quot;</span>, <span class="st">&quot;gray&quot;</span>, <span class="st">&quot;#D55E00&quot;</span>),</span>
<span id="cb34-16"><a href="#cb34-16"></a>          <span class="dt">node_size_axis_label =</span> <span class="st">&quot;OTU count&quot;</span>,</span>
<span id="cb34-17"><a href="#cb34-17"></a>          <span class="dt">node_color_axis_label =</span> <span class="st">&quot;Log 2 median ratio&quot;</span>,</span>
<span id="cb34-18"><a href="#cb34-18"></a>          <span class="dt">layout =</span> <span class="st">&quot;da&quot;</span>,</span>
<span id="cb34-19"><a href="#cb34-19"></a>          <span class="dt">initial_layout =</span> <span class="st">&quot;re&quot;</span>,</span>
<span id="cb34-20"><a href="#cb34-20"></a>          <span class="dt">overlap_avoidance =</span> <span class="dv">2</span>,</span>
<span id="cb34-21"><a href="#cb34-21"></a>          <span class="dt">output_file =</span> <span class="st">&quot;files/trepo/taxa/figures/differential_heat_tree_merged_Bacteria.png&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/trepo/taxa/differential_heat_tree_merged_Bacteria.png" width="1500" /></p>
</div>
<h2 id="archaea">Archaea</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>arch_only &lt;-<span class="st"> </span>obj <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter_taxa</span>(taxon_names <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Archaea&quot;</span>), <span class="dt">subtaxa =</span> <span class="ot">TRUE</span>)</span>
<span id="cb35-2"><a href="#cb35-2"></a><span class="kw">range.default</span>(arch_only<span class="op">$</span>data<span class="op">$</span>diff_table<span class="op">$</span>log2_median_ratio, <span class="dt">finite=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>[1] -4.990307  7.108524</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a><span class="co">#set.seed(1999)</span></span>
<span id="cb37-2"><a href="#cb37-2"></a>obj <span class="op">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3"></a><span class="st">  </span><span class="kw">filter_taxa</span>(taxon_names <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Archaea&quot;</span>), <span class="dt">subtaxa =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb37-4"><a href="#cb37-4"></a><span class="co">#  filter_taxa(taxon_ranks == &quot;o&quot;, supertaxa = TRUE) %&gt;% # subset to the order rank</span></span>
<span id="cb37-5"><a href="#cb37-5"></a><span class="co">#  filter_taxa(taxon_names %in% c(&quot;Proteobacteria&quot;, &quot;Bacteroidia&quot;, &quot;Archaea&quot;), subtaxa = TRUE, invert = TRUE) # to remove taxa</span></span>
<span id="cb37-6"><a href="#cb37-6"></a><span class="st">  </span><span class="kw">heat_tree</span>(</span>
<span id="cb37-7"><a href="#cb37-7"></a>          <span class="dt">node_label =</span> taxon_names,</span>
<span id="cb37-8"><a href="#cb37-8"></a>          <span class="dt">node_size =</span> n_obs,</span>
<span id="cb37-9"><a href="#cb37-9"></a>         <span class="co"># node_size_range = c(0.01, 0.05),</span></span>
<span id="cb37-10"><a href="#cb37-10"></a>          <span class="dt">node_label_size_range =</span> <span class="kw">c</span>(<span class="fl">0.008</span>, <span class="fl">0.04</span>),</span>
<span id="cb37-11"><a href="#cb37-11"></a>          <span class="dt">node_color =</span> log2_median_ratio,</span>
<span id="cb37-12"><a href="#cb37-12"></a>          <span class="dt">node_color_interval =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">5</span>, <span class="dv">7</span>),</span>
<span id="cb37-13"><a href="#cb37-13"></a>          <span class="dt">edge_color_interval =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">5</span>, <span class="dv">7</span>),</span>
<span id="cb37-14"><a href="#cb37-14"></a>          <span class="dt">node_color_trans =</span> <span class="st">&quot;area&quot;</span>,</span>
<span id="cb37-15"><a href="#cb37-15"></a>          <span class="dt">node_color_range =</span> <span class="kw">c</span>(<span class="st">&quot;#0072B2&quot;</span>, <span class="st">&quot;gray&quot;</span>, <span class="st">&quot;#D55E00&quot;</span>),</span>
<span id="cb37-16"><a href="#cb37-16"></a>          <span class="dt">node_size_axis_label =</span> <span class="st">&quot;OTU count&quot;</span>,</span>
<span id="cb37-17"><a href="#cb37-17"></a>          <span class="dt">node_color_axis_label =</span> <span class="st">&quot;Log 2 median ratio&quot;</span>,</span>
<span id="cb37-18"><a href="#cb37-18"></a>          <span class="dt">layout =</span> <span class="st">&quot;da&quot;</span>,</span>
<span id="cb37-19"><a href="#cb37-19"></a>          <span class="dt">initial_layout =</span> <span class="st">&quot;re&quot;</span>,</span>
<span id="cb37-20"><a href="#cb37-20"></a>          <span class="dt">overlap_avoidance =</span> <span class="dv">2</span>,</span>
<span id="cb37-21"><a href="#cb37-21"></a>          <span class="dt">output_file =</span> <span class="st">&quot;files/trepo/taxa/figures/differential_heat_tree_merged_Archaea.png&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/trepo/taxa/differential_heat_tree_merged_Archaea.png" width="1500" /></p>
</div>
<h2 id="top-phyla">Top Phyla</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>top_phyl &lt;-<span class="st"> </span>obj <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter_taxa</span>(taxon_names <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Proteobacteria&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, <span class="st">&quot;Chloroflexi&quot;</span>, <span class="st">&quot;Planctomycetota&quot;</span>),  <span class="dt">subtaxa =</span> <span class="ot">TRUE</span>)</span>
<span id="cb38-2"><a href="#cb38-2"></a><span class="kw">range.default</span>(top_phyl<span class="op">$</span>data<span class="op">$</span>diff_table<span class="op">$</span>log2_median_ratio, <span class="dt">finite=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>[1] -6.505150  8.071462</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a><span class="co">#set.seed(10)</span></span>
<span id="cb40-2"><a href="#cb40-2"></a>obj <span class="op">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3"></a><span class="st">  </span><span class="kw">filter_taxa</span>(taxon_names <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Proteobacteria&quot;</span>, <span class="st">&quot;Acidobacteriota&quot;</span>, <span class="st">&quot;Chloroflexi&quot;</span>, <span class="st">&quot;Planctomycetota&quot;</span>),</span>
<span id="cb40-4"><a href="#cb40-4"></a>              <span class="dt">subtaxa =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb40-5"><a href="#cb40-5"></a><span class="st">  </span><span class="kw">heat_tree</span>(<span class="dt">node_label =</span> taxon_names,</span>
<span id="cb40-6"><a href="#cb40-6"></a>            <span class="dt">node_size =</span> n_obs,</span>
<span id="cb40-7"><a href="#cb40-7"></a>            <span class="dt">node_color =</span> log2_median_ratio,</span>
<span id="cb40-8"><a href="#cb40-8"></a>            <span class="dt">node_color_range =</span> <span class="kw">c</span>(<span class="st">&quot;#0072B2&quot;</span>, <span class="st">&quot;gray&quot;</span>, <span class="st">&quot;#D55E00&quot;</span>),</span>
<span id="cb40-9"><a href="#cb40-9"></a>            <span class="dt">node_color_interval =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">3</span>, <span class="dv">3</span>),</span>
<span id="cb40-10"><a href="#cb40-10"></a>            <span class="co">#edge_color_interval = c(-5.5, 4),</span></span>
<span id="cb40-11"><a href="#cb40-11"></a>            <span class="co">#node_color_trans = &quot;area&quot;,</span></span>
<span id="cb40-12"><a href="#cb40-12"></a>            <span class="dt">tree_label =</span> taxon_names,</span>
<span id="cb40-13"><a href="#cb40-13"></a>            <span class="dt">initial_layout =</span> <span class="st">&quot;re&quot;</span>, <span class="dt">layout =</span> <span class="st">&quot;da&quot;</span>,</span>
<span id="cb40-14"><a href="#cb40-14"></a>            <span class="dt">node_color_axis_label =</span> <span class="st">&quot;Sum of root reads&quot;</span>,</span>
<span id="cb40-15"><a href="#cb40-15"></a>            <span class="dt">node_size_axis_label =</span> <span class="st">&quot;Number of OTUs&quot;</span>,</span>
<span id="cb40-16"><a href="#cb40-16"></a>            <span class="co">#overlap_avoidance = 2,</span></span>
<span id="cb40-17"><a href="#cb40-17"></a>            <span class="dt">output_file =</span> <span class="st">&quot;files/trepo/taxa/figures/differential_heat_tree_merged_phyla.png&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/trepo/taxa/differential_heat_tree_merged_phyla.png" width="1500" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a><span class="kw">rm</span>(ps_work_merge)</span>
<span id="cb41-2"><a href="#cb41-2"></a><span class="kw">objects</span>()</span>
<span id="cb41-3"><a href="#cb41-3"></a><span class="kw">save.image</span>(<span class="st">&quot;page_build/trepo/taxa_ssu_wf_part_3.rdata&quot;</span>)</span></code></pre></div>
</div>
<details>
<p><summary>Scratch Code</summary></p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a><span class="kw">load</span>(<span class="st">&quot;files/trepo/data-prep/rdata/ssu_merge_taxmap.rdata&quot;</span>)</span>
<span id="cb42-2"><a href="#cb42-2"></a><span class="kw">objects</span>()</span>
<span id="cb42-3"><a href="#cb42-3"></a></span>
<span id="cb42-4"><a href="#cb42-4"></a></span>
<span id="cb42-5"><a href="#cb42-5"></a>obj_merge <span class="op">%&gt;%</span></span>
<span id="cb42-6"><a href="#cb42-6"></a><span class="st">  </span><span class="kw">filter_taxa</span>(<span class="kw">grepl</span>(<span class="dt">pattern =</span> <span class="st">&quot;^[a-zA-Z]+$&quot;</span>, taxon_names)) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># remove &quot;odd&quot; taxa</span></span>
<span id="cb42-7"><a href="#cb42-7"></a><span class="st">  </span><span class="kw">filter_taxa</span>(taxon_ranks <span class="op">==</span><span class="st"> &quot;o&quot;</span>, <span class="dt">supertaxa =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># subset to the order rank</span></span>
<span id="cb42-8"><a href="#cb42-8"></a><span class="st">  </span><span class="kw">heat_tree</span>(<span class="dt">node_label =</span> <span class="kw">gsub</span>(<span class="dt">pattern =</span> <span class="st">&quot;</span><span class="ch">\\</span><span class="st">[|</span><span class="ch">\\</span><span class="st">]&quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;&quot;</span>, taxon_names),</span>
<span id="cb42-9"><a href="#cb42-9"></a>            <span class="dt">node_size =</span> n_obs,</span>
<span id="cb42-10"><a href="#cb42-10"></a>            <span class="dt">node_color =</span> n_obs,</span>
<span id="cb42-11"><a href="#cb42-11"></a>            <span class="dt">node_color_axis_label =</span> <span class="st">&quot;OTU count&quot;</span>,</span>
<span id="cb42-12"><a href="#cb42-12"></a>            <span class="dt">layout =</span> <span class="st">&quot;davidson-harel&quot;</span>, <span class="dt">initial_layout =</span> <span class="st">&quot;reingold-tilford&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a><span class="kw">heat_tree</span>(obj_merge)</span>
<span id="cb43-2"><a href="#cb43-2"></a>obj<span class="op">$</span><span class="kw">taxon_names</span>()</span>
<span id="cb43-3"><a href="#cb43-3"></a>ps_obj_mc_obj<span class="op">$</span><span class="kw">n_obs_1</span>()</span>
<span id="cb43-4"><a href="#cb43-4"></a><span class="kw">heat_tree</span>(obj,</span>
<span id="cb43-5"><a href="#cb43-5"></a>          <span class="dt">node_label =</span> taxon_names,</span>
<span id="cb43-6"><a href="#cb43-6"></a>          <span class="dt">node_size =</span> n_obs,</span>
<span id="cb43-7"><a href="#cb43-7"></a>          <span class="dt">node_color =</span> roots)</span>
<span id="cb43-8"><a href="#cb43-8"></a></span>
<span id="cb43-9"><a href="#cb43-9"></a>obj<span class="op">$</span>data<span class="op">$</span>tax_abund &lt;-<span class="st"> </span><span class="kw">calc_taxon_abund</span>(obj, <span class="st">&quot;otu_counts&quot;</span>,</span>
<span id="cb43-10"><a href="#cb43-10"></a>                                       <span class="dt">cols =</span> sample_data<span class="op">$</span>SampleID,</span>
<span id="cb43-11"><a href="#cb43-11"></a>                                       <span class="dt">groups =</span> sample_data<span class="op">$</span>Type)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a>ssu_make_metac &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ps_obj&quot;</span>)</span>
<span id="cb44-2"><a href="#cb44-2"></a><span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb44-3"><a href="#cb44-3"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_make_metac){</span>
<span id="cb44-4"><a href="#cb44-4"></a>        tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb44-5"><a href="#cb44-5"></a>        tmp_tax &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">tax_table</span>(tmp_get))</span>
<span id="cb44-6"><a href="#cb44-6"></a>        tmp_tax &lt;-<span class="st"> </span>tmp_tax <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;otu_id&quot;</span>)</span>
<span id="cb44-7"><a href="#cb44-7"></a>        tmp_asv &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">t</span>(<span class="kw">otu_table</span>(tmp_get)))</span>
<span id="cb44-8"><a href="#cb44-8"></a>        tmp_asv &lt;-<span class="st"> </span>tmp_asv <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;otu_id&quot;</span>)</span>
<span id="cb44-9"><a href="#cb44-9"></a>        tmp_sam &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(tmp_get))</span>
<span id="cb44-10"><a href="#cb44-10"></a>        tmp_sam[<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb44-11"><a href="#cb44-11"></a>        tmp_sam &lt;-<span class="st"> </span>tmp_sam <span class="op">%&gt;%</span><span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;sample_id&quot;</span>)</span>
<span id="cb44-12"><a href="#cb44-12"></a>        tmp_asv_tax &lt;-<span class="st"> </span><span class="kw">merge</span>(tmp_asv, tmp_tax, <span class="dt">by=</span><span class="st">&quot;otu_id&quot;</span>)</span>
<span id="cb44-13"><a href="#cb44-13"></a>        tmp_asv_tax<span class="op">$</span>ASV_SEQ &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb44-14"><a href="#cb44-14"></a>        tmp_asv_tax<span class="op">$</span>ASV_ID &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb44-15"><a href="#cb44-15"></a>        tmp_obj &lt;-<span class="st"> </span><span class="kw">parse_tax_data</span>(tmp_asv_tax,</span>
<span id="cb44-16"><a href="#cb44-16"></a>                                  <span class="dt">class_cols =</span> <span class="kw">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>,<span class="st">&quot;Class&quot;</span>,</span>
<span id="cb44-17"><a href="#cb44-17"></a>                                                 <span class="st">&quot;Order&quot;</span>,<span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span></span>
<span id="cb44-18"><a href="#cb44-18"></a>                                                 ))</span>
<span id="cb44-19"><a href="#cb44-19"></a>        tmp_obj<span class="op">$</span>data<span class="op">$</span>tax_group_abund &lt;-<span class="st"> </span><span class="kw">calc_taxon_abund</span>(tmp_obj, <span class="st">&quot;tax_data&quot;</span>,</span>
<span id="cb44-20"><a href="#cb44-20"></a>                                       <span class="dt">cols =</span> tmp_sam<span class="op">$</span>sample_id,</span>
<span id="cb44-21"><a href="#cb44-21"></a>                                       <span class="dt">groups =</span> tmp_sam<span class="op">$</span>TEMP)</span>
<span id="cb44-22"><a href="#cb44-22"></a>        tmp_obj<span class="op">$</span>data<span class="op">$</span>tax_total_abund &lt;-<span class="st"> </span><span class="kw">calc_taxon_abund</span>(tmp_obj, <span class="st">&quot;tax_data&quot;</span>,</span>
<span id="cb44-23"><a href="#cb44-23"></a>                                       <span class="dt">cols =</span> tmp_sam<span class="op">$</span>sample_id,</span>
<span id="cb44-24"><a href="#cb44-24"></a>                                       <span class="dt">groups =</span> tmp_sam<span class="op">$</span>DEPTH,</span>
<span id="cb44-25"><a href="#cb44-25"></a>                                       <span class="dt">out_names =</span> <span class="st">&quot;n_reads&quot;</span>)</span>
<span id="cb44-26"><a href="#cb44-26"></a>        tmp_obj<span class="op">$</span>data<span class="op">$</span>tax_samp_abund &lt;-<span class="st"> </span><span class="kw">calc_taxon_abund</span>(tmp_obj, <span class="st">&quot;tax_data&quot;</span>,</span>
<span id="cb44-27"><a href="#cb44-27"></a>                                       <span class="dt">cols =</span> tmp_sam<span class="op">$</span>sample_id)</span>
<span id="cb44-28"><a href="#cb44-28"></a>        tmp_obj<span class="op">$</span>data<span class="op">$</span>tax_samp_occ &lt;-<span class="st"> </span><span class="kw">calc_n_samples</span>(tmp_obj, <span class="st">&quot;tax_samp_abund&quot;</span>,</span>
<span id="cb44-29"><a href="#cb44-29"></a>                                   <span class="dt">groups =</span> tmp_sam<span class="op">$</span>TEMP,</span>
<span id="cb44-30"><a href="#cb44-30"></a>                                   <span class="dt">cols =</span> tmp_sam<span class="op">$</span>sample_id)</span>
<span id="cb44-31"><a href="#cb44-31"></a>        tmp_obj<span class="op">$</span>data<span class="op">$</span>diff_table &lt;-<span class="st"> </span><span class="kw">compare_groups</span>(tmp_obj,</span>
<span id="cb44-32"><a href="#cb44-32"></a>                                      <span class="dt">data =</span> <span class="st">&quot;tax_samp_abund&quot;</span>,</span>
<span id="cb44-33"><a href="#cb44-33"></a>                                      <span class="dt">cols =</span> tmp_sam<span class="op">$</span>sample_id,</span>
<span id="cb44-34"><a href="#cb44-34"></a>                                      <span class="dt">groups =</span> tmp_sam<span class="op">$</span>TEMP)</span>
<span id="cb44-35"><a href="#cb44-35"></a>        tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_mc_obj&quot;</span>))</span>
<span id="cb44-36"><a href="#cb44-36"></a>        <span class="kw">assign</span>(tmp_name, tmp_obj)</span>
<span id="cb44-37"><a href="#cb44-37"></a>        tmp_name2 &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_mc_samp_data&quot;</span>))</span>
<span id="cb44-38"><a href="#cb44-38"></a>        <span class="kw">assign</span>(tmp_name2, tmp_sam)</span>
<span id="cb44-39"><a href="#cb44-39"></a>        <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb44-40"><a href="#cb44-40"></a>}</span>
<span id="cb44-41"><a href="#cb44-41"></a><span class="kw">objects</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a>ps_obj_mc_obj<span class="op">$</span>data<span class="op">$</span>tax_sam</span>
<span id="cb45-2"><a href="#cb45-2"></a><span class="co">#ps_obj_mc_obj$data$diff_table</span></span>
<span id="cb45-3"><a href="#cb45-3"></a><span class="co">#write.table(ps_obj_mc_obj$data$diff_table, &quot;test.txt&quot;, quote = FALSE, sep = &quot;\t&quot;)</span></span>
<span id="cb45-4"><a href="#cb45-4"></a><span class="co">#ps_obj_mc_obj$data$diff_table</span></span>
<span id="cb45-5"><a href="#cb45-5"></a><span class="co">#set.seed(1999)</span></span>
<span id="cb45-6"><a href="#cb45-6"></a>ps_obj_mc_obj <span class="op">%&gt;%</span></span>
<span id="cb45-7"><a href="#cb45-7"></a><span class="st">  </span><span class="kw">filter_taxa</span>(taxon_names <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Bacteria&quot;</span>), <span class="dt">subtaxa =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb45-8"><a href="#cb45-8"></a><span class="co">#  filter_taxa(taxon_ranks == &quot;o&quot;, supertaxa = TRUE) %&gt;% # subset to the order rank</span></span>
<span id="cb45-9"><a href="#cb45-9"></a><span class="co">#  filter_taxa(taxon_names %in% c(&quot;Alphaproteobacteria&quot;, &quot;Firmicutes&quot;, &quot;Gammaproteobacteria&quot;), subtaxa = TRUE, invert = FALSE) %&gt;% #</span></span>
<span id="cb45-10"><a href="#cb45-10"></a></span>
<span id="cb45-11"><a href="#cb45-11"></a><span class="st">  </span><span class="kw">heat_tree</span>(<span class="dt">node_label =</span> taxon_names,</span>
<span id="cb45-12"><a href="#cb45-12"></a>            <span class="dt">node_size =</span> n_obs,</span>
<span id="cb45-13"><a href="#cb45-13"></a>            <span class="dt">node_size_range =</span> <span class="kw">c</span>(<span class="fl">0.01</span>, <span class="fl">0.05</span>),</span>
<span id="cb45-14"><a href="#cb45-14"></a>            <span class="dt">edge_size_range =</span> <span class="kw">c</span>(<span class="fl">0.005</span>, <span class="fl">0.005</span>),</span>
<span id="cb45-15"><a href="#cb45-15"></a>            <span class="dt">node_color =</span> W0,</span>
<span id="cb45-16"><a href="#cb45-16"></a>            <span class="dt">initial_layout =</span> <span class="st">&quot;re&quot;</span>, <span class="dt">layout =</span> <span class="st">&quot;da&quot;</span>,</span>
<span id="cb45-17"><a href="#cb45-17"></a>            <span class="dt">title =</span> <span class="st">&quot;Root sample read depth&quot;</span>,</span>
<span id="cb45-18"><a href="#cb45-18"></a>            <span class="dt">node_color_axis_label =</span> <span class="st">&quot;Sum of root reads&quot;</span>,</span>
<span id="cb45-19"><a href="#cb45-19"></a>            <span class="dt">node_size_axis_label =</span> <span class="st">&quot;Number of OTUs&quot;</span>,</span>
<span id="cb45-20"><a href="#cb45-20"></a>            <span class="dt">output_file =</span> <span class="st">&quot;plot_example0.pdf&quot;</span>)</span>
<span id="cb45-21"><a href="#cb45-21"></a></span>
<span id="cb45-22"><a href="#cb45-22"></a>  <span class="kw">heat_tree</span>(</span>
<span id="cb45-23"><a href="#cb45-23"></a>          <span class="dt">node_size =</span> W8,</span>
<span id="cb45-24"><a href="#cb45-24"></a>          <span class="dt">node_label =</span> <span class="kw">paste0</span>(taxon_names, <span class="st">&quot; (&quot;</span>, n_obs, <span class="st">&quot;)&quot;</span> ) ,</span>
<span id="cb45-25"><a href="#cb45-25"></a>          <span class="dt">node_size_range =</span> <span class="kw">c</span>(<span class="fl">0.01</span>, <span class="fl">0.04</span>),</span>
<span id="cb45-26"><a href="#cb45-26"></a>          <span class="co">#node_label_size_range = c(0.008, 0.04),</span></span>
<span id="cb45-27"><a href="#cb45-27"></a>          <span class="dt">node_color =</span> W8,</span>
<span id="cb45-28"><a href="#cb45-28"></a>          <span class="co">#node_color_interval = c(0, 0.05),</span></span>
<span id="cb45-29"><a href="#cb45-29"></a>          <span class="co">#edge_color_interval = c(0, 0.05),</span></span>
<span id="cb45-30"><a href="#cb45-30"></a>          <span class="co">#node_color_trans = &quot;area&quot;,</span></span>
<span id="cb45-31"><a href="#cb45-31"></a>          <span class="co">#node_color_range = c(&quot;#D55E00&quot;, &quot;gray&quot;, &quot;#0072B2&quot;),</span></span>
<span id="cb45-32"><a href="#cb45-32"></a>          <span class="dt">node_size_axis_label =</span> <span class="st">&quot;OTU count&quot;</span>,</span>
<span id="cb45-33"><a href="#cb45-33"></a>          <span class="dt">node_color_axis_label =</span> <span class="st">&quot;Log 2 median ratio&quot;</span>,</span>
<span id="cb45-34"><a href="#cb45-34"></a>          <span class="dt">layout =</span> <span class="st">&quot;da&quot;</span>,</span>
<span id="cb45-35"><a href="#cb45-35"></a>          <span class="dt">initial_layout =</span> <span class="st">&quot;re&quot;</span>,</span>
<span id="cb45-36"><a href="#cb45-36"></a>          <span class="dt">overlap_avoidance =</span> <span class="dv">2</span>,</span>
<span id="cb45-37"><a href="#cb45-37"></a>          <span class="dt">output_file =</span> <span class="st">&quot;differential_heat_tree.pdf&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a>swel_col &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#ff6db6&quot;</span>, <span class="st">&quot;#24ff24&quot;</span>, <span class="st">&quot;#db6d00&quot;</span>)</span>
<span id="cb46-2"><a href="#cb46-2"></a><span class="kw">write.table</span>(ps_obj_mc_obj<span class="op">$</span>data<span class="op">$</span>diff_table, <span class="st">&quot;test.txt&quot;</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb46-3"><a href="#cb46-3"></a>ps_obj_mc_obj<span class="op">$</span><span class="kw">n_obs</span>()</span>
<span id="cb46-4"><a href="#cb46-4"></a>ps_obj_mc_obj <span class="op">%&gt;%</span></span>
<span id="cb46-5"><a href="#cb46-5"></a><span class="kw">filter_taxa</span>(taxon_names <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Bacteria&quot;</span>), <span class="dt">subtaxa =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb46-6"><a href="#cb46-6"></a><span class="kw">heat_tree_matrix</span>(</span>
<span id="cb46-7"><a href="#cb46-7"></a>                 <span class="dt">data =</span> <span class="st">&quot;diff_table&quot;</span>,</span>
<span id="cb46-8"><a href="#cb46-8"></a>                 <span class="dt">node_label =</span> taxon_names,</span>
<span id="cb46-9"><a href="#cb46-9"></a>                 <span class="dt">node_size =</span> n_obs_<span class="dv">1</span>, <span class="co"># n_obs is a function that calculates the number of OTUs per taxon</span></span>
<span id="cb46-10"><a href="#cb46-10"></a>                 <span class="co">#node_label_size_range = c(0.008, 0.04),</span></span>
<span id="cb46-11"><a href="#cb46-11"></a>                 <span class="dt">node_color =</span> roots, <span class="co"># A column from `obj$data$diff_table`</span></span>
<span id="cb46-12"><a href="#cb46-12"></a>                 <span class="co">#node_color_range = swel_col, # The built-in palette for diverging data</span></span>
<span id="cb46-13"><a href="#cb46-13"></a>                 <span class="co">#node_color_trans = &quot;linear&quot;, # The default is scaled by circle area</span></span>
<span id="cb46-14"><a href="#cb46-14"></a>                 <span class="co">#node_color_interval = c(0, 0.05),  # The range of `log2_median_ratio` to display</span></span>
<span id="cb46-15"><a href="#cb46-15"></a>                 <span class="co">#edge_color_interval = c(0, 0.05), # The range of `log2_median_ratio` to display</span></span>
<span id="cb46-16"><a href="#cb46-16"></a>                 <span class="dt">node_size_axis_label =</span> <span class="st">&quot;Number of OTUs&quot;</span>,</span>
<span id="cb46-17"><a href="#cb46-17"></a>                 <span class="dt">node_color_axis_label =</span> <span class="st">&quot;Log2 ratio median proportions&quot;</span>,</span>
<span id="cb46-18"><a href="#cb46-18"></a>                 <span class="dt">layout =</span> <span class="st">&quot;fr&quot;</span>, <span class="co"># The primary layout algorithm</span></span>
<span id="cb46-19"><a href="#cb46-19"></a>                 <span class="dt">initial_layout =</span> <span class="st">&quot;reingold-tilford&quot;</span>, <span class="co"># The layout algorithm that initializes node locations</span></span>
<span id="cb46-20"><a href="#cb46-20"></a>                 <span class="dt">output_file =</span> <span class="st">&quot;differential_heat_tree2.pdf&quot;</span>) <span class="co"># Saves the plot as a png file</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_data_sets){</span>
<span id="cb47-2"><a href="#cb47-2"></a>    tmp_clean &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_proteo_clean&quot;</span>))</span>
<span id="cb47-3"><a href="#cb47-3"></a>    tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_clean)</span>
<span id="cb47-4"><a href="#cb47-4"></a>    tmp_ps &lt;-<span class="st"> </span><span class="kw">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) otu<span class="op">/</span><span class="kw">sum</span>(otu))</span>
<span id="cb47-5"><a href="#cb47-5"></a>    tmp_ps<span class="op">@</span>phy_tree &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb47-6"><a href="#cb47-6"></a>    tmp_prune &lt;-<span class="st"> </span><span class="kw">prune_samples</span>(<span class="kw">sample_sums</span>(tmp_ps) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, tmp_ps)</span>
<span id="cb47-7"><a href="#cb47-7"></a>    tmp_tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="kw">ntaxa</span>(tmp_prune), <span class="dt">rooted =</span> <span class="ot">TRUE</span>, <span class="dt">tip.label =</span> <span class="kw">taxa_names</span>(tmp_prune))</span>
<span id="cb47-8"><a href="#cb47-8"></a>    tmp_ps_merge &lt;-<span class="st"> </span><span class="kw">merge_phyloseq</span>(tmp_prune,</span>
<span id="cb47-9"><a href="#cb47-9"></a>                               sample_data,</span>
<span id="cb47-10"><a href="#cb47-10"></a>                               tmp_tree)</span>
<span id="cb47-11"><a href="#cb47-11"></a>    tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_proteo_clean_trans&quot;</span>))</span>
<span id="cb47-12"><a href="#cb47-12"></a>    <span class="kw">assign</span>(tmp_name, tmp_ps_merge)</span>
<span id="cb47-13"><a href="#cb47-13"></a>    <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb47-14"><a href="#cb47-14"></a>}</span>
<span id="cb47-15"><a href="#cb47-15"></a><span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_agg&quot;</span>) ssu_ps_pime_clean_Phylum_agg</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a>carbom_chloro &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(ssu_ps_pime_proteo_clean_trans, Phylum <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Firmicutes&quot;</span>))</span>
<span id="cb48-2"><a href="#cb48-2"></a>testplot &lt;-<span class="st"> </span><span class="kw">plot_bar</span>(carbom_chloro, <span class="dt">x=</span><span class="st">&quot;Class&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;Class&quot;</span>, <span class="dt">facet_grid =</span> <span class="op">~</span>TEMP) <span class="op">+</span></span>
<span id="cb48-3"><a href="#cb48-3"></a><span class="st">            </span><span class="kw">geom_bar</span>(<span class="kw">aes</span>(<span class="dt">color=</span>Class, <span class="dt">fill=</span>Class), <span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span><span class="st">&quot;stack&quot;</span>)</span>
<span id="cb48-4"><a href="#cb48-4"></a></span>
<span id="cb48-5"><a href="#cb48-5"></a>carbom_chloro &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(ssu_ps_pime_proteo_clean_trans, Phylum <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Alphaproteobacteria&quot;</span>, <span class="st">&quot;Gammaproteobacteria&quot;</span>, <span class="st">&quot;Firmicutes&quot;</span>))</span>
<span id="cb48-6"><a href="#cb48-6"></a>carbom.ord &lt;-<span class="st"> </span><span class="kw">ordinate</span>(carbom_chloro, <span class="st">&quot;NMDS&quot;</span>, <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb48-7"><a href="#cb48-7"></a>testplot &lt;-<span class="st"> </span><span class="kw">plot_ordination</span>(ssu_ps_work_proteo_clean_trans, carbom.ord, <span class="dt">type=</span><span class="st">&quot;taxa&quot;</span>, <span class="dt">color=</span><span class="st">&quot;Family&quot;</span>,</span>
<span id="cb48-8"><a href="#cb48-8"></a>                  <span class="dt">title=</span><span class="st">&quot;OTUs&quot;</span>, <span class="dt">label=</span><span class="st">&quot;Family&quot;</span>) <span class="op">+</span></span>
<span id="cb48-9"><a href="#cb48-9"></a><span class="st">                  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>Phylum)</span>
<span id="cb48-10"><a href="#cb48-10"></a></span>
<span id="cb48-11"><a href="#cb48-11"></a></span>
<span id="cb48-12"><a href="#cb48-12"></a>testplot</span>
<span id="cb48-13"><a href="#cb48-13"></a><span class="kw">dev.off</span>()</span>
<span id="cb48-14"><a href="#cb48-14"></a><span class="kw">png</span>(<span class="st">&quot;figures/trepo-taxa/ssu/testplot.png&quot;</span>, <span class="dt">height =</span> <span class="dv">20</span>, <span class="dt">width =</span> <span class="dv">70</span>,</span>
<span id="cb48-15"><a href="#cb48-15"></a>    <span class="dt">units =</span> <span class="st">&#39;cm&#39;</span>, <span class="dt">res =</span> <span class="dv">600</span>, <span class="dt">bg =</span> <span class="st">&quot;white&quot;</span>)</span>
<span id="cb48-16"><a href="#cb48-16"></a>testplot</span>
<span id="cb48-17"><a href="#cb48-17"></a><span class="kw">dev.off</span>()</span>
<span id="cb48-18"><a href="#cb48-18"></a><span class="kw">pdf</span>(<span class="st">&quot;figures/trepo-taxa/ssu/testplot.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">10</span>, <span class="dt">width =</span> <span class="dv">12</span>)</span>
<span id="cb48-19"><a href="#cb48-19"></a>testplot</span>
<span id="cb48-20"><a href="#cb48-20"></a><span class="kw">dev.off</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a>subset_level &lt;-<span class="st"> &quot;Phylum&quot;</span></span>
<span id="cb49-2"><a href="#cb49-2"></a>subset_name &lt;-<span class="st"> &quot;Alphaproteobacteria&quot;</span></span>
<span id="cb49-3"><a href="#cb49-3"></a><span class="cf">for</span> ( i <span class="cf">in</span> ssu_data_sets){</span>
<span id="cb49-4"><a href="#cb49-4"></a>  tmp_get_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(., <span class="st">&quot;_proteo_clean&quot;</span>))</span>
<span id="cb49-5"><a href="#cb49-5"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_get_name)</span>
<span id="cb49-6"><a href="#cb49-6"></a>  tmp_ps &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(tmp_get, <span class="kw">get</span>(subset_level) <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(subset_name))</span>
<span id="cb49-7"><a href="#cb49-7"></a>  tmp_sub_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_clean_sub_&quot;</span>, subset_name))</span>
<span id="cb49-8"><a href="#cb49-8"></a>  <span class="kw">assign</span>(tmp_sub_name, tmp_ps)</span>
<span id="cb49-9"><a href="#cb49-9"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb49-10"><a href="#cb49-10"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a>top_hits &lt;-<span class="st"> </span><span class="dv">12</span></span>
<span id="cb50-2"><a href="#cb50-2"></a>top_level &lt;-<span class="st"> &quot;Family&quot;</span></span>
<span id="cb50-3"><a href="#cb50-3"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_data_sets){</span>
<span id="cb50-4"><a href="#cb50-4"></a>  tmp_get_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(., <span class="st">&quot;_clean_sub_&quot;</span>, subset_name))</span>
<span id="cb50-5"><a href="#cb50-5"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_get_name)</span>
<span id="cb50-6"><a href="#cb50-6"></a>  tmp_otu &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">otu_table</span>(tmp_get)))</span>
<span id="cb50-7"><a href="#cb50-7"></a>  tmp_otu[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(tmp_otu, as.numeric)</span>
<span id="cb50-8"><a href="#cb50-8"></a>  tmp_otu &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(tmp_otu)</span>
<span id="cb50-9"><a href="#cb50-9"></a>  tmp_tax &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">data.frame</span>(<span class="kw">tax_table</span>(tmp_get)))</span>
<span id="cb50-10"><a href="#cb50-10"></a>  tmp_samples &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(tmp_get))</span>
<span id="cb50-11"><a href="#cb50-11"></a>  tmp_clean_df &lt;-<span class="st"> </span><span class="kw">merge_phyloseq</span>(<span class="kw">otu_table</span>(tmp_otu, <span class="dt">taxa_are_rows =</span> <span class="ot">TRUE</span>),</span>
<span id="cb50-12"><a href="#cb50-12"></a>                          <span class="kw">tax_table</span>(tmp_tax, tmp_tax),</span>
<span id="cb50-13"><a href="#cb50-13"></a>                          <span class="kw">sample_data</span>(tmp_samples))</span>
<span id="cb50-14"><a href="#cb50-14"></a>  tmp_agg_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_clean_sub_&quot;</span>, subset_name, <span class="st">&quot;_agg&quot;</span>))</span>
<span id="cb50-15"><a href="#cb50-15"></a>  tmp_agg_df &lt;-<span class="st"> </span>microbiome<span class="op">::</span><span class="kw">aggregate_top_taxa</span>(tmp_clean_df,</span>
<span id="cb50-16"><a href="#cb50-16"></a>                                                    <span class="dt">top =</span> top_hits,</span>
<span id="cb50-17"><a href="#cb50-17"></a>                                                    <span class="dt">level =</span> top_level)</span>
<span id="cb50-18"><a href="#cb50-18"></a>  <span class="kw">assign</span>(tmp_agg_name, tmp_agg_df)</span>
<span id="cb50-19"><a href="#cb50-19"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb50-20"><a href="#cb50-20"></a>}</span>
<span id="cb50-21"><a href="#cb50-21"></a><span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_sub&quot;</span>)</span>
<span id="cb50-22"><a href="#cb50-22"></a><span class="kw">objects</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_sub.*_agg$&quot;</span>)) {</span>
<span id="cb51-2"><a href="#cb51-2"></a>  tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="st"> </span><span class="kw">paste0</span>(., <span class="st">&quot;_order&quot;</span>))</span>
<span id="cb51-3"><a href="#cb51-3"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb51-4"><a href="#cb51-4"></a>  tmp_list &lt;-<span class="st"> </span><span class="kw">get_taxa_unique</span>(tmp_get, <span class="dt">taxonomic.rank =</span> <span class="kw">rank_names</span>(tmp_get)[<span class="dv">1</span>],</span>
<span id="cb51-5"><a href="#cb51-5"></a>                        <span class="dt">errorIfNULL =</span> <span class="ot">TRUE</span>)</span>
<span id="cb51-6"><a href="#cb51-6"></a>  <span class="kw">assign</span>(tmp_name, tmp_list)</span>
<span id="cb51-7"><a href="#cb51-7"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb51-8"><a href="#cb51-8"></a>}</span>
<span id="cb51-9"><a href="#cb51-9"></a><span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_sub&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_sub.*_agg$&quot;</span>)) {</span>
<span id="cb52-2"><a href="#cb52-2"></a>  tmp_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_tax&quot;</span>))</span>
<span id="cb52-3"><a href="#cb52-3"></a>  tmp_agg &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_order&quot;</span>))</span>
<span id="cb52-4"><a href="#cb52-4"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb52-5"><a href="#cb52-5"></a>  tmp_get_agg &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_agg)</span>
<span id="cb52-6"><a href="#cb52-6"></a>  tmp_df &lt;-<span class="st"> </span>tmp_get <span class="op">%&gt;%</span></span>
<span id="cb52-7"><a href="#cb52-7"></a><span class="st">            </span><span class="kw">transform_sample_counts</span>(<span class="cf">function</span>(x) {x<span class="op">/</span><span class="kw">sum</span>(x)} ) <span class="op">%&gt;%</span></span>
<span id="cb52-8"><a href="#cb52-8"></a><span class="st">            </span><span class="kw">psmelt</span>()</span>
<span id="cb52-9"><a href="#cb52-9"></a>  tmp_df[[top_level]] &lt;-<span class="st"> </span>gdata<span class="op">::</span><span class="kw">reorder.factor</span>(tmp_df[[top_level]],</span>
<span id="cb52-10"><a href="#cb52-10"></a>                                                  <span class="dt">new.order =</span> <span class="kw">rev</span>(tmp_get_agg))</span>
<span id="cb52-11"><a href="#cb52-11"></a>  tmp_df &lt;-<span class="st"> </span>tmp_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">get</span>(top_level))</span>
<span id="cb52-12"><a href="#cb52-12"></a>  <span class="kw">assign</span>(tmp_name, tmp_df)</span>
<span id="cb52-13"><a href="#cb52-13"></a>  <span class="kw">print</span>(<span class="kw">c</span>(i,tmp_name, tmp_agg))</span>
<span id="cb52-14"><a href="#cb52-14"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb52-15"><a href="#cb52-15"></a>}</span>
<span id="cb52-16"><a href="#cb52-16"></a><span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_tax&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_sub.*_tax&quot;</span>)) {</span>
<span id="cb53-2"><a href="#cb53-2"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb53-3"><a href="#cb53-3"></a>  tmp_levels &lt;-<span class="st"> </span><span class="kw">levels</span>(tmp_get[[top_level]])</span>
<span id="cb53-4"><a href="#cb53-4"></a>  <span class="kw">print</span>(<span class="kw">c</span>(i, tmp_levels))</span>
<span id="cb53-5"><a href="#cb53-5"></a>}</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a>ssu_ps_pime_<span class="dv">0</span>_clean_sub_Firmicutes_agg_order &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Bacillaceae&quot;</span> <span class="st">&quot;o_Bacillales&quot;</span>     <span class="st">&quot;Planococcaceae&quot;</span>   <span class="st">&quot;Paenibacillaceae&quot;</span>)</span>
<span id="cb54-2"><a href="#cb54-2"></a>ssu_ps_pime_<span class="dv">4</span>_clean_sub_Firmicutes_agg_order</span>
<span id="cb54-3"><a href="#cb54-3"></a>ssu_ps_pime_<span class="dv">8</span>_clean_sub_Firmicutes_agg_order</span>
<span id="cb54-4"><a href="#cb54-4"></a>ssu_ps_pime_clean_sub_Firmicutes_agg_order</span>
<span id="cb54-5"><a href="#cb54-5"></a>ssu_ps_work_<span class="dv">0</span>_clean_sub_Firmicutes_agg_order</span>
<span id="cb54-6"><a href="#cb54-6"></a>ssu_ps_work_<span class="dv">4</span>_clean_sub_Firmicutes_agg_order</span>
<span id="cb54-7"><a href="#cb54-7"></a>ssu_ps_work_<span class="dv">8</span>_clean_sub_Firmicutes_agg_order</span>
<span id="cb54-8"><a href="#cb54-8"></a>ssu_ps_work_clean_sub_Firmicutes_agg_order</span></code></pre></div>
</div>
<ol start="5" type="1">
<li>Plot the data for a single phyloseq object. Here you use an aggragated tax file.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a>ssu_ps_work_Alphaproteobacteria_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(ssu_ps_work_clean_sub_Alphaproteobacteria_agg_tax,</span>
<span id="cb55-2"><a href="#cb55-2"></a>                <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(TEMP),</span>
<span id="cb55-3"><a href="#cb55-3"></a>                    <span class="dt">y =</span> Abundance, <span class="dt">fill =</span> <span class="kw">get</span>(top_level))) <span class="op">+</span></span>
<span id="cb55-4"><a href="#cb55-4"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></span>
<span id="cb55-5"><a href="#cb55-5"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> ssu_colvec.tax) <span class="op">+</span></span>
<span id="cb55-6"><a href="#cb55-6"></a><span class="st">  </span><span class="co">#scale_x_discrete(&quot;Temperature&quot;, expand = waiver(), position = &quot;bottom&quot;, drop = FALSE) +</span></span>
<span id="cb55-7"><a href="#cb55-7"></a><span class="st">  </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></span>
<span id="cb55-8"><a href="#cb55-8"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> top_level)) <span class="op">+</span></span>
<span id="cb55-9"><a href="#cb55-9"></a><span class="st">  </span><span class="co">#guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +</span></span>
<span id="cb55-10"><a href="#cb55-10"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Relative Abundance (% total reads)&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Temperature&quot;</span>) <span class="op">+</span></span>
<span id="cb55-11"><a href="#cb55-11"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb55-12"><a href="#cb55-12"></a>        <span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb55-13"><a href="#cb55-13"></a>        <span class="dt">plot.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb55-14"><a href="#cb55-14"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb55-15"><a href="#cb55-15"></a>        <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb55-16"><a href="#cb55-16"></a>        <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>),</span>
<span id="cb55-17"><a href="#cb55-17"></a>        <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>))</span>
<span id="cb55-18"><a href="#cb55-18"></a>ssu_ps_work_Alphaproteobacteria_plot</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a>ssu_split_df &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ssu_ps_work&quot;</span>, <span class="st">&quot;ssu_ps_pime_0&quot;</span>, <span class="st">&quot;ssu_ps_pime_4&quot;</span>, <span class="st">&quot;ssu_ps_pime_8&quot;</span>)</span>
<span id="cb56-2"><a href="#cb56-2"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_split_df) {</span>
<span id="cb56-3"><a href="#cb56-3"></a>  tmp_level_get &lt;-<span class="st"> </span><span class="kw">get</span>(i)</span>
<span id="cb56-4"><a href="#cb56-4"></a>  tmp_level &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(tmp_level_get))</span>
<span id="cb56-5"><a href="#cb56-5"></a>  tmp_level &lt;-<span class="st"> </span>tmp_level[<span class="kw">order</span>(tmp_level<span class="op">$</span>TEMP), ]</span>
<span id="cb56-6"><a href="#cb56-6"></a>  tmp_level &lt;-<span class="st"> </span><span class="kw">as.vector</span>(tmp_level<span class="op">$</span>SamName)</span>
<span id="cb56-7"><a href="#cb56-7"></a></span>
<span id="cb56-8"><a href="#cb56-8"></a>  tmp_agg_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_clean_sub_&quot;</span>, subset_name, <span class="st">&quot;_agg_tax&quot;</span>))</span>
<span id="cb56-9"><a href="#cb56-9"></a>  tmp_get &lt;-<span class="st"> </span><span class="kw">get</span>(tmp_agg_name)</span>
<span id="cb56-10"><a href="#cb56-10"></a>  tmp_df &lt;-<span class="st"> </span>reshape<span class="op">::</span><span class="kw">melt</span>(tmp_get, <span class="dt">id.vars =</span> <span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;TEMP&quot;</span>, <span class="st">&quot;Abundance&quot;</span>, top_level))</span>
<span id="cb56-11"><a href="#cb56-11"></a>  tmp_plot_name &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(i, <span class="op">~</span><span class="kw">paste0</span>(., <span class="st">&quot;_&quot;</span>, subset_name, <span class="st">&quot;_melt_plot&quot;</span>))</span>
<span id="cb56-12"><a href="#cb56-12"></a></span>
<span id="cb56-13"><a href="#cb56-13"></a>  tmp_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tmp_df,</span>
<span id="cb56-14"><a href="#cb56-14"></a>                <span class="kw">aes</span>(<span class="dt">x =</span> Sample,</span>
<span id="cb56-15"><a href="#cb56-15"></a>                    <span class="dt">y =</span> Abundance, <span class="dt">fill =</span> <span class="kw">get</span>(top_level))) <span class="op">+</span></span>
<span id="cb56-16"><a href="#cb56-16"></a><span class="st">    </span><span class="kw">facet_grid</span>(. <span class="op">~</span>TEMP, <span class="dt">scale =</span> <span class="st">&quot;free_x&quot;</span>, <span class="dt">space =</span> <span class="st">&quot;free_x&quot;</span>)<span class="op">+</span></span>
<span id="cb56-17"><a href="#cb56-17"></a><span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>)  <span class="op">+</span></span>
<span id="cb56-18"><a href="#cb56-18"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> ssu_colvec.tax) <span class="op">+</span></span>
<span id="cb56-19"><a href="#cb56-19"></a><span class="st">    </span><span class="co">#scale_x_discrete(&quot;Treatment&quot;, expand = waiver(),</span></span>
<span id="cb56-20"><a href="#cb56-20"></a><span class="st">    </span><span class="co">#                 position = &quot;bottom&quot;, drop = FALSE, limits = tmp_level) +</span></span>
<span id="cb56-21"><a href="#cb56-21"></a><span class="st">    </span><span class="kw">theme_cowplot</span>() <span class="op">+</span></span>
<span id="cb56-22"><a href="#cb56-22"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">title =</span> top_level, <span class="dt">reverse =</span> <span class="ot">FALSE</span>,</span>
<span id="cb56-23"><a href="#cb56-23"></a>                               <span class="dt">keywidth =</span> <span class="fl">0.7</span>, <span class="dt">keyheight =</span> <span class="fl">0.7</span>)) <span class="op">+</span></span>
<span id="cb56-24"><a href="#cb56-24"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb56-25"><a href="#cb56-25"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb56-26"><a href="#cb56-26"></a>        <span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb56-27"><a href="#cb56-27"></a>        <span class="dt">plot.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="dt">colour =</span> <span class="ot">NA</span>),</span>
<span id="cb56-28"><a href="#cb56-28"></a>        <span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb56-29"><a href="#cb56-29"></a>        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>),</span>
<span id="cb56-30"><a href="#cb56-30"></a>        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>),</span>
<span id="cb56-31"><a href="#cb56-31"></a>        <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb56-32"><a href="#cb56-32"></a>        <span class="co">#legend.key.size = unit(1.5, &quot;cm&quot;),</span></span>
<span id="cb56-33"><a href="#cb56-33"></a>        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>),</span>
<span id="cb56-34"><a href="#cb56-34"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">6</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb56-35"><a href="#cb56-35"></a>        <span class="dt">strip.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">angle =</span> <span class="dv">0</span>),</span>
<span id="cb56-36"><a href="#cb56-36"></a>        <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="ot">NULL</span>)</span>
<span id="cb56-37"><a href="#cb56-37"></a>  <span class="kw">assign</span>(tmp_plot_name, tmp_plot)</span>
<span id="cb56-38"><a href="#cb56-38"></a>  <span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>(<span class="dt">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb56-39"><a href="#cb56-39"></a>}</span>
<span id="cb56-40"><a href="#cb56-40"></a><span class="kw">objects</span>(<span class="dt">pattern=</span><span class="st">&quot;_plot&quot;</span>)</span>
<span id="cb56-41"><a href="#cb56-41"></a>ssu_ps_work_Alphaproteobacteria_melt_plot</span>
<span id="cb56-42"><a href="#cb56-42"></a>ssu_ps_pime_<span class="dv">0</span>_Alphaproteobacteria_melt_plot</span>
<span id="cb56-43"><a href="#cb56-43"></a>ssu_ps_pime_<span class="dv">4</span>_Alphaproteobacteria_melt_plot</span>
<span id="cb56-44"><a href="#cb56-44"></a>ssu_ps_pime_<span class="dv">8</span>_Alphaproteobacteria_melt_plot</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1"></a>ssu_ps_work_</span>
<span id="cb57-2"><a href="#cb57-2"></a>carbom_chloro_trans &lt;-<span class="st"> </span>carbom_chloro <span class="op">%&gt;%</span></span>
<span id="cb57-3"><a href="#cb57-3"></a><span class="st">        </span><span class="kw">transform_sample_counts</span>(<span class="cf">function</span>(x) {x<span class="op">/</span><span class="kw">sum</span>(x)} ) <span class="op">%&gt;%</span></span>
<span id="cb57-4"><a href="#cb57-4"></a><span class="st">        </span><span class="kw">psmelt</span>()</span>
<span id="cb57-5"><a href="#cb57-5"></a>ssu_Fractions_tax<span class="op">$</span>Phylum &lt;-<span class="st"> </span>gdata<span class="op">::</span><span class="kw">reorder.factor</span>(ssu_Fractions_tax<span class="op">$</span>Phylum,</span>
<span id="cb57-6"><a href="#cb57-6"></a>                                                  <span class="dt">new.order =</span> ssu_tax_order)</span>
<span id="cb57-7"><a href="#cb57-7"></a>ssu_Fractions_tax &lt;-<span class="st"> </span>ssu_Fractions_tax <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(Phylum)</span>
<span id="cb57-8"><a href="#cb57-8"></a></span>
<span id="cb57-9"><a href="#cb57-9"></a></span>
<span id="cb57-10"><a href="#cb57-10"></a></span>
<span id="cb57-11"><a href="#cb57-11"></a><span class="kw">plot_bar</span>(carbom_chloro_trans, <span class="dt">x =</span> <span class="st">&quot;Family&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;Family&quot;</span>, <span class="dt">facet_grid =</span> DEPTH<span class="op">~</span>TEMP) <span class="op">+</span></span>
<span id="cb57-12"><a href="#cb57-12"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="kw">aes</span>(<span class="dt">color =</span> Family, <span class="dt">fill =</span> Family), <span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="st">&quot;stack&quot;</span>)</span></code></pre></div>
</div>
<p>tax_tab &lt;- as.data.frame(tax_table(ps_work)) tax_tab &lt;- tibble::rownames_to_column(tax_tab, asv_id) tax_tab$ASV_SEQ &lt;- NULL colnames(tax_tab)[colnames(tax_tab) == ASV_ID] &lt;- ASV write.table(tax_tab, tables/trepo-data-prep/ssu/tax_tab_mc_ssu18.txt, quote = FALSE, sep = ", row.names = FALSE)</p>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<h2 id="source-code" class="appendix">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/tropical-repo/web/blob/master/trepo-taxa.Rmd">clicking this link</a>.</p>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/tropical-repo/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/tropical-repo/web/">https://github.com/tropical-repo/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p> Copyright 2020 The TROPICAL-REPO Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
