<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>TropicalRepo: 4 Prevalence Interval for Microbiome Evaluation (PIME)</title>

  <meta property="description" itemprop="description" content="Reproducible workflow to filter ASVs &amp; OTUs by prevalence intervals. In this workflow, we first rarefy the data, then split by predictor variable, calculate the best prevalence, and estimate the error in prediction. We end by saving objects for downstream analysis."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>
  <link rel="icon" type="image/vnd.microsoft.icon" href="assets/favicon.ico"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="TropicalRepo: 4 Prevalence Interval for Microbiome Evaluation (PIME)"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Reproducible workflow to filter ASVs &amp; OTUs by prevalence intervals. In this workflow, we first rarefy the data, then split by predictor variable, calculate the best prevalence, and estimate the error in prediction. We end by saving objects for downstream analysis."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="TropicalRepo"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="TropicalRepo: 4 Prevalence Interval for Microbiome Evaluation (PIME)"/>
  <meta property="twitter:description" content="Reproducible workflow to filter ASVs &amp; OTUs by prevalence intervals. In this workflow, we first rarefy the data, then split by predictor variable, calculate the best prevalence, and estimate the error in prediction. We end by saving objects for downstream analysis."/>

  <!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["4 Prevalence Interval for Microbiome Evaluation (PIME)"]},{"type":"character","attributes":{},"value":["Reproducible workflow to filter ASVs & OTUs by prevalence intervals. In this workflow, we first rarefy the data, then split by predictor variable, calculate the best prevalence, and estimate the error in prediction. We end by saving objects for downstream analysis.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
  <script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.17/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
  <link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
  <script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
  <script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
  <link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
  <script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
  <link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
  <script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
  <link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Roboto:300|Nunito+Sans:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    margin-top: 1em;
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1.3em;
  	font-family: 'Lato';
    font-style: normal;
    margin: 0.5em 0 0.5em;
  }

  d-article li {
    margin: 0px 0px 4px;
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #323232;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #FC6464;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;

    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header  {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .headroom {
      z-index: 1000;
      position: fixed;
      top: 0;
      left: 0;
      right: 0;
  }
  .distill-site-nav {
      background-color: #00203F;
      font-size: 15px;
      font-weight: 300;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: hsla(206, 95%, 20%, 0.2);
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }


  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;
  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;
  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;
  }

  summary:focus {
    outline: none;
  }

  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }


  /********************************/
  /* dygraph*/
  /********************************/

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"4 Prevalence Interval for Microbiome Evaluation (PIME)","description":"Reproducible workflow to filter ASVs & OTUs by prevalence intervals. In this workflow, we first rarefy the data, then split by predictor variable, calculate the best prevalence, and estimate the error in prediction. We end by saving objects for downstream analysis.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a class="logo" href="index.html">
<img src="assets/icon.png" alt="Logo"/>
</a>
<a href="index.html" class="title">TropicalRepo</a>
<input id="distill-search" class="nav-search hidden" type="text" placeholder="Search..."/>
</div>
<div class="nav-right">
<a href="trepo-summary.html">Summary</a>
<a href="trepo-physical.html">Physical</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Macrofauna
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<hr/>
<a href="trepo-macrotaxa.html">4. Macrofauna Taxa</a>
<a href="trepo-temporal.html">3. Temporal Diversity</a>
<a href="trepo-macrobeta.html">2. Beta Diversity</a>
<a href="trepo-macroalpha.html">1. Alpha Diversity</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Microbial
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Diversity Workflows</span>
<hr/>
<a href="trepo-metadata.html">9. Metadata</a>
<a href="trepo-da.html">8. DA ASVs &amp; OTUs</a>
<a href="trepo-beta.html">7. Beta Diversity</a>
<a href="trepo-alpha.html">6. Alpha Diversity</a>
<a href="trepo-taxa.html">5. Taxonomic</a>
<hr/>
<span class="nav-dropdown-header">Processing Workflows</span>
<a href="trepo-pime.html">4. PIME</a>
<a href="trepo-otu.html">3. OTU Clustering</a>
<a href="trepo-data-prep.html">2. Data Set Prep</a>
<a href="trepo-dada2.html">1. DADA2</a>
<hr/>
</div>
</div>
<a href="trepo-data.html">Data</a>
<a href="https://github.com/tropical-repo/web/" aria-label="Link to source">
<i class="fab fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>4 Prevalence Interval for Microbiome Evaluation (PIME)</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Reproducible workflow to filter ASVs &amp; OTUs by prevalence intervals. In this workflow, we first rarefy the data, then split by predictor variable, calculate the best prevalence, and estimate the error in prediction. We end by saving objects for downstream analysis.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#summary-of-results">Summary of Results</a>
<ul>
<li><a href="#full-data-set-asv-vs.-otu">Full Data Set ASV vs. OTU</a></li>
<li><a href="#merged-data-set-asv-vs.-otu">Merged Data Set ASV vs. OTU</a></li>
</ul></li>
<li><a href="#prerequisites">Prerequisites</a></li>
<li><a href="#merged-samples-asv-data">Merged Samples (ASV Data)</a>
<ul>
<li><a href="#setup">Setup</a></li>
<li><a href="#split-by-predictor-variable">Split by Predictor Variable</a></li>
<li><a href="#calculate-prevalence-intervals">Calculate Prevalence Intervals</a></li>
<li><a href="#calculate-best-prevalence">Calculate Best Prevalence</a></li>
<li><a href="#best-prevalence-summary">Best Prevalence Summary</a></li>
<li><a href="#estimate-error-in-prediction">Estimate Error in Prediction</a></li>
<li><a href="#save-phyloseq-pime-objects">Save Phyloseq PIME objects</a></li>
<li><a href="#create-ampvis2-pime-object">Create Ampvis2 PIME Object</a></li>
<li><a href="#summary">Summary</a></li>
</ul></li>
<li><a href="#merged-samples-otu-data">Merged Samples (OTU Data)</a>
<ul>
<li><a href="#setup-1">Setup</a></li>
<li><a href="#split-by-predictor-variable-1">Split by Predictor Variable</a></li>
<li><a href="#calculate-prevalence-intervals-1">Calculate Prevalence Intervals</a></li>
<li><a href="#calculate-best-prevalence-1">Calculate Best Prevalence</a></li>
<li><a href="#best-prevalence-summary-1">Best Prevalence Summary</a></li>
<li><a href="#estimate-error-in-prediction-1">Estimate Error in Prediction</a></li>
<li><a href="#save-phyloseq-pime-objects-1">Save Phyloseq PIME objects</a></li>
<li><a href="#create-ampvis2-pime-object-1">Create Ampvis2 PIME Object</a></li>
<li><a href="#summary-1">Summary</a></li>
</ul></li>
<li><a href="#all-samples-asv-data">All Samples (ASV Data)</a>
<ul>
<li><a href="#setup-2">Setup</a></li>
<li><a href="#split-by-predictor-variable-2">Split by Predictor Variable</a></li>
<li><a href="#calculate-prevalence-intervals-2">Calculate Prevalence Intervals</a></li>
<li><a href="#calculate-best-prevalence-2">Calculate Best Prevalence</a></li>
<li><a href="#best-prevalence-summary-2">Best Prevalence Summary</a></li>
<li><a href="#estimate-error-in-prediction-2">Estimate Error in Prediction</a></li>
<li><a href="#save-phyloseq-pime-objects-2">Save Phyloseq PIME objects</a></li>
<li><a href="#create-ampvis2-pime-object-2">Create Ampvis2 PIME Object</a></li>
<li><a href="#summary-2">Summary</a></li>
</ul></li>
<li><a href="#all-samples-otu-data">All Samples (OTU Data)</a>
<ul>
<li><a href="#setup-3">Setup</a></li>
<li><a href="#split-by-predictor-variable-3">Split by Predictor Variable</a></li>
<li><a href="#calculate-prevalence-intervals-3">Calculate Prevalence Intervals</a></li>
<li><a href="#calculate-best-prevalence-3">Calculate Best Prevalence</a></li>
<li><a href="#best-prevalence-summary-3">Best Prevalence Summary</a></li>
<li><a href="#estimate-error-in-prediction-3">Estimate Error in Prediction</a></li>
<li><a href="#save-phyloseq-pime-objects-3">Save Phyloseq PIME objects</a></li>
<li><a href="#create-ampvis2-pime-object-3">Create Ampvis2 PIME Object</a></li>
<li><a href="#summary-3">Summary</a></li>
<li><a href="#source-code">Source Code</a></li>
</ul></li>
</ul>
</nav>
</div>
<blockquote>
<p>Hit the <em>Hide Code</em> button to hide the R code (visible by default).</p>
</blockquote>
<aside>
<div class="layout-chunk" data-layout="l-body">
<input onclick="codefolder(&#39;pre.sourceCode&#39;);" type="button" value="Hide Code" id="codefolder-button" style="position: absolute; right: 14%; z-index: 200"/>
<script>
  function codefolder(query) {

    var x = document.querySelectorAll(query);
    if (x.length === 0) return;

    function toggle_vis(o) {
      var d = o.style.display;
      o.style.display = (d === 'block' || d === '') ? 'none':'block';
    }

    for (i = 0; i < x.length; i++) {
      var y = x[i];
      toggle_vis(y);
    }

    var elem = document.getElementById("codefolder-button");
    if (elem.value === "Hide Code") elem.value = "Show Code";
    else elem.value = "Hide Code";
  }
</script>
</div>
</aside>
<h1 id="summary-of-results">Summary of Results</h1>
<p>In this workflow, we used PIME (Prevalence Interval for Microbiome Evaluation) to filter the FULL 16S rRNA (ASV &amp; OTU) and the MERGED by replicates (ASV &amp; OTU) data sets. The first step in the process is to rarefy the data and then proceed with the filtering. Here is a summary of the results. Tables show comparisons of ASV vs. OTU data sets.</p>
<h2 id="full-data-set-asv-vs.-otu">Full Data Set ASV vs. OTU</h2>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:right;">
no. samples
</th>
<th style="text-align:right;">
ASV reads
</th>
<th style="text-align:right;">
OTU reads
</th>
<th style="text-align:right;">
no. ASVs
</th>
<th style="text-align:right;">
no. OTUs
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FULL data set
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
6457054
</td>
<td style="text-align:right;">
6457054
</td>
<td style="text-align:right;">
47725
</td>
<td style="text-align:right;">
21906
</td>
</tr>
<tr>
<td style="text-align:left;">
Rarefied data
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
820496
</td>
<td style="text-align:right;">
820496
</td>
<td style="text-align:right;">
27818
</td>
<td style="text-align:right;">
13450
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME filtered data
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
711567
</td>
<td style="text-align:right;">
740985
</td>
<td style="text-align:right;">
7795
</td>
<td style="text-align:right;">
3371
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (ALMR)
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
179992
</td>
<td style="text-align:right;">
186549
</td>
<td style="text-align:right;">
3417
</td>
<td style="text-align:right;">
1719
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (CRIS)
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
176867
</td>
<td style="text-align:right;">
184235
</td>
<td style="text-align:right;">
3336
</td>
<td style="text-align:right;">
1713
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PAST)
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
175867
</td>
<td style="text-align:right;">
184770
</td>
<td style="text-align:right;">
3471
</td>
<td style="text-align:right;">
1771
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PUCL)
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
178841
</td>
<td style="text-align:right;">
185431
</td>
<td style="text-align:right;">
3859
</td>
<td style="text-align:right;">
1893
</td>
</tr>
</tbody>
</table>
</div>
<h2 id="merged-data-set-asv-vs.-otu">Merged Data Set ASV vs. OTU</h2>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:right;">
no. samples
</th>
<th style="text-align:right;">
ASV reads
</th>
<th style="text-align:right;">
OTU reads
</th>
<th style="text-align:right;">
no. ASVs
</th>
<th style="text-align:right;">
no. OTUs
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FULL data set
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
6457054
</td>
<td style="text-align:right;">
6457054
</td>
<td style="text-align:right;">
47725
</td>
<td style="text-align:right;">
21910
</td>
</tr>
<tr>
<td style="text-align:left;">
Rarefied data
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
1339728
</td>
<td style="text-align:right;">
1339728
</td>
<td style="text-align:right;">
32423
</td>
<td style="text-align:right;">
15619
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME filtered data
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
1236856
</td>
<td style="text-align:right;">
1269326
</td>
<td style="text-align:right;">
11066
</td>
<td style="text-align:right;">
5012
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (ALMR)
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
312348
</td>
<td style="text-align:right;">
319103
</td>
<td style="text-align:right;">
5222
</td>
<td style="text-align:right;">
2684
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (CRIS)
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
307830
</td>
<td style="text-align:right;">
316360
</td>
<td style="text-align:right;">
5136
</td>
<td style="text-align:right;">
2699
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PAST)
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
306844
</td>
<td style="text-align:right;">
315960
</td>
<td style="text-align:right;">
5345
</td>
<td style="text-align:right;">
2702
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PUCL)
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
309834
</td>
<td style="text-align:right;">
317903
</td>
<td style="text-align:right;">
5521
</td>
<td style="text-align:right;">
2820
</td>
</tr>
</tbody>
</table>
</div>
<h1 id="prerequisites">Prerequisites</h1>
<p>This workflow contains diversity assessments for the 16S rRNA data set. In order to run the workflow, you either need to first run the <a href="trepo-dada2.html">DADA2 Workflow</a>, the <a href="trepo-data-prep.html">Data Preparation workflow</a>, and the <a href="trepo-otu.html">OTU clustering workflow</a>. See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
<h1 id="merged-samples-asv-data">Merged Samples (ASV Data)</h1>
<p>We will use PIME (Prevalence Interval for Microbiome Evaluation) to create a filtered data set.</p>
<h2 id="setup">Setup</h2>
<p>First, choose a phyloseq object and a sample data frame</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds <span class="ot">&lt;-</span> ssu_ps_work_merge</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>ssu_which_pime <span class="ot">&lt;-</span> <span class="st">&quot;ssu_pime_ds&quot;</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">### USE THIS TO CREATE NEW SAMPLE VARIABLE FOR SPLITTING</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="do">##tmp_samp_data &lt;- sample_data(ssu_pime_ds) </span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_samp_data$YR_TREAT &lt;- paste(tmp_samp_data$YEAR, tmp_samp_data$TREAT, sep=&quot;_&quot;)</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_ps &lt;- merge_phyloseq(ssu_pime_ds, tmp_samp_data)   </span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_pime_ds &lt;- tmp_ps  </span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">#rm(list = ls(pattern = &quot;tmp_&quot;))</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 47725 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 47725 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>And then rarefy the data.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ssu_pime_sample_d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(<span class="fu">otu_table</span>(ssu_pime_ds)))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>ssu_pime_sample_d <span class="ot">&lt;-</span> ssu_pime_sample_d <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">total_reads =</span> <span class="dv">1</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds <span class="ot">&lt;-</span> <span class="fu">rarefy_even_depth</span>(ssu_pime_ds, </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sample.size =</span> <span class="fu">min</span>(ssu_pime_sample_d<span class="sc">$</span>total_reads), </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">trimOTUs =</span> <span class="cn">TRUE</span>, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">rngseed =</span> <span class="dv">119</span>)</span></code></pre></div>
</div>
<pre><code>`set.seed(119)` was used to initialize repeatable random subsampling.
Please record this for your records so others can reproduce.
Try `set.seed(119); .Random.seed` for the full vector

15302 OTUs were removed because they are no longer 
present in any sample after random subsampling
</code></pre>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 32423 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 32423 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>The first step in PIME is to define if the microbial community presents a high relative abundance of taxa with low prevalence, which is considered as noise in PIME analysis. This is calculated by random forests analysis and is the baseline noise detection.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>ssu_pime.oob.error <span class="ot">&lt;-</span> <span class="fu">pime.oob.error</span>(ssu_pime_ds, <span class="st">&quot;SITE&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] 0.01923077</code></pre>
</div>
<h2 id="split-by-predictor-variable">Split by Predictor Variable</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ssu_pime_ds))</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>ssu_per_variable_obj <span class="ot">&lt;-</span> <span class="fu">pime.split.by.variable</span>(ssu_pime_ds, <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>ssu_per_variable_obj</span></code></pre></div>
</div>
<details>
<summary>
Detailed results for splitting by sample variables
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$ALMR
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 13892 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 13892 taxa by 8 taxonomic ranks ]

$CRIS
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 15024 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 15024 taxa by 8 taxonomic ranks ]

$PAST
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 15585 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 15585 taxa by 8 taxonomic ranks ]

$PUCL
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 14893 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 14893 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</details>
<h2 id="calculate-prevalence-intervals">Calculate Prevalence Intervals</h2>
<p>Using the output of <code>pime.split.by.variable</code>, we calculate the prevalence intervals with the function <code>pime.prevalence</code>. This function estimates the highest prevalence possible (no empty ASV table), calculates prevalence for taxa, starting at 5 maximum prevalence possible (no empty ASV table or dropping samples). After prevalence calculation, each prevalence interval are merged.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>ssu_prevalences <span class="ot">&lt;-</span> <span class="fu">pime.prevalence</span>(ssu_per_variable_obj)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>ssu_prevalences</span></code></pre></div>
</div>
<details>
<summary>
Detailed results for all prevalences intervals
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$`5`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 16911 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 16911 taxa by 8 taxonomic ranks ]

$`10`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 15028 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 15028 taxa by 8 taxonomic ranks ]

$`15`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 13479 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 13479 taxa by 8 taxonomic ranks ]

$`20`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 11066 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 11066 taxa by 8 taxonomic ranks ]

$`25`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 10039 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 10039 taxa by 8 taxonomic ranks ]

$`30`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 9145 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 9145 taxa by 8 taxonomic ranks ]

$`35`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7668 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 7668 taxa by 8 taxonomic ranks ]

$`40`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7052 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 7052 taxa by 8 taxonomic ranks ]

$`45`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 6479 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 6479 taxa by 8 taxonomic ranks ]

$`50`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5411 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5411 taxa by 8 taxonomic ranks ]

$`55`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4934 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 4934 taxa by 8 taxonomic ranks ]

$`60`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4495 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 4495 taxa by 8 taxonomic ranks ]

$`65`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4089 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 4089 taxa by 8 taxonomic ranks ]

$`70`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3312 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3312 taxa by 8 taxonomic ranks ]

$`75`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2987 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2987 taxa by 8 taxonomic ranks ]

$`80`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2672 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2672 taxa by 8 taxonomic ranks ]

$`85`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2021 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2021 taxa by 8 taxonomic ranks ]

$`90`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1693 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1693 taxa by 8 taxonomic ranks ]

$`95`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1354 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1354 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</details>
<h2 id="calculate-best-prevalence">Calculate Best Prevalence</h2>
<p>Finally, we use the function <code>pime.best.prevalence</code> to calculate the best prevalence. The function uses randomForest to build random forests trees for samples classification and variable importance computation. It performs classifications for each prevalence interval returned by <code>pime.prevalence</code>. Variable importance is calculated, returning the Mean Decrease Accuracy (MDA), Mean Decrease Impurity (MDI), overall and by sample group, and taxonomy for each ASV. PIME keeps the top 30 variables with highest MDA each prevalence level.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1911</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>ssu_best.prev <span class="ot">&lt;-</span> <span class="fu">pime.best.prevalence</span>(ssu_prevalences, <span class="st">&quot;SITE&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>         Interval OOB error rate (%)  OTUs   Nseqs
1   Prevalence 5%               1.92 16911 1307494
2  Prevalence 10%               0.96 15028 1292012
3  Prevalence 15%               0.96 13479 1274752
4  Prevalence 20%                  0 11066 1236856
5  Prevalence 25%                  0 10039 1216233
6  Prevalence 30%                  0  9145 1195045
7  Prevalence 35%                  0  7668 1153904
8  Prevalence 40%                  0  7052 1131893
9  Prevalence 45%                  0  6479 1109363
10 Prevalence 50%                  0  5411 1061161
11 Prevalence 55%                  0  4934 1037216
12 Prevalence 60%                  0  4495 1012617
13 Prevalence 65%                  0  4089  987006
14 Prevalence 70%                  0  3312  929687
15 Prevalence 75%                  0  2987  898727
16 Prevalence 80%                  0  2672  867681
17 Prevalence 85%                  0  2021  787247
18 Prevalence 90%                  0  1693  738742
19 Prevalence 95%                  0  1354  669000</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best <span class="ot">&lt;-</span> ssu_best.prev<span class="sc">$</span><span class="st">`</span><span class="at">OOB error</span><span class="st">`</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>)] <span class="ot">&lt;-</span> <span class="fu">sapply</span>(ssu_what_is_best[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>)], as.numeric)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best <span class="ot">&lt;-</span> ssu_what_is_best <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;OOB_error_rate&quot;</span> <span class="ot">=</span> <span class="st">&quot;OOB error rate (%)&quot;</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best<span class="sc">$</span>Interval <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(ssu_what_is_best<span class="sc">$</span>Interval, <span class="st">&quot;%&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>ssu_best <span class="ot">&lt;-</span> <span class="fu">with</span>(ssu_what_is_best, Interval[<span class="fu">which.min</span>(OOB_error_rate)])</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>ssu_best <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;`&quot;</span>, ssu_best, <span class="st">&quot;`&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>ssu_prev_choice <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_best.prev$`Importance`$&quot;</span>, ssu_best, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>ssu_imp_best <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_prev_choice)))</span></code></pre></div>
</div>
<p>Looks like the lowest OOB error rate (%) that retains the most ASVs is 0% from <code>Prevalence 20</code>. We will use this interval.</p>
<h2 id="best-prevalence-summary">Best Prevalence Summary</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_imp_best[, 13:14] &lt;- list(NULL)</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>ssu_imp_best[,(<span class="fu">ncol</span>(ssu_imp_best)<span class="sc">-</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">ncol</span>(ssu_imp_best)] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="cn">NULL</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>ssu_imp_best <span class="ot">&lt;-</span> ssu_imp_best <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;ASV_ID&quot;</span> <span class="ot">=</span> <span class="st">&quot;SequenceID&quot;</span>)</span></code></pre></div>
</div>
<p></br></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="856uu6dn9soetf8m9lwa" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="856uu6dn9soetf8m9lwa">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000681818181818\" data-max=\"0.015661904761905\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000307692307692\" data-max=\"0.014079797979798\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000595238095238\" data-max=\"0.01281341991342\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000153846153846\" data-max=\"0.018136181139123\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.003764191871962\" data-max=\"0.006350691597212\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.20872645770905\" data-max=\"0.568330928313938\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Table of ASVs (max 30)\n                                     from the chosen prevalence\n                                     interval.<\/em>\n<\/caption>","data":[["ASV607","ASV1273","ASV2890","ASV244","ASV319","ASV616","ASV80","ASV204","ASV1840","ASV108","ASV78","ASV1053","ASV280","ASV481","ASV304","ASV712","ASV187","ASV1451","ASV17","ASV10","ASV477","ASV61","ASV246","ASV89","ASV213","ASV2207","ASV132","ASV737","ASV46","ASV437"],[0.00146212121212121,0.000681818181818182,0.00278205128205128,0.00728513708513709,0.00511428571428571,0.0056,0.00221349206349206,0.00494444444444444,0.0139787545787546,0.00222727272727273,0.0140725274725275,0.0156619047619048,0.00167777777777778,0.00367207792207792,0.00181587301587302,0.0053056277056277,0.00294949494949495,0.0028469696969697,0.0109888888888889,0.00341368631368631,0.003996336996337,0.00474350649350649,0.00395108225108225,0.00327272727272727,0.00385353535353535,0.00326767676767677,0.00411825396825397,0.00395757575757576,0.0040989898989899,0.00475],[0.00572857142857143,0.005895670995671,0.0129441558441558,0.00580606060606061,0.0058525641025641,0.00598888888888889,0.014079797979798,0.00473181818181818,0.000785714285714286,0.00392395382395382,0.00104848484848485,0.000727272727272727,0.011384188034188,0.00488034188034188,0.00236743256743257,0.00095,0.00445953654188948,0.00256557886557887,0.00663589743589744,0.00853614718614719,0.00371111111111111,0.00382222222222222,0.00383484848484848,0.004204329004329,0.00508311688311688,0.000307692307692308,0.00593668831168831,0.00392857142857143,0.00446825396825397,0.00354327894327894],[0.00106666666666667,0.000595238095238095,0.00295485625485625,0.0053979797979798,0.00687676767676768,0.0064547619047619,0.00226118326118326,0.00508888888888889,0.00443535353535354,0.00206926406926407,0.00416623376623377,0.00436026196026196,0.0015547619047619,0.00628571428571429,0.00218181818181818,0.0128134199134199,0.00395952380952381,0.0016049062049062,0.000715151515151515,0.00370079365079365,0.00423686868686869,0.0042,0.00491111111111111,0.0049984126984127,0.00407264957264957,0.0123358585858586,0.0029,0.00281367521367521,0.00355,0.00381515151515151],[0.0168823898323898,0.0181361811391223,0.00307272727272727,0.00472727272727273,0.00469090909090909,0.0051974025974026,0.00332374847374847,0.00616969696969697,0.00179365079365079,0.0112554112554113,0.000861538461538462,0.000342857142857143,0.00362424242424242,0.00423686868686869,0.0111893939393939,0.000598290598290598,0.00764069264069264,0.00987525252525252,0.00162171717171717,0.00186666666666667,0.0050954156954157,0.00443333333333333,0.00397727272727273,0.00411499611499612,0.00341414141414141,0.000153846153846154,0.00351746031746032,0.00453939393939394,0.00322727272727273,0.00325194805194805],[0.00635069159721112,0.00627714122124267,0.00572101034039416,0.0056875104427736,0.00556643167449619,0.00549851695734049,0.00514515133988818,0.00514034200296754,0.0049978924409769,0.004948554057909,0.00489446174991746,0.0048812474459892,0.00480785956176729,0.00473850940508042,0.00465510179999344,0.00462931352349808,0.00454867871811479,0.00444294811428958,0.00434699456030725,0.0043149702801316,0.00426734994103415,0.00425213279851631,0.00415459102301208,0.00403084634663582,0.00402653088225844,0.00397554677590451,0.00396363459787639,0.00380120086312037,0.00379946793483379,0.00376419187196272],[0.562220936948756,0.568330928313938,0.433750452709712,0.26411295322709,0.342977877027627,0.314072101723896,0.50636112162448,0.353329130340031,0.486169871410437,0.348230972311986,0.42343599162829,0.499346598730814,0.373148848759659,0.268855555464351,0.407906974956208,0.521096232403404,0.363264535118141,0.428848920792194,0.463402480336471,0.486682497262282,0.268934783364287,0.212533268613752,0.263102079073881,0.251118223261502,0.30420897300129,0.415202574729119,0.250095314993289,0.20872645770905,0.21066960267246,0.212505853133922],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Archaea","Bacteria","Bacteria","Bacteria","Bacteria"],["Planctomycetota","Nitrospirota","Acidobacteriota","Actinobacteriota","Proteobacteria","Nitrospirota","Proteobacteria","Acidobacteriota","Nanoarchaeota","Nitrospirota","Acidobacteriota","AncK6","Acidobacteriota","Acidobacteriota","Acidobacteriota","Acidobacteriota","Nitrospirota","Proteobacteria","Nitrospirota","Acidobacteriota","Acidobacteriota","Proteobacteria","Acidobacteriota","Proteobacteria","Asgardarchaeota","Thermoplasmatota","Acidobacteriota","Proteobacteria","Desulfobacterota","Desulfobacterota"],["vadinHA49","Thermodesulfovibrionia","Subgroup_22","Acidimicrobiia","Gammaproteobacteria","Thermodesulfovibrionia","Gammaproteobacteria","Thermoanaerobaculia","Nanoarchaeia","Thermodesulfovibrionia","Aminicenantia","p_AncK6","Thermoanaerobaculia","Aminicenantia","Aminicenantia","Aminicenantia","Thermodesulfovibrionia","Alphaproteobacteria","Nitrospiria","Thermoanaerobaculia","Thermoanaerobaculia","Alphaproteobacteria","Aminicenantia","Gammaproteobacteria","Odinarchaeia","Thermoplasmata","Aminicenantia","Gammaproteobacteria","Desulfobacteria","Desulfobacteria"],["c_vadinHA49","c_Thermodesulfovibrionia","c_Subgroup_22","Actinomarinales","Cellvibrionales","c_Thermodesulfovibrionia","AT-s2-59","Thermoanaerobaculales","Woesearchaeales","c_Thermodesulfovibrionia","Aminicenantales","p_AncK6","Thermoanaerobaculales","Aminicenantales","Aminicenantales","Aminicenantales","c_Thermodesulfovibrionia","Kiloniellales","Nitrospirales","Thermoanaerobaculales","Thermoanaerobaculales","Kiloniellales","Aminicenantales","Ectothiorhodospirales","c_Odinarchaeia","Marine_Benthic_Group_D_and_DHVEG-1","Aminicenantales","SS1-B-09-64","Desulfatiglandales","Desulfatiglandales"],["c_vadinHA49","c_Thermodesulfovibrionia","c_Subgroup_22","o_Actinomarinales","Spongiibacteraceae","c_Thermodesulfovibrionia","o_AT-s2-59","Thermoanaerobaculaceae","GW2011_GWC1_47_15","c_Thermodesulfovibrionia","o_Aminicenantales","p_AncK6","Thermoanaerobaculaceae","o_Aminicenantales","o_Aminicenantales","o_Aminicenantales","c_Thermodesulfovibrionia","Kiloniellaceae","Nitrospiraceae","Thermoanaerobaculaceae","Thermoanaerobaculaceae","Kiloniellaceae","o_Aminicenantales","Thioalkalispiraceae","c_Odinarchaeia","o_Marine_Benthic_Group_D_and_DHVEG-1","o_Aminicenantales","o_SS1-B-09-64","Desulfatiglandaceae","Desulfatiglandaceae"],["c_vadinHA49","c_Thermodesulfovibrionia","c_Subgroup_22","o_Actinomarinales","BD1-7_clade","c_Thermodesulfovibrionia","o_AT-s2-59","Subgroup_10","f_GW2011_GWC1_47_15","c_Thermodesulfovibrionia","o_Aminicenantales","p_AncK6","Subgroup_10","o_Aminicenantales","o_Aminicenantales","o_Aminicenantales","c_Thermodesulfovibrionia","f_Kiloniellaceae","Nitrospira","Subgroup_10","Subgroup_10","f_Kiloniellaceae","o_Aminicenantales","Thiohalophilus","c_Odinarchaeia","o_Marine_Benthic_Group_D_and_DHVEG-1","o_Aminicenantales","o_SS1-B-09-64","Desulfatiglans","Desulfatiglans"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ASV_ID<\/th>\n      <th>ALMR<\/th>\n      <th>CRIS<\/th>\n      <th>PAST<\/th>\n      <th>PUCL<\/th>\n      <th>MeanDecreaseAccuracy<\/th>\n      <th>MeanDecreaseGini<\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\nvar value=data[10]; $(this.api().cell(row, 10).node()).css({'font-size':'80%'});\nvar value=data[11]; $(this.api().cell(row, 11).node()).css({'font-size':'80%'});\nvar value=data[12]; $(this.api().cell(row, 12).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>Now we need to create a phyloseq object of ASVs at this cutoff (<code>Prevalence 20</code>).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>ssu_best_val <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(ssu_best, <span class="st">&quot;Prevalence &quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>ssu_best_val <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_prevalences$&quot;</span>, ssu_best_val, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>ssu_prevalence_best <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_best_val)))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_prevalence_best, <span class="st">&quot;rdata/trepo/ssu_merge_asv_prevalence_best.rds&quot;</span>)</span></code></pre></div>
</div>
<p>And look at a summary of the data.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 11066 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 11066 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>10654</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>12171</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>1236856</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>11893</td>
</tr>
<tr class="odd">
<td>Mean number of reads</td>
<td>11893</td>
</tr>
<tr class="even">
<td>Median number of reads</td>
<td>11922</td>
</tr>
<tr class="odd">
<td>Total ASVS</td>
<td>11066</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.742</td>
</tr>
</tbody>
</table>
<h2 id="estimate-error-in-prediction">Estimate Error in Prediction</h2>
<p>Using the function <code>pime.error.prediction</code> we can estimate the error in prediction. For each prevalence interval, this function randomizes the samples labels into arbitrary groupings using <code>n</code> random permutations, defined by the <code>bootstrap</code> value. For each, randomized and prevalence filtered, data set the OOB error rate is calculated to estimate whether the original differences in groups of samples occur by chance. Results are in a list containing a table and a box plot summarizing the results.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>ssu_randomized <span class="ot">&lt;-</span> <span class="fu">pime.error.prediction</span>(ssu_pime_ds, <span class="st">&quot;TEMP&quot;</span>,</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">bootstrap =</span> <span class="dv">100</span>, <span class="at">parallel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">max.prev =</span> <span class="dv">95</span>)</span></code></pre></div>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>It is also possible to estimate the variation of OOB error for each prevalence interval filtering. This is done by running the random forests classification for <code>n</code> times, determined by the <code>bootstrap</code> value. The function will return a box plot figure and a table for each classification error.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>ssu_replicated.oob.error <span class="ot">&lt;-</span> <span class="fu">pime.oob.replicate</span>(ssu_prevalences, <span class="st">&quot;SITE&quot;</span>, </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">bootstrap =</span> <span class="dv">100</span>, <span class="at">parallel =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="trepo-pime_files/figure-html5/unnamed-chunk-31-1.png" width="624" /></p>
</div>
<p>To obtain the confusion matrix from random forests classification use the following:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>ssu_prev_confuse <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_best.prev$`Confusion`$&quot;</span>, ssu_best, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_prev_confuse)))</span></code></pre></div>
</div>
<h2 id="save-phyloseq-pime-objects">Save Phyloseq PIME objects</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime <span class="ot">&lt;-</span> ssu_prevalence_best</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ssu_ps_pime), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ssu_ps_pime))</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ssu_ps_pime,</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>                               sample_data,</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>                               ssu_ps_pime_tree)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_asv_pime.rds&quot;</span>)</span></code></pre></div>
</div>
<h3 id="split-save-by-predictor-variable">Split &amp; save by predictor variable</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime_split <span class="ot">&lt;-</span> <span class="fu">pime.split.by.variable</span>(ssu_ps_pime, <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>ALMR, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_asv_pime_ALMR.rds&quot;</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>CRIS, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_asv_pime_CRIS.rds&quot;</span>)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>PAST, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_asv_pime_PAST.rds&quot;</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>PUCL, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_asv_pime_PUCL.rds&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$ALMR
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5222 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5222 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 5222 tips and 5221 internal nodes ]

$CRIS
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5136 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5136 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 5136 tips and 5135 internal nodes ]

$PAST
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5345 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5345 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 5345 tips and 5344 internal nodes ]

$PUCL
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5521 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5521 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 5521 tips and 5520 internal nodes ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="create-ampvis2-pime-object">Create Ampvis2 PIME Object</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>ssu_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ssu_ps_pime)))</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>ssu_otu[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(ssu_otu, as.numeric)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>ssu_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(ssu_otu)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>ssu_tax <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu_ps_pime)))</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>ssu_samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ssu_ps_pime))</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>ssu_ps_amp <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(ssu_otu, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>),</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">tax_table</span>(ssu_tax, ssu_tax),</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">sample_data</span>(ssu_samples))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>ssu_amp_asv_pime  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">otu_table</span>(ssu_ps_amp))</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>ssu_amp_asv_pime <span class="ot">&lt;-</span> ssu_amp_asv_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu_ps_amp))</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime <span class="ot">&lt;-</span> ssu_amp_tax_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ssu_amp_tax_pime)[<span class="fu">colnames</span>(ssu_amp_tax_pime) <span class="sc">==</span> <span class="st">&quot;ASV_ID&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Species&quot;</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>ssu_amp_asv_tax_pime <span class="ot">&lt;-</span> <span class="fu">left_join</span>(ssu_amp_asv_pime, ssu_amp_tax_pime, <span class="at">by =</span> <span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>ssu_samp_data_t <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu_merge_samp_data)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_samp_data_t[,7:9] &lt;- NULL</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>ssu_amp_data_pime <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(ssu_amp_asv_tax_pime, <span class="at">metadata =</span> ssu_samp_data_t)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">amp_alphadiv</span>(</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>    ssu_amp_data_pime,</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">measure =</span> <span class="st">&quot;observed&quot;</span>,</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">richness =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>  ssu_diversity_pime <span class="sc">%&gt;%</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;total_reads&quot;</span> <span class="ot">=</span> <span class="st">&quot;Reads&quot;</span>,</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;total_asvs&quot;</span> <span class="ot">=</span> <span class="st">&quot;ObservedOTUs&quot;</span>)</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime<span class="sc">$</span>coverage <span class="ot">=</span> <span class="fu">cut</span>(ssu_diversity_pime<span class="sc">$</span>total_reads,</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5000</span>, <span class="dv">25000</span>, <span class="dv">100000</span>, <span class="dv">200000</span>),</span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>                          <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;low (&lt; 5k)&quot;</span>, <span class="st">&quot;medium (5-25k)&quot;</span>,</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;high (25-100k)&quot;</span>, <span class="st">&quot;extra_high (&gt; 100k)&quot;</span>))</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span> ssu_diversity_pime[<span class="fu">order</span>(ssu_diversity_pime<span class="sc">$</span>SamName), ]</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>ssu_amp_pime <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(ssu_amp_asv_tax_pime,</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>                          <span class="at">metadata =</span> ssu_diversity_pime, <span class="at">tree =</span> ssu_ps_pime_tree)</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>ssu_samp_pime <span class="ot">&lt;-</span> ssu_diversity_pime</span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>ssu_samp_pime <span class="ot">&lt;-</span> ssu_samp_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>ampvis2 object with 4 elements. 
Summary of OTU table:
     Samples         OTUs  Total#Reads    Min#Reads    Max#Reads 
         104        11066      1236856        10654        12171 
Median#Reads    Avg#Reads 
       11922     11892.85 

Assigned taxonomy:
      Kingdom        Phylum         Class         Order        Family 
  11066(100%) 11043(99.79%) 11063(99.97%) 11057(99.92%) 11037(99.74%) 
        Genus       Species 
11065(99.99%)   11066(100%) 

Metadata variables: 9 
 SamName, SITE, WEEK, REGION, SEASON, REP, total_reads, total_asvs, coverage</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_amp_pime, <span class="st">&quot;files/trepo/pime/rdata/ssu_merge_asv_amp_pime.rds&quot;</span>)</span></code></pre></div>
</div>
<h2 id="summary">Summary</h2>
<p>Let’s take a look at a table of sample information. Any header with the <code>_p</code> suffix is the <em>PIME filtered</em> data.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="k9rn0xxltiqkg0bowglj" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="k9rn0xxltiqkg0bowglj">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"10654\" data-max=\"12171\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"2132\" data-max=\"3588\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Sample summary table.\n            Use the buttons to navigate through the table or\n            download a copy.<\/em>\n<\/caption>","data":[["ALMR_W01_IB_HS","ALMR_W03_IB_HS","ALMR_W05_IB_HS","ALMR_W07_IB_HS","ALMR_W09_IB_HS","ALMR_W11_IB_HS","ALMR_W13_IB_NS","ALMR_W15_IB_NS","ALMR_W17_IB_NS","ALMR_W19_IB_NS","ALMR_W21_IB_NS","ALMR_W23_IB_NS","ALMR_W25_IB_NS","ALMR_W27_IB_NS","ALMR_W29_IB_NS","ALMR_W31_IB_NS","ALMR_W33_IB_NS","ALMR_W35_IB_NS","ALMR_W37_IB_NS","ALMR_W39_IB_NS","ALMR_W41_IB_HS","ALMR_W43_IB_HS","ALMR_W45_IB_HS","ALMR_W47_IB_HS","ALMR_W49_IB_HS","ALMR_W51_IB_HS","CRIS_W01_OB_HS","CRIS_W03_OB_HS","CRIS_W05_OB_HS","CRIS_W07_OB_HS","CRIS_W09_OB_HS","CRIS_W11_OB_HS","CRIS_W13_OB_NS","CRIS_W15_OB_NS","CRIS_W17_OB_NS","CRIS_W19_OB_NS","CRIS_W21_OB_NS","CRIS_W23_OB_NS","CRIS_W25_OB_NS","CRIS_W27_OB_NS","CRIS_W29_OB_NS","CRIS_W31_OB_NS","CRIS_W33_OB_NS","CRIS_W35_OB_NS","CRIS_W37_OB_NS","CRIS_W39_OB_NS","CRIS_W41_OB_HS","CRIS_W43_OB_HS","CRIS_W45_OB_HS","CRIS_W47_OB_HS","CRIS_W49_OB_HS","CRIS_W51_OB_HS","PAST_W01_IB_HS","PAST_W03_IB_HS","PAST_W05_IB_HS","PAST_W07_IB_HS","PAST_W09_IB_HS","PAST_W11_IB_HS","PAST_W13_IB_NS","PAST_W15_IB_NS","PAST_W17_IB_NS","PAST_W19_IB_NS","PAST_W21_IB_NS","PAST_W23_IB_NS","PAST_W25_IB_NS","PAST_W27_IB_NS","PAST_W29_IB_NS","PAST_W31_IB_NS","PAST_W33_IB_NS","PAST_W35_IB_NS","PAST_W37_IB_NS","PAST_W39_IB_NS","PAST_W41_IB_HS","PAST_W43_IB_HS","PAST_W45_IB_HS","PAST_W47_IB_HS","PAST_W49_IB_HS","PAST_W51_IB_HS","PUCL_W01_OB_HS","PUCL_W03_OB_HS","PUCL_W05_OB_HS","PUCL_W07_OB_HS","PUCL_W09_OB_HS","PUCL_W11_OB_HS","PUCL_W13_OB_NS","PUCL_W15_OB_NS","PUCL_W17_OB_NS","PUCL_W19_OB_NS","PUCL_W21_OB_NS","PUCL_W23_OB_NS","PUCL_W25_OB_NS","PUCL_W27_OB_NS","PUCL_W29_OB_NS","PUCL_W31_OB_NS","PUCL_W33_OB_NS","PUCL_W35_OB_NS","PUCL_W37_OB_NS","PUCL_W39_OB_NS","PUCL_W41_OB_HS","PUCL_W43_OB_HS","PUCL_W45_OB_HS","PUCL_W47_OB_HS","PUCL_W49_OB_HS","PUCL_W51_OB_HS"],["ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL"],["W01","W03","W05","W07","W09","W11","W13","W15","W17","W19","W21","W23","W25","W27","W29","W31","W33","W35","W37","W39","W41","W43","W45","W47","W49","W51","W01","W03","W05","W07","W09","W11","W13","W15","W17","W19","W21","W23","W25","W27","W29","W31","W33","W35","W37","W39","W41","W43","W45","W47","W49","W51","W01","W03","W05","W07","W09","W11","W13","W15","W17","W19","W21","W23","W25","W27","W29","W31","W33","W35","W37","W39","W41","W43","W45","W47","W49","W51","W01","W03","W05","W07","W09","W11","W13","W15","W17","W19","W21","W23","W25","W27","W29","W31","W33","W35","W37","W39","W41","W43","W45","W47","W49","W51"],["IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB"],["HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS"],["R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R"],[12003,12137,12091,11923,11893,11865,12042,12082,11956,12155,11992,11922,12035,12006,12053,12171,11957,11968,12090,12125,12083,12167,11985,11829,11899,11919,11969,11690,12049,11871,11765,11972,11822,11595,11973,11826,11894,11877,11924,11894,11937,11759,11650,11990,11775,11898,11786,11856,11871,11713,11691,11783,11902,11628,11941,11791,11935,11763,12055,11246,12008,11999,11824,11976,11878,11997,11605,12007,10654,12098,11996,11923,11888,11897,11673,11455,11926,11779,11974,12078,11958,11920,12024,12066,11890,11901,11864,11868,11960,12020,11861,11810,11931,11922,11780,12026,12058,11865,11978,12028,11818,11671,11916,11647],[2879,2807,2894,3122,2891,2615,2831,2710,3031,2485,2545,3080,2601,3035,2464,2448,3233,3091,2233,3006,2833,2673,2921,3227,3137,3047,2886,3003,2700,2931,2882,2620,2872,3006,2656,2901,2587,2667,2482,2636,2476,2528,3137,2731,2634,2728,2132,2646,2650,2942,3051,2963,2975,2698,3027,3022,3119,2627,2842,2277,3020,2608,2916,2711,3075,2455,2574,2690,2649,2817,2598,2584,2757,2575,3393,3163,3008,2984,3085,3225,3193,2966,2937,2809,3146,3076,2890,2935,2829,2659,3088,2655,2789,3458,3578,3104,2891,2561,3063,2402,3566,3588,3403,3018]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>SamName<\/th>\n      <th>SITE<\/th>\n      <th>WEEK<\/th>\n      <th>REGION<\/th>\n      <th>SEASON<\/th>\n      <th>REP<\/th>\n      <th>total_reads_p<\/th>\n      <th>total_asvs_p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[5,15],"columnDefs":[{"className":"dt-right","targets":[6,7]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>And here is how the data sets changed through the PIME filtering process.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<table>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
object name
</th>
<th style="text-align:right;">
no. samples
</th>
<th style="text-align:right;">
total reads
</th>
<th style="text-align:right;">
total asvs
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FULL data set
</td>
<td style="text-align:left;">
ssu_ps_work_merge
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
6457054
</td>
<td style="text-align:right;">
47725
</td>
</tr>
<tr>
<td style="text-align:left;">
Rarefied data
</td>
<td style="text-align:left;">
ssu_pime_ds
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
1339728
</td>
<td style="text-align:right;">
32423
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME filtered data
</td>
<td style="text-align:left;">
ssu_ps_pime
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
1236856
</td>
<td style="text-align:right;">
11066
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (ALMR)
</td>
<td style="text-align:left;">
ssu_ps_pime_ALMR
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
312348
</td>
<td style="text-align:right;">
5222
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (CRIS)
</td>
<td style="text-align:left;">
ssu_ps_pime_CRIS
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
307830
</td>
<td style="text-align:right;">
5136
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PAST)
</td>
<td style="text-align:left;">
ssu_ps_pime_PAST
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
306844
</td>
<td style="text-align:right;">
5345
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PUCL)
</td>
<td style="text-align:left;">
ssu_ps_pime_PUCL
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
309834
</td>
<td style="text-align:right;">
5521
</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="merged-samples-otu-data">Merged Samples (OTU Data)</h1>
<h2 id="setup-1">Setup</h2>
<p>First, choose a phyloseq object and a sample data frame</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds <span class="ot">&lt;-</span> ssu_ps_work_merge_otu</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>ssu_which_pime <span class="ot">&lt;-</span> <span class="st">&quot;ssu_pime_ds&quot;</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="do">### USE THIS TO CREATE NEW SAMPLE VARIABLE FOR SPLITTING</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_samp_data &lt;- sample_data(ssu_pime_ds)  </span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_samp_data$YR_TREAT &lt;- paste(tmp_samp_data$YEAR, tmp_samp_data$TREAT, sep=&quot;_&quot;)  </span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_ps &lt;- merge_phyloseq(ssu_pime_ds, tmp_samp_data)   </span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_pime_ds &lt;- tmp_ps  </span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="co">#rm(list = ls(pattern = &quot;tmp_&quot;))</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 21910 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 21910 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<p>Then rarefy the data.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>ssu_pime_sample_d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(<span class="fu">otu_table</span>(ssu_pime_ds)))</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>ssu_pime_sample_d <span class="ot">&lt;-</span> ssu_pime_sample_d <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">total_reads =</span> <span class="dv">1</span>)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds <span class="ot">&lt;-</span> <span class="fu">rarefy_even_depth</span>(ssu_pime_ds, </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">sample.size =</span> <span class="fu">min</span>(ssu_pime_sample_d<span class="sc">$</span>total_reads), </span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">trimOTUs =</span> <span class="cn">TRUE</span>, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">rngseed =</span> <span class="dv">119</span>)</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds</span></code></pre></div>
</div>
<pre><code>`set.seed(119)` was used to initialize repeatable random subsampling.
Please record this for your records so others can reproduce.
Try `set.seed(119); .Random.seed` for the full vector

6291 OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 15619 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 15619 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<p>The first step in PIME is to define if the microbial community presents a high relative abundance of taxa with low prevalence, which is considered as noise in PIME analysis. This is calculated by random forests analysis and is the baseline noise detection.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>ssu_pime.oob.error <span class="ot">&lt;-</span> <span class="fu">pime.oob.error</span>(ssu_pime_ds, <span class="st">&quot;SITE&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] 0.01923077</code></pre>
</div>
<h2 id="split-by-predictor-variable-1">Split by Predictor Variable</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ssu_pime_ds))</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>ssu_per_variable_obj <span class="ot">&lt;-</span> <span class="fu">pime.split.by.variable</span>(ssu_pime_ds, <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>ssu_per_variable_obj</span></code></pre></div>
</div>
<details>
<summary>
Detailed results for splitting by sample variables
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$ALMR
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7622 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 7622 taxa by 7 taxonomic ranks ]

$CRIS
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 8030 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 8030 taxa by 7 taxonomic ranks ]

$PAST
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 8314 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 8314 taxa by 7 taxonomic ranks ]

$PUCL
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7664 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 7664 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</details>
<h2 id="calculate-prevalence-intervals-1">Calculate Prevalence Intervals</h2>
<p>Using the output of <code>pime.split.by.variable</code>, we calculate the prevalence intervals with the function <code>pime.prevalence</code>. This function estimates the highest prevalence possible (no empty ASV table), calculates prevalence for taxa, starting at 5 maximum prevalence possible (no empty ASV table or dropping samples). After prevalence calculation, each prevalence interval are merged.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>ssu_prevalences <span class="ot">&lt;-</span> <span class="fu">pime.prevalence</span>(ssu_per_variable_obj)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>ssu_prevalences</span></code></pre></div>
</div>
<details>
<summary>
Detailed results for all prevalences intervals
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$`5`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 8646 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 8646 taxa by 7 taxonomic ranks ]

$`10`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7612 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 7612 taxa by 7 taxonomic ranks ]

$`15`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 6907 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 6907 taxa by 7 taxonomic ranks ]

$`20`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5843 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5843 taxa by 7 taxonomic ranks ]

$`25`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5404 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5404 taxa by 7 taxonomic ranks ]

$`30`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5012 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5012 taxa by 7 taxonomic ranks ]

$`35`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4329 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 4329 taxa by 7 taxonomic ranks ]

$`40`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4019 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 4019 taxa by 7 taxonomic ranks ]

$`45`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3750 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3750 taxa by 7 taxonomic ranks ]

$`50`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3266 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3266 taxa by 7 taxonomic ranks ]

$`55`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3023 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3023 taxa by 7 taxonomic ranks ]

$`60`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2819 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2819 taxa by 7 taxonomic ranks ]

$`65`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2614 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2614 taxa by 7 taxonomic ranks ]

$`70`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2254 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2254 taxa by 7 taxonomic ranks ]

$`75`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2078 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2078 taxa by 7 taxonomic ranks ]

$`80`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1901 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1901 taxa by 7 taxonomic ranks ]

$`85`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1577 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1577 taxa by 7 taxonomic ranks ]

$`90`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1390 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1390 taxa by 7 taxonomic ranks ]

$`95`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1181 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1181 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</details>
<h2 id="calculate-best-prevalence-1">Calculate Best Prevalence</h2>
<p>Finally, we use the function <code>pime.best.prevalence</code> to calculate the best prevalence. The function uses randomForest to build random forests trees for samples classification and variable importance computation. It performs classifications for each prevalence interval returned by <code>pime.prevalence</code>. Variable importance is calculated, returning the Mean Decrease Accuracy (MDA), Mean Decrease Impurity (MDI), overall and by sample group, and taxonomy for each ASV. PIME keeps the top 30 variables with highest MDA each prevalence level.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1911</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>ssu_best.prev <span class="ot">&lt;-</span> <span class="fu">pime.best.prevalence</span>(ssu_prevalences, <span class="st">&quot;SITE&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>         Interval OOB error rate (%) OTUs   Nseqs
1   Prevalence 5%               1.92 8646 1324657
2  Prevalence 10%               1.92 7612 1316668
3  Prevalence 15%               1.92 6907 1308415
4  Prevalence 20%               0.96 5843 1289737
5  Prevalence 25%               0.96 5404 1279705
6  Prevalence 30%                  0 5012 1269326
7  Prevalence 35%                  0 4329 1248043
8  Prevalence 40%                  0 4019 1236511
9  Prevalence 45%                  0 3750 1224041
10 Prevalence 50%                  0 3266 1198754
11 Prevalence 55%                  0 3023 1184862
12 Prevalence 60%                  0 2819 1171279
13 Prevalence 65%                  0 2614 1157015
14 Prevalence 70%                  0 2254 1124049
15 Prevalence 75%                  0 2078 1105437
16 Prevalence 80%                  0 1901 1085627
17 Prevalence 85%                  0 1577 1033225
18 Prevalence 90%                  0 1390 1000019
19 Prevalence 95%                  0 1181  953215</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best <span class="ot">&lt;-</span> ssu_best.prev<span class="sc">$</span><span class="st">`</span><span class="at">OOB error</span><span class="st">`</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>)] <span class="ot">&lt;-</span> <span class="fu">sapply</span>(ssu_what_is_best[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>)], as.numeric)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best <span class="ot">&lt;-</span> ssu_what_is_best <span class="sc">%&gt;%</span> </span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;OOB_error_rate&quot;</span> <span class="ot">=</span> <span class="st">&quot;OOB error rate (%)&quot;</span>)</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best<span class="sc">$</span>Interval <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(ssu_what_is_best<span class="sc">$</span>Interval, <span class="st">&quot;%&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>ssu_best <span class="ot">&lt;-</span> <span class="fu">with</span>(ssu_what_is_best, Interval[<span class="fu">which.min</span>(OOB_error_rate)])</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>ssu_best <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;`&quot;</span>, ssu_best, <span class="st">&quot;`&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>ssu_prev_choice <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_best.prev$`Importance`$&quot;</span>, ssu_best, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>ssu_imp_best <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_prev_choice)))</span></code></pre></div>
</div>
<p>Looks like the lowest OOB error rate (%) that retains the most ASVs is 0% from <code>Prevalence 30</code>. We will use this interval.</p>
<h2 id="best-prevalence-summary-1">Best Prevalence Summary</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_imp_best[, 13:14] &lt;- list(NULL)</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>ssu_imp_best[,(<span class="fu">ncol</span>(ssu_imp_best)<span class="sc">-</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">ncol</span>(ssu_imp_best)] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="cn">NULL</span>)</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>ssu_imp_best <span class="ot">&lt;-</span> ssu_imp_best <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;ASV_ID&quot;</span> <span class="ot">=</span> <span class="st">&quot;SequenceID&quot;</span>)</span></code></pre></div>
</div>
<p></br></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="pf0wohnh74w860fku3z5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="pf0wohnh74w860fku3z5">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000363636363636\" data-max=\"0.028539704739705\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000535714285714\" data-max=\"0.015901298701299\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000353846153846\" data-max=\"0.009451315351316\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.0005\" data-max=\"0.018936507936508\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.004180458089332\" data-max=\"0.009747423041805\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.272072220940584\" data-max=\"0.823860868640045\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Table of ASVs (max 30)\n                                     from the chosen prevalence\n                                     interval.<\/em>\n<\/caption>","data":[["OTU1348","OTU185","OTU1318","OTU85","OTU137","OTU33","OTU78","OTU583","OTU175","OTU50","OTU2539","OTU53","OTU625","OTU144","OTU11","OTU42","OTU463","OTU407","OTU149","OTU589","OTU565","OTU178","OTU29","OTU1319","OTU6","OTU302","OTU1072","OTU356","OTU2012","OTU620"],[0.0285397047397047,0.0032,0.0188895604395604,0.00233333333333333,0.00391313131313131,0.00161111111111111,0.0165586691086691,0.0125883116883117,0.00794003219003219,0.0081765873015873,0.00124603174603175,0.00980964035964036,0.000585858585858586,0.0115984126984127,0.00308658008658009,0.00867756132756133,0.0052525641025641,0.00376666666666667,0.00558412698412698,0.00491507936507937,0.00249393939393939,0.0141477633477633,0.00465952380952381,0.0105840270840271,0.000363636363636364,0.00547350427350427,0.0112068154068154,0.00160927960927961,0.00100793650793651,0.00138585858585859],[0.000966666666666667,0.00652828282828283,0.00135677655677656,0.00686031746031746,0.00421507381507381,0.0047510101010101,0.000535714285714286,0.00881802641802642,0.00459545454545454,0.00797319347319347,0.0159012987012987,0.00391111111111111,0.00488011988011988,0.00433113553113553,0.00433888888888889,0.00998001443001443,0.00598571428571429,0.00282004662004662,0.00588939393939394,0.00485714285714286,0.014170202020202,0.00225,0.00573923853923854,0.00118253968253968,0.00507380952380952,0.00459393939393939,0.000926739926739927,0.00406418026418026,0.00413085248085248,0.00312121212121212],[0.00945131535131535,0.00378066378066378,0.00701767676767677,0.00337619047619048,0.0018031746031746,0.000353846153846154,0.00377289377289377,0.000734798534798535,0.00564529914529915,0.00347698412698413,0.00213286713286713,0.00691991341991342,0.000967676767676768,0.0042478354978355,0.0025982683982684,0.00239848484848485,0.00437387057387057,0.00260839160839161,0.00435721500721501,0.00525299145299145,0.00266495726495726,0.00212878787878788,0.00520634920634921,0.0058955377955378,0.000931818181818182,0.00325714285714286,0.00516688311688312,0.00679529914529914,0.00131515151515152,0.00139166666666667],[0.000963636363636364,0.0171136363636364,0.00264568764568765,0.0152730158730159,0.0181865079365079,0.0189365079365079,0.0041,0.00308766233766234,0.00672429792429792,0.00407820512820513,0.0053445831945832,0.00323426573426573,0.0162740370740371,0.0020979020979021,0.0114873737373737,0.00229032634032634,0.00625,0.0117767676767677,0.00332121212121212,0.00488528138528139,0.0018525974025974,0.00175757575757576,0.00247619047619048,0.0012,0.0104883116883117,0.00464285714285714,0.0005,0.0041038961038961,0.0110285547785548,0.012147619047619],[0.00974742304180489,0.00737268855258721,0.006848048297599,0.00665019689903402,0.00655082433673765,0.00653009659737164,0.00640507496562199,0.00632855742405026,0.00604319923793608,0.00592768677846638,0.00579393728998992,0.00572018249544939,0.0053474936322694,0.00533742934429712,0.00531690063952286,0.00530878219414805,0.00527854304324893,0.00526221852224646,0.00493310928683201,0.00482141111614796,0.00472552982035447,0.00471817804930204,0.00463180683125964,0.00461193077822328,0.00452889995258416,0.00448510406721941,0.00433302688797667,0.00423482658482659,0.00421434287596411,0.00418045808933202],[0.823860868640045,0.60008607509562,0.75958582413688,0.548784124863157,0.570588609716861,0.668533893868542,0.668207280443019,0.580540951297943,0.390041806960582,0.625436397151645,0.52860447373036,0.543707180462436,0.480957349079564,0.348736298443531,0.470046129424268,0.607891749736829,0.311123556627624,0.430725933383862,0.307445515500523,0.322519902749965,0.516557526756948,0.532693673789023,0.272072220940585,0.405856654522712,0.357751935666672,0.303929217179622,0.424514706981583,0.291108340690182,0.42122369740837,0.409001514920423],["Archaea","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Bacteria"],["Nanoarchaeota","Acidobacteriota","Chloroflexi","Nitrospirota","Acidobacteriota","Chloroflexi","Crenarchaeota","Proteobacteria","Acidobacteriota","Chloroflexi","Planctomycetota","Chloroflexi","Acidobacteriota","Proteobacteria","Crenarchaeota","Nitrospirota","Acidobacteriota","Nitrospirota","Acidobacteriota","Nitrospirota","Nitrospinota","Calditrichota","Asgardarchaeota","Calditrichota","Acidobacteriota","Proteobacteria","Chloroflexi","Thermoplasmatota","Schekmanbacteria","Nitrospirota"],["Nanoarchaeia","Thermoanaerobaculia","Anaerolineae","Thermodesulfovibrionia","Aminicenantia","Anaerolineae","Bathyarchaeia","Gammaproteobacteria","Thermoanaerobaculia","Anaerolineae","Phycisphaerae","Anaerolineae","Thermoanaerobaculia","Gammaproteobacteria","Bathyarchaeia","Nitrospiria","Thermoanaerobaculia","Thermodesulfovibrionia","Aminicenantia","Nitrospiria","Nitrospinia","Calditrichia","Lokiarchaeia","Calditrichia","Aminicenantia","Alphaproteobacteria","Dehalococcoidia","Thermoplasmata","p_Schekmanbacteria","Thermodesulfovibrionia"],["Woesearchaeales","Thermoanaerobaculales","SBR1031","c_Thermodesulfovibrionia","Aminicenantales","Anaerolineales","c_Bathyarchaeia","c_Gammaproteobacteria","Thermoanaerobaculales","Anaerolineales","Phycisphaerales","Anaerolineales","Thermoanaerobaculales","Nitrosococcales","c_Bathyarchaeia","Nitrospirales","Thermoanaerobaculales","c_Thermodesulfovibrionia","Aminicenantales","Nitrospirales","Nitrospinales","Calditrichales","c_Lokiarchaeia","Calditrichales","Aminicenantales","Rhodobacterales","Sh765B-AG-111","Marine_Benthic_Group_D_and_DHVEG-1","p_Schekmanbacteria","c_Thermodesulfovibrionia"],["GW2011_GWC1_47_15","Thermoanaerobaculaceae","o_SBR1031","c_Thermodesulfovibrionia","o_Aminicenantales","Anaerolineaceae","c_Bathyarchaeia","c_Gammaproteobacteria","Thermoanaerobaculaceae","Anaerolineaceae","Phycisphaeraceae","Anaerolineaceae","Thermoanaerobaculaceae","Nitrosococcaceae","c_Bathyarchaeia","Nitrospiraceae","Thermoanaerobaculaceae","c_Thermodesulfovibrionia","o_Aminicenantales","Nitrospiraceae","Nitrospinaceae","Calditrichaceae","c_Lokiarchaeia","Calditrichaceae","o_Aminicenantales","Rhodobacteraceae","o_Sh765B-AG-111","o_Marine_Benthic_Group_D_and_DHVEG-1","p_Schekmanbacteria","c_Thermodesulfovibrionia"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ASV_ID<\/th>\n      <th>ALMR<\/th>\n      <th>CRIS<\/th>\n      <th>PAST<\/th>\n      <th>PUCL<\/th>\n      <th>MeanDecreaseAccuracy<\/th>\n      <th>MeanDecreaseGini<\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\nvar value=data[10]; $(this.api().cell(row, 10).node()).css({'font-size':'80%'});\nvar value=data[11]; $(this.api().cell(row, 11).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>Now we need to create a phyloseq object of ASVs at this cutoff (<code>Prevalence 30</code>).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>ssu_best_val <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(ssu_best, <span class="st">&quot;Prevalence &quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>ssu_best_val <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_prevalences$&quot;</span>, ssu_best_val, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>ssu_prevalence_best <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_best_val)))</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_prevalence_best, <span class="st">&quot;rdata/trepo/ssu_merge_otu_prevalence_best.rds&quot;</span>)</span></code></pre></div>
</div>
<p>And look at a summary of the data.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5012 taxa and 104 samples ]
sample_data() Sample Data:       [ 104 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5012 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>11568</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>12393</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>1269326</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>12205</td>
</tr>
<tr class="odd">
<td>Mean number of reads</td>
<td>12205</td>
</tr>
<tr class="even">
<td>Median number of reads</td>
<td>12224</td>
</tr>
<tr class="odd">
<td>Total ASVS</td>
<td>5012</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.642</td>
</tr>
</tbody>
</table>
<h2 id="estimate-error-in-prediction-1">Estimate Error in Prediction</h2>
<p>Using the function <code>pime.error.prediction</code> we can estimate the error in prediction. For each prevalence interval, this function randomizes the samples labels into arbitrary groupings using <code>n</code> random permutations, defined by the <code>bootstrap</code> value. For each, randomized and prevalence filtered, data set the OOB error rate is calculated to estimate whether the original differences in groups of samples occur by chance. Results are in a list containing a table and a box plot summarizing the results.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>ssu_randomized <span class="ot">&lt;-</span> <span class="fu">pime.error.prediction</span>(ssu_pime_ds, <span class="st">&quot;TEMP&quot;</span>,</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">bootstrap =</span> <span class="dv">100</span>, <span class="at">parallel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">max.prev =</span> <span class="dv">95</span>)</span></code></pre></div>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>It is also possible to estimate the variation of OOB error for each prevalence interval filtering. This is done by running the random forests classification for <code>n</code> times, determined by the <code>bootstrap</code> value. The function will return a box plot figure and a table for each classification error.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>ssu_replicated.oob.error <span class="ot">&lt;-</span> <span class="fu">pime.oob.replicate</span>(ssu_prevalences, <span class="st">&quot;SITE&quot;</span>, </span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">bootstrap =</span> <span class="dv">100</span>, <span class="at">parallel =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="trepo-pime_files/figure-html5/unnamed-chunk-73-1.png" width="624" /></p>
</div>
<p>To obtain the confusion matrix from random forests classification use the following:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>ssu_prev_confuse <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_best.prev$`Confusion`$&quot;</span>, ssu_best, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_prev_confuse)))</span></code></pre></div>
</div>
<h2 id="save-phyloseq-pime-objects-1">Save Phyloseq PIME objects</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime <span class="ot">&lt;-</span> ssu_prevalence_best</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ssu_ps_pime), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ssu_ps_pime))</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ssu_ps_pime,</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>                               sample_data,</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>                               ssu_ps_pime_tree)</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_otu_pime.rds&quot;</span>)</span></code></pre></div>
</div>
<h3 id="split-save-by-predictor-variable-1">Split &amp; save by predictor variable</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime_split <span class="ot">&lt;-</span> <span class="fu">pime.split.by.variable</span>(ssu_ps_pime, <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>ALMR, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_otu_pime_ALMR.rds&quot;</span>)</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>CRIS, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_otu_pime_CRIS.rds&quot;</span>)</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>PAST, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_otu_pime_PAST.rds&quot;</span>)</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>PUCL, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_merge_otu_pime_PUCL.rds&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$ALMR
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2684 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2684 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 2684 tips and 2683 internal nodes ]

$CRIS
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2699 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2699 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 2699 tips and 2698 internal nodes ]

$PAST
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2702 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2702 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 2702 tips and 2701 internal nodes ]

$PUCL
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2820 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2820 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 2820 tips and 2819 internal nodes ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="create-ampvis2-pime-object-1">Create Ampvis2 PIME Object</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>ssu_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ssu_ps_pime)))</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>ssu_otu[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(ssu_otu, as.numeric)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>ssu_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(ssu_otu)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>ssu_tax <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu_ps_pime)))</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>ssu_samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ssu_ps_pime))</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>ssu_ps_amp <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(ssu_otu, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>),</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">tax_table</span>(ssu_tax, ssu_tax),</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">sample_data</span>(ssu_samples))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>ssu_amp_otu_pime  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">otu_table</span>(ssu_ps_amp))</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>ssu_amp_otu_pime <span class="ot">&lt;-</span> ssu_amp_otu_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu_ps_amp))</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime <span class="ot">&lt;-</span> ssu_amp_tax_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="do">## THE NEXT  COMMAND SHOULD BE REMOVED, THIS IS A TEMP FIX</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a><span class="do">## BECAUSE OTU_ID&#39;S not INCLUDED FROM OTU WF</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime<span class="sc">$</span>ASV_ID <span class="ot">&lt;-</span> ssu_amp_tax_pime<span class="sc">$</span>OTU</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a><span class="do">#########</span></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ssu_amp_tax_pime)[<span class="fu">colnames</span>(ssu_amp_tax_pime) <span class="sc">==</span> <span class="st">&quot;ASV_ID&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Species&quot;</span></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>ssu_amp_otu_tax_pime <span class="ot">&lt;-</span> <span class="fu">left_join</span>(ssu_amp_otu_pime, ssu_amp_tax_pime, <span class="at">by =</span> <span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>ssu_samp_data_t <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu_samp_data)</span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_samp_data_t[,7:9] &lt;- NULL</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>ssu_amp_data_pime <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(ssu_amp_otu_tax_pime, <span class="at">metadata =</span> ssu_samp_data_t)</span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span></span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">amp_alphadiv</span>(</span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>    ssu_amp_data_pime,</span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">measure =</span> <span class="st">&quot;observed&quot;</span>,</span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">richness =</span> <span class="cn">FALSE</span>)</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span></span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a>  ssu_diversity_pime <span class="sc">%&gt;%</span></span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;total_reads&quot;</span> <span class="ot">=</span> <span class="st">&quot;Reads&quot;</span>,</span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;total_otus&quot;</span> <span class="ot">=</span> <span class="st">&quot;ObservedOTUs&quot;</span>)</span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime<span class="sc">$</span>coverage <span class="ot">=</span> <span class="fu">cut</span>(ssu_diversity_pime<span class="sc">$</span>total_reads,</span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5000</span>, <span class="dv">25000</span>, <span class="dv">100000</span>, <span class="dv">200000</span>),</span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a>                          <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;low (&lt; 5k)&quot;</span>, <span class="st">&quot;medium (5-25k)&quot;</span>,</span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;high (25-100k)&quot;</span>, <span class="st">&quot;extra_high (&gt; 100k)&quot;</span>))</span>
<span id="cb52-34"><a href="#cb52-34" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span> ssu_diversity_pime[<span class="fu">order</span>(ssu_diversity_pime<span class="sc">$</span>SamName), ]</span>
<span id="cb52-35"><a href="#cb52-35" aria-hidden="true" tabindex="-1"></a>ssu_amp_pime <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(ssu_amp_otu_tax_pime,</span>
<span id="cb52-36"><a href="#cb52-36" aria-hidden="true" tabindex="-1"></a>                          <span class="at">metadata =</span> ssu_diversity_pime, <span class="at">tree =</span> ssu_ps_pime_tree)</span>
<span id="cb52-37"><a href="#cb52-37" aria-hidden="true" tabindex="-1"></a>ssu_samp_pime <span class="ot">&lt;-</span> ssu_diversity_pime</span>
<span id="cb52-38"><a href="#cb52-38" aria-hidden="true" tabindex="-1"></a>ssu_samp_pime <span class="ot">&lt;-</span> ssu_samp_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>ampvis2 object with 4 elements. 
Summary of OTU table:
     Samples         OTUs  Total#Reads    Min#Reads    Max#Reads 
         104         5012      1269326        11568        12393 
Median#Reads    Avg#Reads 
       12224     12205.06 

Assigned taxonomy:
     Kingdom       Phylum        Class        Order       Family 
  5012(100%) 4993(99.62%) 5010(99.96%) 5006(99.88%) 5000(99.76%) 
       Genus      Species 
5010(99.96%)   5012(100%) 

Metadata variables: 9 
 SamName, SITE, WEEK, REGION, SEASON, REP, total_reads, total_otus, coverage</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_amp_pime, <span class="st">&quot;files/trepo/pime/rdata/ssu_merge_otu_amp_pime.rds&quot;</span>)</span></code></pre></div>
</div>
<h2 id="summary-1">Summary</h2>
<p>Let’s take a look at a table of sample information. Any header with the <code>_p</code> suffix is the <em>PIME filtered</em> data.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="rpirfampwp2p4vv05q7i" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="rpirfampwp2p4vv05q7i">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"11568\" data-max=\"12393\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"1444\" data-max=\"2152\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Sample summary table.\n            Use the buttons to navigate through the table or\n            download a copy.<\/em>\n<\/caption>","data":[["ALMR_W01_IB_HS","ALMR_W03_IB_HS","ALMR_W05_IB_HS","ALMR_W07_IB_HS","ALMR_W09_IB_HS","ALMR_W11_IB_HS","ALMR_W13_IB_NS","ALMR_W15_IB_NS","ALMR_W17_IB_NS","ALMR_W19_IB_NS","ALMR_W21_IB_NS","ALMR_W23_IB_NS","ALMR_W25_IB_NS","ALMR_W27_IB_NS","ALMR_W29_IB_NS","ALMR_W31_IB_NS","ALMR_W33_IB_NS","ALMR_W35_IB_NS","ALMR_W37_IB_NS","ALMR_W39_IB_NS","ALMR_W41_IB_HS","ALMR_W43_IB_HS","ALMR_W45_IB_HS","ALMR_W47_IB_HS","ALMR_W49_IB_HS","ALMR_W51_IB_HS","CRIS_W01_OB_HS","CRIS_W03_OB_HS","CRIS_W05_OB_HS","CRIS_W07_OB_HS","CRIS_W09_OB_HS","CRIS_W11_OB_HS","CRIS_W13_OB_NS","CRIS_W15_OB_NS","CRIS_W17_OB_NS","CRIS_W19_OB_NS","CRIS_W21_OB_NS","CRIS_W23_OB_NS","CRIS_W25_OB_NS","CRIS_W27_OB_NS","CRIS_W29_OB_NS","CRIS_W31_OB_NS","CRIS_W33_OB_NS","CRIS_W35_OB_NS","CRIS_W37_OB_NS","CRIS_W39_OB_NS","CRIS_W41_OB_HS","CRIS_W43_OB_HS","CRIS_W45_OB_HS","CRIS_W47_OB_HS","CRIS_W49_OB_HS","CRIS_W51_OB_HS","PAST_W01_IB_HS","PAST_W03_IB_HS","PAST_W05_IB_HS","PAST_W07_IB_HS","PAST_W09_IB_HS","PAST_W11_IB_HS","PAST_W13_IB_NS","PAST_W15_IB_NS","PAST_W17_IB_NS","PAST_W19_IB_NS","PAST_W21_IB_NS","PAST_W23_IB_NS","PAST_W25_IB_NS","PAST_W27_IB_NS","PAST_W29_IB_NS","PAST_W31_IB_NS","PAST_W33_IB_NS","PAST_W35_IB_NS","PAST_W37_IB_NS","PAST_W39_IB_NS","PAST_W41_IB_HS","PAST_W43_IB_HS","PAST_W45_IB_HS","PAST_W47_IB_HS","PAST_W49_IB_HS","PAST_W51_IB_HS","PUCL_W01_OB_HS","PUCL_W03_OB_HS","PUCL_W05_OB_HS","PUCL_W07_OB_HS","PUCL_W09_OB_HS","PUCL_W11_OB_HS","PUCL_W13_OB_NS","PUCL_W15_OB_NS","PUCL_W17_OB_NS","PUCL_W19_OB_NS","PUCL_W21_OB_NS","PUCL_W23_OB_NS","PUCL_W25_OB_NS","PUCL_W27_OB_NS","PUCL_W29_OB_NS","PUCL_W31_OB_NS","PUCL_W33_OB_NS","PUCL_W35_OB_NS","PUCL_W37_OB_NS","PUCL_W39_OB_NS","PUCL_W41_OB_HS","PUCL_W43_OB_HS","PUCL_W45_OB_HS","PUCL_W47_OB_HS","PUCL_W49_OB_HS","PUCL_W51_OB_HS"],["ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL"],["W01","W03","W05","W07","W09","W11","W13","W15","W17","W19","W21","W23","W25","W27","W29","W31","W33","W35","W37","W39","W41","W43","W45","W47","W49","W51","W01","W03","W05","W07","W09","W11","W13","W15","W17","W19","W21","W23","W25","W27","W29","W31","W33","W35","W37","W39","W41","W43","W45","W47","W49","W51","W01","W03","W05","W07","W09","W11","W13","W15","W17","W19","W21","W23","W25","W27","W29","W31","W33","W35","W37","W39","W41","W43","W45","W47","W49","W51","W01","W03","W05","W07","W09","W11","W13","W15","W17","W19","W21","W23","W25","W27","W29","W31","W33","W35","W37","W39","W41","W43","W45","W47","W49","W51"],["IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB"],["HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS"],["R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R","R"],[12319,12388,12347,12224,12240,12118,12268,12342,12203,12351,12224,12246,12280,12250,12307,12376,12216,12230,12347,12376,12311,12393,12276,12114,12167,12190,12254,12070,12254,12180,12144,12293,12144,12002,12270,12124,12178,12245,12300,12187,12283,12112,11943,12264,12234,12202,12173,12152,12147,12125,11956,12124,12259,12000,12167,12173,12162,12120,12280,12014,12269,12287,12131,12244,12202,12256,12120,12204,11568,12338,12221,12289,12217,12231,12050,11904,12239,12015,12319,12331,12268,12252,12309,12336,12204,12166,12241,12192,12271,12241,12179,12203,12254,12207,12064,12294,12289,12306,12221,12358,12134,11996,12200,12068],[1810,1722,1774,1911,1794,1631,1756,1715,1850,1583,1629,1872,1635,1873,1591,1586,1942,1876,1506,1875,1733,1691,1830,2002,1903,1878,1865,1879,1720,1825,1819,1719,1825,1876,1680,1828,1726,1697,1617,1676,1618,1707,1996,1707,1763,1736,1444,1750,1726,1866,1922,1812,1812,1716,1894,1862,1841,1666,1751,1514,1861,1613,1841,1688,1894,1570,1624,1673,1762,1769,1657,1682,1725,1677,2050,1936,1844,1815,1913,1969,1993,1850,1823,1778,1987,1902,1814,1851,1777,1698,1925,1712,1754,2077,2118,1956,1795,1674,1923,1569,2151,2152,2071,1863]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>SamName<\/th>\n      <th>SITE<\/th>\n      <th>WEEK<\/th>\n      <th>REGION<\/th>\n      <th>SEASON<\/th>\n      <th>REP<\/th>\n      <th>total_reads_p<\/th>\n      <th>total_otus_p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[5,15],"columnDefs":[{"className":"dt-right","targets":[6,7]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>And here is how the data sets changed through the PIME filtering process.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<table>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
object name
</th>
<th style="text-align:right;">
no. samples
</th>
<th style="text-align:right;">
total reads
</th>
<th style="text-align:right;">
total otus
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FULL data set
</td>
<td style="text-align:left;">
ssu_ps_work_merge_otu
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
6457054
</td>
<td style="text-align:right;">
21910
</td>
</tr>
<tr>
<td style="text-align:left;">
Rarefied data
</td>
<td style="text-align:left;">
ssu_pime_ds
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
1339728
</td>
<td style="text-align:right;">
15619
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME filtered data
</td>
<td style="text-align:left;">
ssu_ps_pime
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
1269326
</td>
<td style="text-align:right;">
5012
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (ALMR)
</td>
<td style="text-align:left;">
ssu_ps_pime_ALMR
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
319103
</td>
<td style="text-align:right;">
2684
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (CRIS)
</td>
<td style="text-align:left;">
ssu_ps_pime_CRIS
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
316360
</td>
<td style="text-align:right;">
2699
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PAST)
</td>
<td style="text-align:left;">
ssu_ps_pime_PAST
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
315960
</td>
<td style="text-align:right;">
2702
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PUCL)
</td>
<td style="text-align:left;">
ssu_ps_pime_PUCL
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
317903
</td>
<td style="text-align:right;">
2820
</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="all-samples-asv-data">All Samples (ASV Data)</h1>
<p>We will use PIME (Prevalence Interval for Microbiome Evaluation) to create a filtered data set.</p>
<h2 id="setup-2">Setup</h2>
<p>First, choose a phyloseq object and a sample data frame</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds <span class="ot">&lt;-</span> ssu_ps_work</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>ssu_which_pime <span class="ot">&lt;-</span> <span class="st">&quot;ssu_pime_ds&quot;</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="do">### USE THIS TO CREATE NEW SAMPLE VARIABLE FOR SPLITTING</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="do">##tmp_samp_data &lt;- sample_data(ssu_pime_ds) </span></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_samp_data$YR_TREAT &lt;- paste(tmp_samp_data$YEAR, tmp_samp_data$TREAT, sep=&quot;_&quot;)</span></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_ps &lt;- merge_phyloseq(ssu_pime_ds, tmp_samp_data)   </span></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_pime_ds &lt;- tmp_ps  </span></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a><span class="co">#rm(list = ls(pattern = &quot;tmp_&quot;))</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 47725 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 47725 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>ssu_pime_sample_d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(<span class="fu">otu_table</span>(ssu_pime_ds)))</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>ssu_pime_sample_d <span class="ot">&lt;-</span> ssu_pime_sample_d <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">total_reads =</span> <span class="dv">1</span>)</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds <span class="ot">&lt;-</span> <span class="fu">rarefy_even_depth</span>(ssu_pime_ds, </span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sample.size =</span> <span class="fu">min</span>(ssu_pime_sample_d<span class="sc">$</span>total_reads), </span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">trimOTUs =</span> <span class="cn">TRUE</span>, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">rngseed =</span> <span class="dv">119</span>)</span></code></pre></div>
</div>
<pre><code>`set.seed(119)` was used to initialize repeatable random subsampling.
Please record this for your records so others can reproduce.
Try `set.seed(119); .Random.seed` for the full vector

19907 OTUs were removed because they are no longer 
present in any sample after random subsampling
</code></pre>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 27818 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 27818 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<p>The first step in PIME is to define if the microbial community presents a high relative abundance of taxa with low prevalence, which is considered as noise in PIME analysis. This is calculated by random forests analysis and is the baseline noise detection.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>ssu_pime.oob.error <span class="ot">&lt;-</span> <span class="fu">pime.oob.error</span>(ssu_pime_ds, <span class="st">&quot;SITE&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] 0.02302632</code></pre>
</div>
<h2 id="split-by-predictor-variable-2">Split by Predictor Variable</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ssu_pime_ds))</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>ssu_per_variable_obj <span class="ot">&lt;-</span> <span class="fu">pime.split.by.variable</span>(ssu_pime_ds, <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>ssu_per_variable_obj</span></code></pre></div>
</div>
<details>
<summary>
Detailed results for splitting by sample variables
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$ALMR
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 12119 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 12119 taxa by 8 taxonomic ranks ]

$CRIS
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 12988 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 12988 taxa by 8 taxonomic ranks ]

$PAST
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 13539 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 13539 taxa by 8 taxonomic ranks ]

$PUCL
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 12939 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 12939 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</details>
<h2 id="calculate-prevalence-intervals-2">Calculate Prevalence Intervals</h2>
<p>Using the output of <code>pime.split.by.variable</code>, we calculate the prevalence intervals with the function <code>pime.prevalence</code>. This function estimates the highest prevalence possible (no empty ASV table), calculates prevalence for taxa, starting at 5 maximum prevalence possible (no empty ASV table or dropping samples). After prevalence calculation, each prevalence interval are merged.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>ssu_prevalences <span class="ot">&lt;-</span> <span class="fu">pime.prevalence</span>(ssu_per_variable_obj)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>ssu_prevalences</span></code></pre></div>
</div>
<details>
<summary>
Detailed results for all prevalences intervals
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$`5`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 11795 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 11795 taxa by 8 taxonomic ranks ]

$`10`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7795 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 7795 taxa by 8 taxonomic ranks ]

$`15`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 5610 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 5610 taxa by 8 taxonomic ranks ]

$`20`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4226 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 4226 taxa by 8 taxonomic ranks ]

$`25`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3341 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3341 taxa by 8 taxonomic ranks ]

$`30`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2819 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2819 taxa by 8 taxonomic ranks ]

$`35`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2297 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2297 taxa by 8 taxonomic ranks ]

$`40`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1854 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1854 taxa by 8 taxonomic ranks ]

$`45`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1537 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1537 taxa by 8 taxonomic ranks ]

$`50`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1269 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1269 taxa by 8 taxonomic ranks ]

$`55`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1123 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1123 taxa by 8 taxonomic ranks ]

$`60`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 962 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 962 taxa by 8 taxonomic ranks ]

$`65`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 813 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 813 taxa by 8 taxonomic ranks ]

$`70`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 679 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 679 taxa by 8 taxonomic ranks ]

$`75`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 551 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 551 taxa by 8 taxonomic ranks ]

$`80`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 481 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 481 taxa by 8 taxonomic ranks ]

$`85`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 371 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 371 taxa by 8 taxonomic ranks ]

$`90`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 264 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 264 taxa by 8 taxonomic ranks ]

$`95`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 168 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 168 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</details>
<h2 id="calculate-best-prevalence-2">Calculate Best Prevalence</h2>
<p>Finally, we use the function <code>pime.best.prevalence</code> to calculate the best prevalence. The function uses randomForest to build random forests trees for samples classification and variable importance computation. It performs classifications for each prevalence interval returned by <code>pime.prevalence</code>. Variable importance is calculated, returning the Mean Decrease Accuracy (MDA), Mean Decrease Impurity (MDI), overall and by sample group, and taxonomy for each ASV. PIME keeps the top 30 variables with highest MDA each prevalence level.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1911</span>)</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>ssu_best.prev <span class="ot">&lt;-</span> <span class="fu">pime.best.prevalence</span>(ssu_prevalences, <span class="st">&quot;SITE&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>         Interval OOB error rate (%)  OTUs  Nseqs
1   Prevalence 5%               0.33 11795 768526
2  Prevalence 10%                  0  7795 711567
3  Prevalence 15%                  0  5610 661834
4  Prevalence 20%                  0  4226 617129
5  Prevalence 25%                  0  3341 578009
6  Prevalence 30%                  0  2819 549694
7  Prevalence 35%                  0  2297 515763
8  Prevalence 40%                  0  1854 479844
9  Prevalence 45%                  0  1537 450527
10 Prevalence 50%                  0  1269 423402
11 Prevalence 55%                  0  1123 402686
12 Prevalence 60%                  0   962 378215
13 Prevalence 65%                  0   813 352608
14 Prevalence 70%                  0   679 324801
15 Prevalence 75%                  0   551 294565
16 Prevalence 80%                  0   481 273339
17 Prevalence 85%                  0   371 240756
18 Prevalence 90%                  0   264 202921
19 Prevalence 95%                  0   168 158804</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best <span class="ot">&lt;-</span> ssu_best.prev<span class="sc">$</span><span class="st">`</span><span class="at">OOB error</span><span class="st">`</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>)] <span class="ot">&lt;-</span> <span class="fu">sapply</span>(ssu_what_is_best[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>)], as.numeric)</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best <span class="ot">&lt;-</span> ssu_what_is_best <span class="sc">%&gt;%</span> </span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;OOB_error_rate&quot;</span> <span class="ot">=</span> <span class="st">&quot;OOB error rate (%)&quot;</span>)</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best<span class="sc">$</span>Interval <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(ssu_what_is_best<span class="sc">$</span>Interval, <span class="st">&quot;%&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>ssu_best <span class="ot">&lt;-</span> <span class="fu">with</span>(ssu_what_is_best, Interval[<span class="fu">which.min</span>(OOB_error_rate)])</span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>ssu_best <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;`&quot;</span>, ssu_best, <span class="st">&quot;`&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>ssu_prev_choice <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_best.prev$`Importance`$&quot;</span>, ssu_best, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>ssu_imp_best <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_prev_choice)))</span></code></pre></div>
</div>
<p>Looks like the lowest OOB error rate (%) that retains the most ASVs is 0% from <code>Prevalence 10</code>. We will use this interval.</p>
<h2 id="best-prevalence-summary-2">Best Prevalence Summary</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_imp_best[, 13:14] &lt;- list(NULL)</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>ssu_imp_best[,(<span class="fu">ncol</span>(ssu_imp_best)<span class="sc">-</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">ncol</span>(ssu_imp_best)] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="cn">NULL</span>)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>ssu_imp_best <span class="ot">&lt;-</span> ssu_imp_best <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;ASV_ID&quot;</span> <span class="ot">=</span> <span class="st">&quot;SequenceID&quot;</span>)</span></code></pre></div>
</div>
<p></br></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="jl2ze7jfzrnn9ytdht9x" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="jl2ze7jfzrnn9ytdht9x">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.002022612656222\" data-max=\"0.020816286714352\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.00008\" data-max=\"0.028103804205769\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000327948847964\" data-max=\"0.011945103602368\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000238095238095\" data-max=\"0.042421368998594\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.004789801014035\" data-max=\"0.016169335003166\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.853133321622435\" data-max=\"3.26954138887383\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Table of ASVs (max 30)\n                                     from the chosen prevalence\n                                     interval.<\/em>\n<\/caption>","data":[["ASV310","ASV10","ASV19","ASV41","ASV15","ASV558","ASV2","ASV17","ASV11","ASV3","ASV9","ASV531","ASV27","ASV57","ASV24","ASV154","ASV859","ASV243","ASV51","ASV89","ASV6","ASV58","ASV112","ASV78","ASV1007","ASV71","ASV68","ASV187","ASV25","ASV106"],[0.00554870825330176,0.011575460634284,0.011546561665377,0.00944154404209758,0.00804435598750115,0.00428725810255046,0.0179137506729781,0.00985830171560417,0.00808480113677662,0.010070507419701,0.00446327017357051,0.0020226126562225,0.00671669200306117,0.0090986557170965,0.00679799687869427,0.00782533405000796,0.00434445393977037,0.0208162867143519,0.0105052061731372,0.00645988432731686,0.00497395610507401,0.00544924862812794,0.00338262563523433,0.0159041527408059,0.00379830360382955,0.0166013736196239,0.00397280241634524,0.00411719355912904,0.00779431709414468,0.00349989446963863],[0.0113536320104729,0.0281038042057689,0.0117685219240705,0.0121777128906964,0.0101250449767077,0.00782756767595146,0.00684639012935732,0.0204347311199406,0.0113424725416898,0.00808295742077722,0.00519383250464019,0.022078199208113,0.00911612085724868,0.00791228209741534,0.00747790687275026,0.00905352752930339,0.0071505324841169,8e-05,0.00781419230709553,0.00653252044252044,0.00218339094225063,0.00113206105218063,0.00948206188299045,0.000922094508301405,0.0074376800478379,0.00119245569245569,0.00784927477291815,0.00166487949986829,0.00425062444197906,0.00489489885593334],[0.00573164526798712,0.0019399675328791,0.0119451036023672,0.00563610997816504,0.00804654925218612,0.0041185298035298,0.00497318683158938,0.00139117742325215,0.00628531119568675,0.00645123884926607,0.00476574566124967,0.00211364307605542,0.00631320926706621,0.00449489014497635,0.00376574978221775,0.00338216241816185,0.00383088303727855,0.00304708810254075,0.00032794884796463,0.00312593694174962,0.0024123428260193,0.00212298121668437,0.00457730225318968,0.00377530350476846,0.0036126036874725,0.00206287950066283,0.00449494336586661,0.00446000693750694,0.00463324087390118,0.00543419406623242],[0.0424213689985935,0.0117760371795199,0.00975915138570344,0.00838231399219163,0.00793467928031332,0.0169406758790379,0.00411380484872771,0.000701699380563336,0.00674897365511822,0.00546468296381763,0.0148232625654522,0.00285503776064121,0.00533119389344034,0.00411368861024033,0.00723293381248756,0.00529399146295113,0.00917098226156902,0.000238095238095238,0.00586623969765625,0.00558145347695257,0.011021127297467,0.0132486268802192,0.00458337898660479,0.00112859335038363,0.00626433252778569,0.001740500778687,0.00514885851864073,0.0106622353980075,0.00399089671614409,0.00544499192040774],[0.0161693350031659,0.0132591434968995,0.0111236441443563,0.00875176469218689,0.00850190516821503,0.00824401718326511,0.00816591441082952,0.00800223962178549,0.00795718491015643,0.00739555897593019,0.0072293630226664,0.00719848893866112,0.00684531773088194,0.00631217805108937,0.00630256220406663,0.00627766047374075,0.00611367506820535,0.00610670127893827,0.00597279296439194,0.0055248398292405,0.00542910193305607,0.00541219435675353,0.00538644335329824,0.00537901231100101,0.00537583940036713,0.00531952105704878,0.00529460819971727,0.00528403643768353,0.00493742485188554,0.00478980101403518],[3.26954138887383,3.17652125174905,1.88455809103126,1.36942530982996,1.54041281349542,1.90497234987096,1.51397143721584,1.5987291543215,1.22657841308271,1.33708830258795,1.80863893306407,1.2247939978363,1.48126574185278,1.29095908243943,1.11773803671703,1.18347887237273,1.98426490645245,1.55070041279661,1.12778442354173,0.872665634546511,1.02177360810821,1.50216603436174,0.956830809351021,1.62088493205187,1.50519957677425,1.37210259449198,1.10679849619978,1.35011246674115,1.43054144982062,0.853133321622435],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],["Nitrospirota","Acidobacteriota","Proteobacteria","Proteobacteria","Nitrospirota","Nitrospirota","Proteobacteria","Nitrospirota","Desulfobacterota","Proteobacteria","Chloroflexi","Chloroflexi","Desulfobacterota","Chloroflexi","k_Archaea","Proteobacteria","Nitrospirota","Chloroflexi","Chloroflexi","Proteobacteria","Acidobacteriota","Acidobacteriota","Desulfobacterota","Acidobacteriota","Chloroflexi","Proteobacteria","Desulfobacterota","Nitrospirota","Chloroflexi","Acidobacteriota"],["Thermodesulfovibrionia","Thermoanaerobaculia","Gammaproteobacteria","Gammaproteobacteria","Thermodesulfovibrionia","Thermodesulfovibrionia","Gammaproteobacteria","Nitrospiria","Syntrophobacteria","Gammaproteobacteria","Anaerolineae","Anaerolineae","Syntrophobacteria","Anaerolineae","k_Archaea","Gammaproteobacteria","Thermodesulfovibrionia","Anaerolineae","Anaerolineae","Gammaproteobacteria","Aminicenantia","Thermoanaerobaculia","Syntrophobacteria","Aminicenantia","Dehalococcoidia","Gammaproteobacteria","Desulfobacteria","Thermodesulfovibrionia","Anaerolineae","Thermoanaerobaculia"],["c_Thermodesulfovibrionia","Thermoanaerobaculales","Steroidobacterales","Ectothiorhodospirales","c_Thermodesulfovibrionia","c_Thermodesulfovibrionia","Ectothiorhodospirales","Nitrospirales","Syntrophobacterales","Ectothiorhodospirales","Anaerolineales","Anaerolineales","Syntrophobacterales","Anaerolineales","k_Archaea","c_Gammaproteobacteria","c_Thermodesulfovibrionia","c_Anaerolineae","Anaerolineales","Ectothiorhodospirales","Aminicenantales","Thermoanaerobaculales","Syntrophobacterales","Aminicenantales","c_Dehalococcoidia","Ectothiorhodospirales","Desulfobacterales","c_Thermodesulfovibrionia","Anaerolineales","Thermoanaerobaculales"],["c_Thermodesulfovibrionia","Thermoanaerobaculaceae","Woeseiaceae","Thioalkalispiraceae","c_Thermodesulfovibrionia","c_Thermodesulfovibrionia","Thioalkalispiraceae","Nitrospiraceae","o_Syntrophobacterales","Thioalkalispiraceae","Anaerolineaceae","Anaerolineaceae","o_Syntrophobacterales","Anaerolineaceae","k_Archaea","c_Gammaproteobacteria","c_Thermodesulfovibrionia","c_Anaerolineae","Anaerolineaceae","Thioalkalispiraceae","o_Aminicenantales","Thermoanaerobaculaceae","o_Syntrophobacterales","o_Aminicenantales","c_Dehalococcoidia","Thioalkalispiraceae","Desulfosarcinaceae","c_Thermodesulfovibrionia","Anaerolineaceae","Thermoanaerobaculaceae"],["c_Thermodesulfovibrionia","Subgroup_10","Woeseia","Thiohalophilus","c_Thermodesulfovibrionia","c_Thermodesulfovibrionia","Thiohalophilus","Nitrospira","o_Syntrophobacterales","Thiohalophilus","f_Anaerolineaceae","f_Anaerolineaceae","o_Syntrophobacterales","f_Anaerolineaceae","k_Archaea","c_Gammaproteobacteria","c_Thermodesulfovibrionia","c_Anaerolineae","f_Anaerolineaceae","Thiohalophilus","o_Aminicenantales","Subgroup_10","o_Syntrophobacterales","o_Aminicenantales","c_Dehalococcoidia","Thiohalophilus","Sva0081_sediment_group","c_Thermodesulfovibrionia","f_Anaerolineaceae","Subgroup_10"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ASV_ID<\/th>\n      <th>ALMR<\/th>\n      <th>CRIS<\/th>\n      <th>PAST<\/th>\n      <th>PUCL<\/th>\n      <th>MeanDecreaseAccuracy<\/th>\n      <th>MeanDecreaseGini<\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\nvar value=data[10]; $(this.api().cell(row, 10).node()).css({'font-size':'80%'});\nvar value=data[11]; $(this.api().cell(row, 11).node()).css({'font-size':'80%'});\nvar value=data[12]; $(this.api().cell(row, 12).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>Now we need to create a phyloseq object of ASVs at this cutoff (<code>Prevalence 10</code>).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>ssu_best_val <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(ssu_best, <span class="st">&quot;Prevalence &quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>ssu_best_val <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_prevalences$&quot;</span>, ssu_best_val, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>ssu_prevalence_best <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_best_val)))</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_prevalence_best, <span class="st">&quot;rdata/trepo/ssu_asv_prevalence_best.rds&quot;</span>)</span></code></pre></div>
</div>
<p>And look at a summary of the data.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7795 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 7795 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>2027</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>2482</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>711567</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>2341</td>
</tr>
<tr class="odd">
<td>Mean number of reads</td>
<td>2341</td>
</tr>
<tr class="even">
<td>Median number of reads</td>
<td>2347.5</td>
</tr>
<tr class="odd">
<td>Total ASVS</td>
<td>7795</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.863</td>
</tr>
</tbody>
</table>
<h2 id="estimate-error-in-prediction-2">Estimate Error in Prediction</h2>
<p>Using the function <code>pime.error.prediction</code> we can estimate the error in prediction. For each prevalence interval, this function randomizes the samples labels into arbitrary groupings using <code>n</code> random permutations, defined by the <code>bootstrap</code> value. For each, randomized and prevalence filtered, data set the OOB error rate is calculated to estimate whether the original differences in groups of samples occur by chance. Results are in a list containing a table and a box plot summarizing the results.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>ssu_randomized <span class="ot">&lt;-</span> <span class="fu">pime.error.prediction</span>(ssu_pime_ds, <span class="st">&quot;TEMP&quot;</span>,</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">bootstrap =</span> <span class="dv">100</span>, <span class="at">parallel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">max.prev =</span> <span class="dv">95</span>)</span></code></pre></div>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>It is also possible to estimate the variation of OOB error for each prevalence interval filtering. This is done by running the random forests classification for <code>n</code> times, determined by the <code>bootstrap</code> value. The function will return a box plot figure and a table for each classification error.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>ssu_replicated.oob.error <span class="ot">&lt;-</span> <span class="fu">pime.oob.replicate</span>(ssu_prevalences, <span class="st">&quot;SITE&quot;</span>, </span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">bootstrap =</span> <span class="dv">100</span>, <span class="at">parallel =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="trepo-pime_files/figure-html5/unnamed-chunk-115-1.png" width="624" /></p>
</div>
<p>To obtain the confusion matrix from random forests classification use the following:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>ssu_prev_confuse <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_best.prev$`Confusion`$&quot;</span>, ssu_best, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_prev_confuse)))</span></code></pre></div>
</div>
<h2 id="save-phyloseq-pime-objects-2">Save Phyloseq PIME objects</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime <span class="ot">&lt;-</span> ssu_prevalence_best</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ssu_ps_pime), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ssu_ps_pime))</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ssu_ps_pime,</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>                               sample_data,</span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>                               ssu_ps_pime_tree)</span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_asv_pime.rds&quot;</span>)</span></code></pre></div>
</div>
<h3 id="split-save-by-predictor-variable-2">Split &amp; save by predictor variable</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime_split <span class="ot">&lt;-</span> <span class="fu">pime.split.by.variable</span>(ssu_ps_pime, <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>ALMR, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_asv_pime_ALMR.rds&quot;</span>)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>CRIS, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_asv_pime_CRIS.rds&quot;</span>)</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>PAST, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_asv_pime_PAST.rds&quot;</span>)</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>PUCL, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_asv_pime_PUCL.rds&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$ALMR
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3417 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3417 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 3417 tips and 3416 internal nodes ]

$CRIS
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3336 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3336 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 3336 tips and 3335 internal nodes ]

$PAST
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3471 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3471 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 3471 tips and 3470 internal nodes ]

$PUCL
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3859 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3859 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 3859 tips and 3858 internal nodes ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="create-ampvis2-pime-object-2">Create Ampvis2 PIME Object</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>ssu_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ssu_ps_pime)))</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>ssu_otu[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(ssu_otu, as.numeric)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>ssu_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(ssu_otu)</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>ssu_tax <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu_ps_pime)))</span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>ssu_samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ssu_ps_pime))</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>ssu_ps_amp <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(ssu_otu, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>),</span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">tax_table</span>(ssu_tax, ssu_tax),</span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">sample_data</span>(ssu_samples))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>ssu_amp_asv_pime  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">otu_table</span>(ssu_ps_amp))</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>ssu_amp_asv_pime <span class="ot">&lt;-</span> ssu_amp_asv_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu_ps_amp))</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime <span class="ot">&lt;-</span> ssu_amp_tax_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ssu_amp_tax_pime)[<span class="fu">colnames</span>(ssu_amp_tax_pime) <span class="sc">==</span> <span class="st">&quot;ASV_ID&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Species&quot;</span></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>ssu_amp_asv_tax_pime <span class="ot">&lt;-</span> <span class="fu">left_join</span>(ssu_amp_asv_pime, ssu_amp_tax_pime, <span class="at">by =</span> <span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>ssu_samp_data_t <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu_samp_data)</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_samp_data_t[,7:9] &lt;- NULL</span></span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>ssu_amp_data_pime <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(ssu_amp_asv_tax_pime, <span class="at">metadata =</span> ssu_samp_data_t)</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span></span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">amp_alphadiv</span>(</span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a>    ssu_amp_data_pime,</span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">measure =</span> <span class="st">&quot;observed&quot;</span>,</span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">richness =</span> <span class="cn">FALSE</span>)</span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span></span>
<span id="cb79-19"><a href="#cb79-19" aria-hidden="true" tabindex="-1"></a>  ssu_diversity_pime <span class="sc">%&gt;%</span></span>
<span id="cb79-20"><a href="#cb79-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb79-21"><a href="#cb79-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;total_reads&quot;</span> <span class="ot">=</span> <span class="st">&quot;Reads&quot;</span>,</span>
<span id="cb79-22"><a href="#cb79-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;total_asvs&quot;</span> <span class="ot">=</span> <span class="st">&quot;ObservedOTUs&quot;</span>)</span>
<span id="cb79-23"><a href="#cb79-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-24"><a href="#cb79-24" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime<span class="sc">$</span>coverage <span class="ot">=</span> <span class="fu">cut</span>(ssu_diversity_pime<span class="sc">$</span>total_reads,</span>
<span id="cb79-25"><a href="#cb79-25" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5000</span>, <span class="dv">25000</span>, <span class="dv">100000</span>, <span class="dv">200000</span>),</span>
<span id="cb79-26"><a href="#cb79-26" aria-hidden="true" tabindex="-1"></a>                          <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb79-27"><a href="#cb79-27" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;low (&lt; 5k)&quot;</span>, <span class="st">&quot;medium (5-25k)&quot;</span>,</span>
<span id="cb79-28"><a href="#cb79-28" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;high (25-100k)&quot;</span>, <span class="st">&quot;extra_high (&gt; 100k)&quot;</span>))</span>
<span id="cb79-29"><a href="#cb79-29" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span> ssu_diversity_pime[<span class="fu">order</span>(ssu_diversity_pime<span class="sc">$</span>SamName), ]</span>
<span id="cb79-30"><a href="#cb79-30" aria-hidden="true" tabindex="-1"></a>ssu_amp_pime <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(ssu_amp_asv_tax_pime,</span>
<span id="cb79-31"><a href="#cb79-31" aria-hidden="true" tabindex="-1"></a>                          <span class="at">metadata =</span> ssu_diversity_pime, <span class="at">tree =</span> ssu_ps_pime_tree)</span>
<span id="cb79-32"><a href="#cb79-32" aria-hidden="true" tabindex="-1"></a>ssu_samp_pime <span class="ot">&lt;-</span> ssu_diversity_pime</span>
<span id="cb79-33"><a href="#cb79-33" aria-hidden="true" tabindex="-1"></a>ssu_samp_pime <span class="ot">&lt;-</span> ssu_samp_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>ampvis2 object with 4 elements. 
Summary of OTU table:
     Samples         OTUs  Total#Reads    Min#Reads    Max#Reads 
         304         7795       711567         2027         2482 
Median#Reads    Avg#Reads 
      2347.5      2340.68 

Assigned taxonomy:
     Kingdom       Phylum        Class        Order       Family 
  7795(100%) 7786(99.88%) 7794(99.99%) 7791(99.95%) 7775(99.74%) 
       Genus      Species 
  7795(100%)   7795(100%) 

Metadata variables: 9 
 SamName, SITE, WEEK, REGION, SEASON, REP, total_reads, total_asvs, coverage</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_amp_pime, <span class="st">&quot;files/trepo/pime/rdata/ssu_asv_amp_pime.rds&quot;</span>)</span></code></pre></div>
</div>
<h2 id="summary-2">Summary</h2>
<p>Let’s take a look at a table of sample information. Any header with the <code>_p</code> suffix is the <em>PIME filtered</em> data.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="upipqr0laqwmhi8akm2x" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="upipqr0laqwmhi8akm2x">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"2027\" data-max=\"2482\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"665\" data-max=\"1381\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Sample summary table.\n            Use the buttons to navigate through the table or\n            download a copy.<\/em>\n<\/caption>","data":[["ALMR_W01_IB_HS_R1","ALMR_W01_IB_HS_R2","ALMR_W01_IB_HS_R3","ALMR_W03_IB_HS_R1","ALMR_W03_IB_HS_R2","ALMR_W03_IB_HS_R3","ALMR_W05_IB_HS_R1","ALMR_W05_IB_HS_R2","ALMR_W05_IB_HS_R3","ALMR_W07_IB_HS_R1","ALMR_W07_IB_HS_R2","ALMR_W07_IB_HS_R3","ALMR_W09_IB_HS_R1","ALMR_W09_IB_HS_R2","ALMR_W09_IB_HS_R3","ALMR_W11_IB_HS_R1","ALMR_W11_IB_HS_R2","ALMR_W11_IB_HS_R3","ALMR_W13_IB_NS_R1","ALMR_W13_IB_NS_R2","ALMR_W13_IB_NS_R3","ALMR_W15_IB_NS_R1","ALMR_W15_IB_NS_R2","ALMR_W15_IB_NS_R3","ALMR_W17_IB_NS_R1","ALMR_W17_IB_NS_R2","ALMR_W17_IB_NS_R3","ALMR_W19_IB_NS_R1","ALMR_W19_IB_NS_R2","ALMR_W19_IB_NS_R3","ALMR_W21_IB_NS_R1","ALMR_W21_IB_NS_R2","ALMR_W21_IB_NS_R3","ALMR_W23_IB_NS_R1","ALMR_W23_IB_NS_R2","ALMR_W23_IB_NS_R3","ALMR_W25_IB_NS_R1","ALMR_W25_IB_NS_R2","ALMR_W25_IB_NS_R3","ALMR_W27_IB_NS_R1","ALMR_W27_IB_NS_R2","ALMR_W27_IB_NS_R3","ALMR_W29_IB_NS_R1","ALMR_W29_IB_NS_R2","ALMR_W31_IB_NS_R1","ALMR_W31_IB_NS_R2","ALMR_W31_IB_NS_R3","ALMR_W33_IB_NS_R1","ALMR_W33_IB_NS_R2","ALMR_W33_IB_NS_R3","ALMR_W35_IB_NS_R1","ALMR_W35_IB_NS_R2","ALMR_W35_IB_NS_R3","ALMR_W37_IB_NS_R2","ALMR_W37_IB_NS_R3","ALMR_W39_IB_NS_R1","ALMR_W39_IB_NS_R2","ALMR_W39_IB_NS_R3","ALMR_W41_IB_HS_R1","ALMR_W41_IB_HS_R2","ALMR_W41_IB_HS_R3","ALMR_W43_IB_HS_R1","ALMR_W43_IB_HS_R2","ALMR_W43_IB_HS_R3","ALMR_W45_IB_HS_R1","ALMR_W45_IB_HS_R2","ALMR_W45_IB_HS_R3","ALMR_W47_IB_HS_R1","ALMR_W47_IB_HS_R2","ALMR_W47_IB_HS_R3","ALMR_W49_IB_HS_R1","ALMR_W49_IB_HS_R2","ALMR_W49_IB_HS_R3","ALMR_W51_IB_HS_R1","ALMR_W51_IB_HS_R2","ALMR_W51_IB_HS_R3","CRIS_W01_OB_HS_R1","CRIS_W01_OB_HS_R2","CRIS_W01_OB_HS_R3","CRIS_W03_OB_HS_R1","CRIS_W03_OB_HS_R2","CRIS_W03_OB_HS_R3","CRIS_W05_OB_HS_R1","CRIS_W05_OB_HS_R2","CRIS_W05_OB_HS_R3","CRIS_W07_OB_HS_R1","CRIS_W07_OB_HS_R2","CRIS_W07_OB_HS_R3","CRIS_W09_OB_HS_R1","CRIS_W09_OB_HS_R2","CRIS_W09_OB_HS_R3","CRIS_W11_OB_HS_R1","CRIS_W11_OB_HS_R2","CRIS_W11_OB_HS_R3","CRIS_W13_OB_NS_R1","CRIS_W13_OB_NS_R2","CRIS_W13_OB_NS_R3","CRIS_W15_OB_NS_R1","CRIS_W15_OB_NS_R2","CRIS_W15_OB_NS_R3","CRIS_W17_OB_NS_R1","CRIS_W17_OB_NS_R2","CRIS_W17_OB_NS_R3","CRIS_W19_OB_NS_R1","CRIS_W19_OB_NS_R2","CRIS_W19_OB_NS_R3","CRIS_W21_OB_NS_R1","CRIS_W21_OB_NS_R2","CRIS_W21_OB_NS_R3","CRIS_W23_OB_NS_R1","CRIS_W23_OB_NS_R2","CRIS_W23_OB_NS_R3","CRIS_W25_OB_NS_R1","CRIS_W25_OB_NS_R2","CRIS_W25_OB_NS_R3","CRIS_W27_OB_NS_R1","CRIS_W27_OB_NS_R2","CRIS_W27_OB_NS_R3","CRIS_W29_OB_NS_R1","CRIS_W29_OB_NS_R2","CRIS_W29_OB_NS_R3","CRIS_W31_OB_NS_R1","CRIS_W31_OB_NS_R2","CRIS_W31_OB_NS_R3","CRIS_W33_OB_NS_R1","CRIS_W33_OB_NS_R2","CRIS_W33_OB_NS_R3","CRIS_W35_OB_NS_R1","CRIS_W35_OB_NS_R2","CRIS_W35_OB_NS_R3","CRIS_W37_OB_NS_R1","CRIS_W37_OB_NS_R2","CRIS_W37_OB_NS_R3","CRIS_W39_OB_NS_R1","CRIS_W39_OB_NS_R2","CRIS_W39_OB_NS_R3","CRIS_W41_OB_HS_R1","CRIS_W41_OB_HS_R2","CRIS_W43_OB_HS_R1","CRIS_W43_OB_HS_R2","CRIS_W43_OB_HS_R3","CRIS_W45_OB_HS_R2","CRIS_W45_OB_HS_R3","CRIS_W47_OB_HS_R1","CRIS_W47_OB_HS_R2","CRIS_W47_OB_HS_R3","CRIS_W49_OB_HS_R1","CRIS_W49_OB_HS_R2","CRIS_W49_OB_HS_R3","CRIS_W51_OB_HS_R1","CRIS_W51_OB_HS_R2","CRIS_W51_OB_HS_R3","PAST_W01_IB_HS_R1","PAST_W01_IB_HS_R2","PAST_W01_IB_HS_R3","PAST_W03_IB_HS_R1","PAST_W03_IB_HS_R2","PAST_W03_IB_HS_R3","PAST_W05_IB_HS_R1","PAST_W05_IB_HS_R2","PAST_W05_IB_HS_R3","PAST_W07_IB_HS_R1","PAST_W07_IB_HS_R2","PAST_W07_IB_HS_R3","PAST_W09_IB_HS_R1","PAST_W09_IB_HS_R2","PAST_W09_IB_HS_R3","PAST_W11_IB_HS_R1","PAST_W11_IB_HS_R2","PAST_W11_IB_HS_R3","PAST_W13_IB_NS_R1","PAST_W13_IB_NS_R2","PAST_W13_IB_NS_R3","PAST_W15_IB_NS_R1","PAST_W15_IB_NS_R2","PAST_W15_IB_NS_R3","PAST_W17_IB_NS_R1","PAST_W17_IB_NS_R2","PAST_W17_IB_NS_R3","PAST_W19_IB_NS_R1","PAST_W19_IB_NS_R2","PAST_W19_IB_NS_R3","PAST_W21_IB_NS_R1","PAST_W21_IB_NS_R2","PAST_W21_IB_NS_R3","PAST_W23_IB_NS_R1","PAST_W23_IB_NS_R2","PAST_W23_IB_NS_R3","PAST_W25_IB_NS_R1","PAST_W25_IB_NS_R2","PAST_W25_IB_NS_R3","PAST_W27_IB_NS_R1","PAST_W27_IB_NS_R2","PAST_W29_IB_NS_R1","PAST_W29_IB_NS_R2","PAST_W29_IB_NS_R3","PAST_W31_IB_NS_R1","PAST_W31_IB_NS_R2","PAST_W31_IB_NS_R3","PAST_W33_IB_NS_R1","PAST_W33_IB_NS_R2","PAST_W33_IB_NS_R3","PAST_W35_IB_NS_R1","PAST_W35_IB_NS_R2","PAST_W35_IB_NS_R3","PAST_W37_IB_NS_R1","PAST_W37_IB_NS_R3","PAST_W39_IB_NS_R1","PAST_W39_IB_NS_R2","PAST_W39_IB_NS_R3","PAST_W41_IB_HS_R1","PAST_W41_IB_HS_R2","PAST_W41_IB_HS_R3","PAST_W43_IB_HS_R1","PAST_W43_IB_HS_R2","PAST_W43_IB_HS_R3","PAST_W45_IB_HS_R1","PAST_W45_IB_HS_R2","PAST_W45_IB_HS_R3","PAST_W47_IB_HS_R1","PAST_W47_IB_HS_R2","PAST_W47_IB_HS_R3","PAST_W49_IB_HS_R1","PAST_W49_IB_HS_R2","PAST_W49_IB_HS_R3","PAST_W51_IB_HS_R1","PAST_W51_IB_HS_R2","PAST_W51_IB_HS_R3","PUCL_W01_OB_HS_R1","PUCL_W01_OB_HS_R2","PUCL_W01_OB_HS_R3","PUCL_W03_OB_HS_R1","PUCL_W03_OB_HS_R2","PUCL_W03_OB_HS_R3","PUCL_W05_OB_HS_R1","PUCL_W05_OB_HS_R2","PUCL_W05_OB_HS_R3","PUCL_W07_OB_HS_R1","PUCL_W07_OB_HS_R2","PUCL_W07_OB_HS_R3","PUCL_W09_OB_HS_R1","PUCL_W09_OB_HS_R2","PUCL_W09_OB_HS_R3","PUCL_W11_OB_HS_R1","PUCL_W11_OB_HS_R2","PUCL_W11_OB_HS_R3","PUCL_W13_OB_NS_R1","PUCL_W13_OB_NS_R2","PUCL_W13_OB_NS_R3","PUCL_W15_OB_NS_R1","PUCL_W15_OB_NS_R2","PUCL_W15_OB_NS_R3","PUCL_W17_OB_NS_R1","PUCL_W17_OB_NS_R2","PUCL_W17_OB_NS_R3","PUCL_W19_OB_NS_R1","PUCL_W19_OB_NS_R2","PUCL_W19_OB_NS_R3","PUCL_W21_OB_NS_R1","PUCL_W21_OB_NS_R2","PUCL_W21_OB_NS_R3","PUCL_W23_OB_NS_R1","PUCL_W23_OB_NS_R2","PUCL_W23_OB_NS_R3","PUCL_W25_OB_NS_R1","PUCL_W25_OB_NS_R2","PUCL_W25_OB_NS_R3","PUCL_W27_OB_NS_R1","PUCL_W27_OB_NS_R2","PUCL_W27_OB_NS_R3","PUCL_W29_OB_NS_R1","PUCL_W29_OB_NS_R2","PUCL_W29_OB_NS_R3","PUCL_W31_OB_NS_R1","PUCL_W31_OB_NS_R2","PUCL_W31_OB_NS_R3","PUCL_W33_OB_NS_R1","PUCL_W33_OB_NS_R2","PUCL_W33_OB_NS_R3","PUCL_W35_OB_NS_R1","PUCL_W35_OB_NS_R2","PUCL_W37_OB_NS_R1","PUCL_W37_OB_NS_R2","PUCL_W37_OB_NS_R3","PUCL_W39_OB_NS_R1","PUCL_W39_OB_NS_R2","PUCL_W39_OB_NS_R3","PUCL_W41_OB_HS_R1","PUCL_W41_OB_HS_R2","PUCL_W41_OB_HS_R3","PUCL_W43_OB_HS_R1","PUCL_W43_OB_HS_R3","PUCL_W45_OB_HS_R1","PUCL_W45_OB_HS_R2","PUCL_W45_OB_HS_R3","PUCL_W47_OB_HS_R1","PUCL_W47_OB_HS_R2","PUCL_W47_OB_HS_R3","PUCL_W49_OB_HS_R1","PUCL_W49_OB_HS_R2","PUCL_W49_OB_HS_R3","PUCL_W51_OB_HS_R1","PUCL_W51_OB_HS_R2","PUCL_W51_OB_HS_R3"],["ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL"],["W01","W01","W01","W03","W03","W03","W05","W05","W05","W07","W07","W07","W09","W09","W09","W11","W11","W11","W13","W13","W13","W15","W15","W15","W17","W17","W17","W19","W19","W19","W21","W21","W21","W23","W23","W23","W25","W25","W25","W27","W27","W27","W29","W29","W31","W31","W31","W33","W33","W33","W35","W35","W35","W37","W37","W39","W39","W39","W41","W41","W41","W43","W43","W43","W45","W45","W45","W47","W47","W47","W49","W49","W49","W51","W51","W51","W01","W01","W01","W03","W03","W03","W05","W05","W05","W07","W07","W07","W09","W09","W09","W11","W11","W11","W13","W13","W13","W15","W15","W15","W17","W17","W17","W19","W19","W19","W21","W21","W21","W23","W23","W23","W25","W25","W25","W27","W27","W27","W29","W29","W29","W31","W31","W31","W33","W33","W33","W35","W35","W35","W37","W37","W37","W39","W39","W39","W41","W41","W43","W43","W43","W45","W45","W47","W47","W47","W49","W49","W49","W51","W51","W51","W01","W01","W01","W03","W03","W03","W05","W05","W05","W07","W07","W07","W09","W09","W09","W11","W11","W11","W13","W13","W13","W15","W15","W15","W17","W17","W17","W19","W19","W19","W21","W21","W21","W23","W23","W23","W25","W25","W25","W27","W27","W29","W29","W29","W31","W31","W31","W33","W33","W33","W35","W35","W35","W37","W37","W39","W39","W39","W41","W41","W41","W43","W43","W43","W45","W45","W45","W47","W47","W47","W49","W49","W49","W51","W51","W51","W01","W01","W01","W03","W03","W03","W05","W05","W05","W07","W07","W07","W09","W09","W09","W11","W11","W11","W13","W13","W13","W15","W15","W15","W17","W17","W17","W19","W19","W19","W21","W21","W21","W23","W23","W23","W25","W25","W25","W27","W27","W27","W29","W29","W29","W31","W31","W31","W33","W33","W33","W35","W35","W37","W37","W37","W39","W39","W39","W41","W41","W41","W43","W43","W45","W45","W45","W47","W47","W47","W49","W49","W49","W51","W51","W51"],["IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB"],["HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS"],["R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R1","R2","R3","R1","R2","R3","R1","R2","R3","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R1","R2","R3","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3"],[2349,2458,2353,2416,2415,2413,2416,2350,2393,2306,2338,2345,2335,2371,2317,2316,2398,2305,2388,2453,2361,2371,2427,2389,2384,2333,2349,2414,2366,2475,2348,2372,2326,2398,2323,2358,2419,2347,2420,2285,2373,2319,2401,2388,2418,2411,2435,2356,2312,2328,2347,2354,2383,2380,2377,2394,2367,2385,2403,2395,2384,2429,2401,2411,2354,2392,2369,2274,2310,2325,2334,2328,2293,2353,2303,2306,2341,2377,2375,2262,2239,2285,2369,2482,2306,2321,2307,2382,2392,2273,2322,2350,2338,2366,2272,2302,2409,2356,2291,2314,2372,2373,2367,2304,2353,2335,2283,2382,2330,2253,2333,2360,2358,2400,2367,2358,2325,2386,2322,2422,2324,2302,2303,2304,2244,2264,2342,2382,2382,2328,2309,2288,2320,2317,2340,2380,2290,2339,2274,2349,2349,2352,2305,2243,2281,2303,2251,2100,2296,2293,2353,2346,2372,2346,2370,2248,2289,2253,2329,2330,2314,2262,2310,2319,2295,2355,2349,2335,2319,2277,2362,2412,2370,2205,2215,2238,2335,2344,2366,2428,2337,2352,2346,2330,2315,2315,2385,2389,2363,2339,2303,2366,2377,2320,2210,2223,2392,2308,2319,2108,2027,2043,2369,2405,2398,2340,2359,2361,2316,2388,2375,2324,2302,2258,2343,2330,2250,2260,2261,2239,2368,2197,2305,2387,2395,2330,2341,2252,2354,2397,2386,2401,2367,2398,2349,2407,2347,2317,2355,2334,2385,2387,2414,2383,2408,2421,2381,2288,2351,2385,2331,2312,2374,2303,2365,2361,2347,2291,2356,2403,2377,2350,2385,2385,2353,2347,2332,2337,2360,2322,2378,2365,2368,2342,2337,2293,2365,2276,2306,2332,2377,2409,2379,2406,2334,2306,2382,2394,2357,2332,2372,2362,2276,2287,2339,2284,2299,2267,2319,2382,2449,2255,2384,2322],[969,1128,1155,1068,1062,1062,1154,1103,918,1195,1129,1217,1131,1125,1051,1009,1074,1030,1167,930,1016,1054,1107,917,1111,1107,1128,915,926,933,1004,1004,965,854,1156,933,993,1007,956,1126,1068,1032,979,1016,892,969,1013,1134,1214,1183,1169,1138,1123,918,1023,996,1184,1157,1114,1042,968,730,1016,1040,1092,1048,1185,1099,1191,1213,1187,1128,1187,1181,1125,1144,1111,1137,1124,1102,1142,940,1016,691,1133,1106,1083,1164,1073,1157,1008,994,977,951,1014,1153,883,972,1191,880,995,1049,1018,1159,963,1006,1098,682,912,836,1089,999,895,978,951,1043,1011,964,950,911,944,966,963,906,1207,1129,1135,1021,1026,1069,961,805,1069,1051,1077,942,820,962,979,984,1015,947,1119,1088,1082,1149,1102,724,1150,1117,1062,1120,1129,1058,1123,985,1016,1139,1150,1088,1161,1121,1155,1127,1164,1172,1116,1048,892,968,1079,978,1044,907,927,884,1138,1094,990,923,986,994,1139,1103,976,965,1068,970,1058,1143,1193,945,1002,937,958,952,1111,1041,1027,900,1004,1008,1077,1070,1088,1053,1052,1023,969,886,1102,665,1076,1011,900,794,1224,1208,1225,1172,993,1125,1133,1063,1102,1129,1146,992,1161,1078,1240,1244,1286,1252,1246,1311,1131,1192,1181,1073,1040,1026,1230,1142,1075,1038,950,1229,1228,1200,1134,1151,1163,934,1113,1031,1172,1119,1093,1048,1089,972,1065,971,1250,1183,1217,998,1123,740,1084,966,1041,817,1261,1300,1233,1368,1350,1262,1220,922,1115,1199,902,728,1077,1192,1211,1057,1063,965,1363,1340,1229,1323,1381,1193,1293,1348,1130,1061,1054,1238]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>SamName<\/th>\n      <th>SITE<\/th>\n      <th>WEEK<\/th>\n      <th>REGION<\/th>\n      <th>SEASON<\/th>\n      <th>REP<\/th>\n      <th>total_reads_p<\/th>\n      <th>total_asvs_p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[5,15],"columnDefs":[{"className":"dt-right","targets":[6,7]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>And here is how the data sets changed through the PIME filtering process.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<table>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
object name
</th>
<th style="text-align:right;">
no. samples
</th>
<th style="text-align:right;">
total reads
</th>
<th style="text-align:right;">
total asvs
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FULL data set
</td>
<td style="text-align:left;">
ssu_ps_work
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
6457054
</td>
<td style="text-align:right;">
47725
</td>
</tr>
<tr>
<td style="text-align:left;">
Rarefied data
</td>
<td style="text-align:left;">
ssu_pime_ds
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
820496
</td>
<td style="text-align:right;">
27818
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME filtered data
</td>
<td style="text-align:left;">
ssu_ps_pime
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
711567
</td>
<td style="text-align:right;">
7795
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (ALMR)
</td>
<td style="text-align:left;">
ssu_ps_pime_ALMR
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
179992
</td>
<td style="text-align:right;">
3417
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (CRIS)
</td>
<td style="text-align:left;">
ssu_ps_pime_CRIS
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
176867
</td>
<td style="text-align:right;">
3336
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PAST)
</td>
<td style="text-align:left;">
ssu_ps_pime_PAST
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
175867
</td>
<td style="text-align:right;">
3471
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PUCL)
</td>
<td style="text-align:left;">
ssu_ps_pime_PUCL
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
178841
</td>
<td style="text-align:right;">
3859
</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="all-samples-otu-data">All Samples (OTU Data)</h1>
<p>We will use PIME (Prevalence Interval for Microbiome Evaluation) to create a filtered data set.</p>
<h2 id="setup-3">Setup</h2>
<p>First, choose a phyloseq object and a sample data frame</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds <span class="ot">&lt;-</span> ssu_ps_work_otu</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>ssu_which_pime <span class="ot">&lt;-</span> <span class="st">&quot;ssu_pime_ds&quot;</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a><span class="do">### USE THIS TO CREATE NEW SAMPLE VARIABLE FOR SPLITTING</span></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_samp_data &lt;- sample_data(ssu_pime_ds)  </span></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_samp_data$YR_TREAT &lt;- paste(tmp_samp_data$YEAR, tmp_samp_data$TREAT, sep=&quot;_&quot;)  </span></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_ps &lt;- merge_phyloseq(ssu_pime_ds, tmp_samp_data)   </span></span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_pime_ds &lt;- tmp_ps  </span></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a><span class="co">#rm(list = ls(pattern = &quot;tmp_&quot;))</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 21906 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 21906 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>ssu_pime_sample_d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(<span class="fu">otu_table</span>(ssu_pime_ds)))</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>ssu_pime_sample_d <span class="ot">&lt;-</span> ssu_pime_sample_d <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">total_reads =</span> <span class="dv">1</span>)</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds <span class="ot">&lt;-</span> <span class="fu">rarefy_even_depth</span>(ssu_pime_ds, </span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">sample.size =</span> <span class="fu">min</span>(ssu_pime_sample_d<span class="sc">$</span>total_reads), </span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">trimOTUs =</span> <span class="cn">TRUE</span>, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">rngseed =</span> <span class="dv">119</span>)</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>ssu_pime_ds</span></code></pre></div>
</div>
<pre><code>`set.seed(119)` was used to initialize repeatable random subsampling.
Please record this for your records so others can reproduce.
Try `set.seed(119); .Random.seed` for the full vector

8456 OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 13450 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 13450 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<p>The first step in PIME is to define if the microbial community presents a high relative abundance of taxa with low prevalence, which is considered as noise in PIME analysis. This is calculated by random forests analysis and is the baseline noise detection.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>ssu_pime.oob.error <span class="ot">&lt;-</span> <span class="fu">pime.oob.error</span>(ssu_pime_ds, <span class="st">&quot;SITE&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>[1] 0.01644737</code></pre>
</div>
<h2 id="split-by-predictor-variable-3">Split by Predictor Variable</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ssu_pime_ds))</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>ssu_per_variable_obj <span class="ot">&lt;-</span> <span class="fu">pime.split.by.variable</span>(ssu_pime_ds, <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>ssu_per_variable_obj</span></code></pre></div>
</div>
<details>
<summary>
Detailed results for splitting by sample variables
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$ALMR
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 6631 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 6631 taxa by 7 taxonomic ranks ]

$CRIS
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 6995 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 6995 taxa by 7 taxonomic ranks ]

$PAST
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7174 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 7174 taxa by 7 taxonomic ranks ]

$PUCL
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 6864 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 6864 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</details>
<h2 id="calculate-prevalence-intervals-3">Calculate Prevalence Intervals</h2>
<p>Using the output of <code>pime.split.by.variable</code>, we calculate the prevalence intervals with the function <code>pime.prevalence</code>. This function estimates the highest prevalence possible (no empty ASV table), calculates prevalence for taxa, starting at 5 maximum prevalence possible (no empty ASV table or dropping samples). After prevalence calculation, each prevalence interval are merged.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>ssu_prevalences <span class="ot">&lt;-</span> <span class="fu">pime.prevalence</span>(ssu_per_variable_obj)</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>ssu_prevalences</span></code></pre></div>
</div>
<details>
<summary>
Detailed results for all prevalences intervals
</summary>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$`5`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 6176 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 6176 taxa by 7 taxonomic ranks ]

$`10`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 4310 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 4310 taxa by 7 taxonomic ranks ]

$`15`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3371 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3371 taxa by 7 taxonomic ranks ]

$`20`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2671 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2671 taxa by 7 taxonomic ranks ]

$`25`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2219 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 2219 taxa by 7 taxonomic ranks ]

$`30`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1963 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1963 taxa by 7 taxonomic ranks ]

$`35`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1687 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1687 taxa by 7 taxonomic ranks ]

$`40`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1464 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1464 taxa by 7 taxonomic ranks ]

$`45`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1271 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1271 taxa by 7 taxonomic ranks ]

$`50`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1106 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1106 taxa by 7 taxonomic ranks ]

$`55`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 989 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 989 taxa by 7 taxonomic ranks ]

$`60`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 856 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 856 taxa by 7 taxonomic ranks ]

$`65`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 747 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 747 taxa by 7 taxonomic ranks ]

$`70`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 660 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 660 taxa by 7 taxonomic ranks ]

$`75`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 576 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 576 taxa by 7 taxonomic ranks ]

$`80`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 514 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 514 taxa by 7 taxonomic ranks ]

$`85`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 424 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 424 taxa by 7 taxonomic ranks ]

$`90`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 346 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 346 taxa by 7 taxonomic ranks ]

$`95`
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 243 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 243 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</details>
<h2 id="calculate-best-prevalence-3">Calculate Best Prevalence</h2>
<p>Finally, we use the function <code>pime.best.prevalence</code> to calculate the best prevalence. The function uses randomForest to build random forests trees for samples classification and variable importance computation. It performs classifications for each prevalence interval returned by <code>pime.prevalence</code>. Variable importance is calculated, returning the Mean Decrease Accuracy (MDA), Mean Decrease Impurity (MDI), overall and by sample group, and taxonomy for each ASV. PIME keeps the top 30 variables with highest MDA each prevalence level.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1911</span>)</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>ssu_best.prev <span class="ot">&lt;-</span> <span class="fu">pime.best.prevalence</span>(ssu_prevalences, <span class="st">&quot;SITE&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>         Interval OOB error rate (%) OTUs  Nseqs
1   Prevalence 5%               0.99 6176 795077
2  Prevalence 10%               0.33 4310 766467
3  Prevalence 15%                  0 3371 740985
4  Prevalence 20%                  0 2671 714655
5  Prevalence 25%                  0 2219 691362
6  Prevalence 30%                  0 1963 674292
7  Prevalence 35%                  0 1687 653346
8  Prevalence 40%                  0 1464 632093
9  Prevalence 45%                  0 1271 611427
10 Prevalence 50%                  0 1106 591999
11 Prevalence 55%                  0  989 576446
12 Prevalence 60%                  0  856 554607
13 Prevalence 65%                  0  747 532236
14 Prevalence 70%                  0  660 510859
15 Prevalence 75%                  0  576 483648
16 Prevalence 80%                  0  514 465813
17 Prevalence 85%                  0  424 437508
18 Prevalence 90%                  0  346 398043
19 Prevalence 95%                  0  243 336356</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best <span class="ot">&lt;-</span> ssu_best.prev<span class="sc">$</span><span class="st">`</span><span class="at">OOB error</span><span class="st">`</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>)] <span class="ot">&lt;-</span> <span class="fu">sapply</span>(ssu_what_is_best[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>)], as.numeric)</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best <span class="ot">&lt;-</span> ssu_what_is_best <span class="sc">%&gt;%</span> </span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;OOB_error_rate&quot;</span> <span class="ot">=</span> <span class="st">&quot;OOB error rate (%)&quot;</span>)</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>ssu_what_is_best<span class="sc">$</span>Interval <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(ssu_what_is_best<span class="sc">$</span>Interval, <span class="st">&quot;%&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>ssu_best <span class="ot">&lt;-</span> <span class="fu">with</span>(ssu_what_is_best, Interval[<span class="fu">which.min</span>(OOB_error_rate)])</span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>ssu_best <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;`&quot;</span>, ssu_best, <span class="st">&quot;`&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a>ssu_prev_choice <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_best.prev$`Importance`$&quot;</span>, ssu_best, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a>ssu_imp_best <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_prev_choice)))</span></code></pre></div>
</div>
<p>Looks like the lowest OOB error rate (%) that retains the most ASVs is 0% from <code>Prevalence 15</code>. We will use this interval.</p>
<h2 id="best-prevalence-summary-3">Best Prevalence Summary</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_imp_best[, 13:14] &lt;- list(NULL)</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>ssu_imp_best[,(<span class="fu">ncol</span>(ssu_imp_best)<span class="sc">-</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">ncol</span>(ssu_imp_best)] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="cn">NULL</span>)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>ssu_imp_best <span class="ot">&lt;-</span> ssu_imp_best <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;ASV_ID&quot;</span> <span class="ot">=</span> <span class="st">&quot;SequenceID&quot;</span>)</span></code></pre></div>
</div>
<p></br></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="qx00m3t3ggbavcfsmwqw" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="qx00m3t3ggbavcfsmwqw">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000357142857142\" data-max=\"0.025752891514422\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000831844711311\" data-max=\"0.036087341272488\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.000199582809663\" data-max=\"0.014357745921317\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.001921880012227\" data-max=\"0.048237012729548\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"0.005985410922777\" data-max=\"0.019489986253831\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"1.02290124990258\" data-max=\"4.2497082631301\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Table of ASVs (max 30)\n                                     from the chosen prevalence\n                                     interval.<\/em>\n<\/caption>","data":[["OTU15","OTU405","OTU2","OTU10","OTU52","OTU69","OTU49","OTU239","OTU47","OTU67","OTU43","OTU7","OTU84","OTU9","OTU131","OTU21","OTU138","OTU622","OTU150","OTU276","OTU187","OTU196","OTU342","OTU44","OTU20","OTU179","OTU12","OTU126","OTU4","OTU259"],[0.0208120567345309,0.00575449329517171,0.0250832165581589,0.0146129375933259,0.0256697420336161,0.025752891514422,0.0134976950638837,0.0108158489802718,0.00817541056169385,0.00979621536354295,0.0244061805784944,0.00754822944497683,0.00863664189538036,0.00817838209415517,0.00751081214443283,0.00988197095185277,0.00476968173179294,0.00315389841454358,0.0119092315265235,0.00730155996827298,0.00252234086105054,0.00657898484292062,0.00837455334562156,0.00417197015924704,0.0137410075182189,0.00523173710908531,0.00644366429537175,0.0148152982466386,0.00402457629484853,0.000357142857142857],[0.0360873412724872,0.0139888677637816,0.0132406238718762,0.0124271382062268,0.00729656536274183,0.00918202861952862,0.0181951559165714,0.0103015931761468,0.0112042418256961,0.0117215255512732,0.00800235075494123,0.00848535612604953,0.00880563117118816,0.0123077489177489,0.0088975034927788,0.00971592912144914,0.00126411908583948,0.007994634878736,0.0072856037091076,0.00502511396104604,0.00387129201597254,0.00813189033189033,0.00552972964455723,0.00585625693789838,0.0045787340656864,0.0101155586080586,0.000831844711311752,0.00408434077079108,0.00210565099994642,0.00619695015794059],[0.00903556453358453,0.00537137854021878,0.0143577459213166,0.0101864588400517,0.00900238854793846,0.00535255394071518,0.00216870614011507,0.012547911655559,0.0100866396728966,0.0100253540337634,0.00326135958747718,0.00943158072505188,0.00682168481693219,0.00904370363694008,0.00845613322490964,0.00706960422005798,0.00085378294099095,0.00312792826791714,0.00186269349033532,0.00944023085020328,0.00282614601790014,0.0069483460150071,0.00998948841416233,0.0102723480304899,0.00290659718417783,0.00279176082086768,0.00187828218734912,0.000199582809663455,0.0020015790660952,0.000702447163515017],[0.0109880084766061,0.0482370127295476,0.0128737646866161,0.00927492029990361,0.00571652015414029,0.00222762242828166,0.00703120536882773,0.0058856034263643,0.00936600578792473,0.00669838072153461,0.00192188001222714,0.0108526319821982,0.0116934521146294,0.00599789929368877,0.00755319830376172,0.00553709645474351,0.0237807447743047,0.0163096328843881,0.00823513464203119,0.00703061111192515,0.0192660467740106,0.00699024018991761,0.00395788010407588,0.0068313292367516,0.00677955900905011,0.00819801805446483,0.0154493304396753,0.00554482970612003,0.0169904444901899,0.0161684255778158],[0.0194899862538308,0.0180662117202996,0.016250765059237,0.0117260991513234,0.0115726681431647,0.0105438230910796,0.0100199010627422,0.00984627568979541,0.00957579558928265,0.0094825275034515,0.00930126742087565,0.00900160687679067,0.00896608546019761,0.00872478644808476,0.00805927525692601,0.00793168003505044,0.00758362158028998,0.00749713004865436,0.00730753499389698,0.0071863591561091,0.00711995103015209,0.00706877666048094,0.00686118827198044,0.00685296741411999,0.00681469047358178,0.00655780306598194,0.0062030617509256,0.00606157622076428,0.00603141027840553,0.00598541092277752],[4.2497082631301,3.81677710430543,3.13612773963779,2.02108453237095,2.60216816134054,1.75371830642814,2.5568568188732,1.81766397604372,1.81109838846845,1.65063727167937,2.21856827534313,1.74457918991601,2.15125592130408,1.43633580868833,1.32757780639753,1.28033539544848,1.76225855307437,2.73733263270474,1.58944041019245,1.33559786140354,1.70776495264947,1.34661634410186,1.41188233430539,1.18080793955771,1.54134371101743,1.45686634546911,1.17740809313943,1.02290124990258,1.76101298875005,1.14964640256703],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Bacteria","Bacteria"],["Acidobacteriota","Nitrospirota","Proteobacteria","Proteobacteria","Chloroflexi","Acidobacteriota","Chloroflexi","Proteobacteria","Nitrospirota","k_Archaea","Nitrospirota","Acidobacteriota","Nitrospirota","Desulfobacterota","Proteobacteria","Desulfobacterota","Acidobacteriota","Acidobacteriota","Acidobacteriota","Proteobacteria","Acidobacteriota","Planctomycetota","Proteobacteria","Sva0485","Chloroflexi","Acidobacteriota","Crenarchaeota","Proteobacteria","Acidobacteriota","Acidobacteriota"],["Thermoanaerobaculia","Thermodesulfovibrionia","Gammaproteobacteria","Gammaproteobacteria","Anaerolineae","Thermoanaerobaculia","Anaerolineae","Gammaproteobacteria","Thermodesulfovibrionia","k_Archaea","Nitrospiria","Aminicenantia","Thermodesulfovibrionia","Syntrophobacteria","Gammaproteobacteria","Desulfobacteria","Aminicenantia","Thermoanaerobaculia","Aminicenantia","Gammaproteobacteria","Thermoanaerobaculia","Phycisphaerae","Alphaproteobacteria","p_Sva0485","Anaerolineae","Aminicenantia","Bathyarchaeia","Alphaproteobacteria","Aminicenantia","Thermoanaerobaculia"],["Thermoanaerobaculales","c_Thermodesulfovibrionia","Ectothiorhodospirales","Ectothiorhodospirales","Anaerolineales","Thermoanaerobaculales","Anaerolineales","AT-s2-59","c_Thermodesulfovibrionia","k_Archaea","Nitrospirales","Aminicenantales","c_Thermodesulfovibrionia","Syntrophobacterales","UBA10353_marine_group","Desulfatiglandales","Aminicenantales","Thermoanaerobaculales","Aminicenantales","Steroidobacterales","Thermoanaerobaculales","MSBL9","Kiloniellales","p_Sva0485","c_Anaerolineae","Aminicenantales","c_Bathyarchaeia","Kiloniellales","Aminicenantales","Thermoanaerobaculales"],["Thermoanaerobaculaceae","c_Thermodesulfovibrionia","Thioalkalispiraceae","Thioalkalispiraceae","Anaerolineaceae","Thermoanaerobaculaceae","Anaerolineaceae","o_AT-s2-59","c_Thermodesulfovibrionia","k_Archaea","Nitrospiraceae","o_Aminicenantales","c_Thermodesulfovibrionia","o_Syntrophobacterales","o_UBA10353_marine_group","Desulfatiglandaceae","o_Aminicenantales","Thermoanaerobaculaceae","o_Aminicenantales","Woeseiaceae","Thermoanaerobaculaceae","SG8-4","Kiloniellaceae","p_Sva0485","c_Anaerolineae","o_Aminicenantales","c_Bathyarchaeia","Kiloniellaceae","o_Aminicenantales","Thermoanaerobaculaceae"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ASV_ID<\/th>\n      <th>ALMR<\/th>\n      <th>CRIS<\/th>\n      <th>PAST<\/th>\n      <th>PUCL<\/th>\n      <th>MeanDecreaseAccuracy<\/th>\n      <th>MeanDecreaseGini<\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[[5,10,-1],["5","10","All"]],"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 4, 3, \",\", \".\");\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'80%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'80%'});\nvar value=data[10]; $(this.api().cell(row, 10).node()).css({'font-size':'80%'});\nvar value=data[11]; $(this.api().cell(row, 11).node()).css({'font-size':'80%'});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>Now we need to create a phyloseq object of ASVs at this cutoff (<code>Prevalence 15</code>).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>ssu_best_val <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(ssu_best, <span class="st">&quot;Prevalence &quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>ssu_best_val <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_prevalences$&quot;</span>, ssu_best_val, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>ssu_prevalence_best <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_best_val)))</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_prevalence_best, <span class="st">&quot;rdata/trepo/ssu_otu_prevalence_best.rds&quot;</span>)</span></code></pre></div>
</div>
<p>And look at a summary of the data.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 3371 taxa and 304 samples ]
sample_data() Sample Data:       [ 304 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 3371 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<table>
<thead>
<tr class="header">
<th>Metric</th>
<th>Results</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>2242</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>2557</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>740985</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>2437</td>
</tr>
<tr class="odd">
<td>Mean number of reads</td>
<td>2437</td>
</tr>
<tr class="even">
<td>Median number of reads</td>
<td>2442</td>
</tr>
<tr class="odd">
<td>Total ASVS</td>
<td>3371</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.771</td>
</tr>
</tbody>
</table>
<h2 id="estimate-error-in-prediction-3">Estimate Error in Prediction</h2>
<p>Using the function <code>pime.error.prediction</code> we can estimate the error in prediction. For each prevalence interval, this function randomizes the samples labels into arbitrary groupings using <code>n</code> random permutations, defined by the <code>bootstrap</code> value. For each, randomized and prevalence filtered, data set the OOB error rate is calculated to estimate whether the original differences in groups of samples occur by chance. Results are in a list containing a table and a box plot summarizing the results.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>ssu_randomized <span class="ot">&lt;-</span> <span class="fu">pime.error.prediction</span>(ssu_pime_ds, <span class="st">&quot;TEMP&quot;</span>,</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">bootstrap =</span> <span class="dv">100</span>, <span class="at">parallel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">max.prev =</span> <span class="dv">95</span>)</span></code></pre></div>
</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>It is also possible to estimate the variation of OOB error for each prevalence interval filtering. This is done by running the random forests classification for <code>n</code> times, determined by the <code>bootstrap</code> value. The function will return a box plot figure and a table for each classification error.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>ssu_replicated.oob.error <span class="ot">&lt;-</span> <span class="fu">pime.oob.replicate</span>(ssu_prevalences, <span class="st">&quot;SITE&quot;</span>, </span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">bootstrap =</span> <span class="dv">100</span>, <span class="at">parallel =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="trepo-pime_files/figure-html5/unnamed-chunk-157-1.png" width="624" /></p>
</div>
<p>To obtain the confusion matrix from random forests classification use the following:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>ssu_prev_confuse <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;ssu_best.prev$`Confusion`$&quot;</span>, ssu_best, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> (ssu_prev_confuse)))</span></code></pre></div>
</div>
<h2 id="save-phyloseq-pime-objects-3">Save Phyloseq PIME objects</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime <span class="ot">&lt;-</span> ssu_prevalence_best</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(ssu_ps_pime), <span class="at">rooted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(ssu_ps_pime))</span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ssu_ps_pime,</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>                               sample_data,</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>                               ssu_ps_pime_tree)</span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_otu_pime.rds&quot;</span>)</span></code></pre></div>
</div>
<h3 id="split-save-by-predictor-variable-3">Split &amp; save by predictor variable</h3>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime_split <span class="ot">&lt;-</span> <span class="fu">pime.split.by.variable</span>(ssu_ps_pime, <span class="st">&quot;SITE&quot;</span>)</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>ALMR, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_otu_pime_ALMR.rds&quot;</span>)</span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>CRIS, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_otu_pime_CRIS.rds&quot;</span>)</span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>PAST, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_otu_pime_PAST.rds&quot;</span>)</span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_ps_pime_split<span class="sc">$</span>PUCL, <span class="st">&quot;files/trepo/pime/rdata/ssu_ps_otu_pime_PUCL.rds&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>$ALMR
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1719 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1719 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 1719 tips and 1718 internal nodes ]

$CRIS
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1713 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1713 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 1713 tips and 1712 internal nodes ]

$PAST
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1771 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1771 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 1771 tips and 1770 internal nodes ]

$PUCL
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1893 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 1893 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 1893 tips and 1892 internal nodes ]</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="create-ampvis2-pime-object-3">Create Ampvis2 PIME Object</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>ssu_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(ssu_ps_pime)))</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>ssu_otu[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(ssu_otu, as.numeric)</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>ssu_otu <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(ssu_otu)</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>ssu_tax <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu_ps_pime)))</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>ssu_samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(ssu_ps_pime))</span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>ssu_ps_amp <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(<span class="fu">otu_table</span>(ssu_otu, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>),</span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">tax_table</span>(ssu_tax, ssu_tax),</span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">sample_data</span>(ssu_samples))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>ssu_amp_otu_pime  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">otu_table</span>(ssu_ps_amp))</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>ssu_amp_otu_pime <span class="ot">&lt;-</span> ssu_amp_otu_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime  <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(ssu_ps_amp))</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime <span class="ot">&lt;-</span> ssu_amp_tax_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime<span class="sc">$</span>ASV_SEQ <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a><span class="do">## THE NEXT  COMMAND SHOULD BE REMOVED, THIS IS A TEMP FIX</span></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a><span class="do">## BECAUSE OTU_ID&#39;S not INCLUDED FROM OTU WF</span></span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>ssu_amp_tax_pime<span class="sc">$</span>ASV_ID <span class="ot">&lt;-</span> ssu_amp_tax_pime<span class="sc">$</span>OTU</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a><span class="do">#########</span></span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ssu_amp_tax_pime)[<span class="fu">colnames</span>(ssu_amp_tax_pime) <span class="sc">==</span> <span class="st">&quot;ASV_ID&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Species&quot;</span></span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a>ssu_amp_otu_tax_pime <span class="ot">&lt;-</span> <span class="fu">left_join</span>(ssu_amp_otu_pime, ssu_amp_tax_pime, <span class="at">by =</span> <span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a>ssu_samp_data_t <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ssu_samp_data)</span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a><span class="co">#ssu_samp_data_t[,7:9] &lt;- NULL</span></span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a>ssu_amp_data_pime <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(ssu_amp_otu_tax_pime, <span class="at">metadata =</span> ssu_samp_data_t)</span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span></span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">amp_alphadiv</span>(</span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a>    ssu_amp_data_pime,</span>
<span id="cb106-20"><a href="#cb106-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">measure =</span> <span class="st">&quot;observed&quot;</span>,</span>
<span id="cb106-21"><a href="#cb106-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">richness =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-22"><a href="#cb106-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-23"><a href="#cb106-23" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span></span>
<span id="cb106-24"><a href="#cb106-24" aria-hidden="true" tabindex="-1"></a>  ssu_diversity_pime <span class="sc">%&gt;%</span></span>
<span id="cb106-25"><a href="#cb106-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb106-26"><a href="#cb106-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;total_reads&quot;</span> <span class="ot">=</span> <span class="st">&quot;Reads&quot;</span>,</span>
<span id="cb106-27"><a href="#cb106-27" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;total_otus&quot;</span> <span class="ot">=</span> <span class="st">&quot;ObservedOTUs&quot;</span>)</span>
<span id="cb106-28"><a href="#cb106-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-29"><a href="#cb106-29" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime<span class="sc">$</span>coverage <span class="ot">=</span> <span class="fu">cut</span>(ssu_diversity_pime<span class="sc">$</span>total_reads,</span>
<span id="cb106-30"><a href="#cb106-30" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5000</span>, <span class="dv">25000</span>, <span class="dv">100000</span>, <span class="dv">200000</span>),</span>
<span id="cb106-31"><a href="#cb106-31" aria-hidden="true" tabindex="-1"></a>                          <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb106-32"><a href="#cb106-32" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;low (&lt; 5k)&quot;</span>, <span class="st">&quot;medium (5-25k)&quot;</span>,</span>
<span id="cb106-33"><a href="#cb106-33" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;high (25-100k)&quot;</span>, <span class="st">&quot;extra_high (&gt; 100k)&quot;</span>))</span>
<span id="cb106-34"><a href="#cb106-34" aria-hidden="true" tabindex="-1"></a>ssu_diversity_pime <span class="ot">&lt;-</span> ssu_diversity_pime[<span class="fu">order</span>(ssu_diversity_pime<span class="sc">$</span>SamName), ]</span>
<span id="cb106-35"><a href="#cb106-35" aria-hidden="true" tabindex="-1"></a>ssu_amp_pime <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(ssu_amp_otu_tax_pime,</span>
<span id="cb106-36"><a href="#cb106-36" aria-hidden="true" tabindex="-1"></a>                          <span class="at">metadata =</span> ssu_diversity_pime, <span class="at">tree =</span> ssu_ps_pime_tree)</span>
<span id="cb106-37"><a href="#cb106-37" aria-hidden="true" tabindex="-1"></a>ssu_samp_pime <span class="ot">&lt;-</span> ssu_diversity_pime</span>
<span id="cb106-38"><a href="#cb106-38" aria-hidden="true" tabindex="-1"></a>ssu_samp_pime <span class="ot">&lt;-</span> ssu_samp_pime <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code>ampvis2 object with 4 elements. 
Summary of OTU table:
     Samples         OTUs  Total#Reads    Min#Reads    Max#Reads 
         304         3371       740985         2242         2557 
Median#Reads    Avg#Reads 
        2442      2437.45 

Assigned taxonomy:
     Kingdom       Phylum        Class        Order       Family 
  3371(100%) 3364(99.79%) 3370(99.97%) 3367(99.88%) 3365(99.82%) 
       Genus      Species 
  3371(100%)   3371(100%) 

Metadata variables: 9 
 SamName, SITE, WEEK, REGION, SEASON, REP, total_reads, total_otus, coverage</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ssu_amp_pime, <span class="st">&quot;files/trepo/pime/rdata/ssu_otu_amp_pime.rds&quot;</span>)</span></code></pre></div>
</div>
<h2 id="summary-3">Summary</h2>
<p>Let’s take a look at a table of sample information. Any header with the <code>_p</code> suffix is the <em>PIME filtered</em> data.</p>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="kekt3855pmbio224m21n" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="kekt3855pmbio224m21n">{"x":{"filter":"top","filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"2242\" data-max=\"2557\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none; position: absolute; width: 200px;\">\n      <div data-min=\"486\" data-max=\"978\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Sample summary table.\n            Use the buttons to navigate through the table or\n            download a copy.<\/em>\n<\/caption>","data":[["ALMR_W01_IB_HS_R1","ALMR_W01_IB_HS_R2","ALMR_W01_IB_HS_R3","ALMR_W03_IB_HS_R1","ALMR_W03_IB_HS_R2","ALMR_W03_IB_HS_R3","ALMR_W05_IB_HS_R1","ALMR_W05_IB_HS_R2","ALMR_W05_IB_HS_R3","ALMR_W07_IB_HS_R1","ALMR_W07_IB_HS_R2","ALMR_W07_IB_HS_R3","ALMR_W09_IB_HS_R1","ALMR_W09_IB_HS_R2","ALMR_W09_IB_HS_R3","ALMR_W11_IB_HS_R1","ALMR_W11_IB_HS_R2","ALMR_W11_IB_HS_R3","ALMR_W13_IB_NS_R1","ALMR_W13_IB_NS_R2","ALMR_W13_IB_NS_R3","ALMR_W15_IB_NS_R1","ALMR_W15_IB_NS_R2","ALMR_W15_IB_NS_R3","ALMR_W17_IB_NS_R1","ALMR_W17_IB_NS_R2","ALMR_W17_IB_NS_R3","ALMR_W19_IB_NS_R1","ALMR_W19_IB_NS_R2","ALMR_W19_IB_NS_R3","ALMR_W21_IB_NS_R1","ALMR_W21_IB_NS_R2","ALMR_W21_IB_NS_R3","ALMR_W23_IB_NS_R1","ALMR_W23_IB_NS_R2","ALMR_W23_IB_NS_R3","ALMR_W25_IB_NS_R1","ALMR_W25_IB_NS_R2","ALMR_W25_IB_NS_R3","ALMR_W27_IB_NS_R1","ALMR_W27_IB_NS_R2","ALMR_W27_IB_NS_R3","ALMR_W29_IB_NS_R1","ALMR_W29_IB_NS_R2","ALMR_W31_IB_NS_R1","ALMR_W31_IB_NS_R2","ALMR_W31_IB_NS_R3","ALMR_W33_IB_NS_R1","ALMR_W33_IB_NS_R2","ALMR_W33_IB_NS_R3","ALMR_W35_IB_NS_R1","ALMR_W35_IB_NS_R2","ALMR_W35_IB_NS_R3","ALMR_W37_IB_NS_R2","ALMR_W37_IB_NS_R3","ALMR_W39_IB_NS_R1","ALMR_W39_IB_NS_R2","ALMR_W39_IB_NS_R3","ALMR_W41_IB_HS_R1","ALMR_W41_IB_HS_R2","ALMR_W41_IB_HS_R3","ALMR_W43_IB_HS_R1","ALMR_W43_IB_HS_R2","ALMR_W43_IB_HS_R3","ALMR_W45_IB_HS_R1","ALMR_W45_IB_HS_R2","ALMR_W45_IB_HS_R3","ALMR_W47_IB_HS_R1","ALMR_W47_IB_HS_R2","ALMR_W47_IB_HS_R3","ALMR_W49_IB_HS_R1","ALMR_W49_IB_HS_R2","ALMR_W49_IB_HS_R3","ALMR_W51_IB_HS_R1","ALMR_W51_IB_HS_R2","ALMR_W51_IB_HS_R3","CRIS_W01_OB_HS_R1","CRIS_W01_OB_HS_R2","CRIS_W01_OB_HS_R3","CRIS_W03_OB_HS_R1","CRIS_W03_OB_HS_R2","CRIS_W03_OB_HS_R3","CRIS_W05_OB_HS_R1","CRIS_W05_OB_HS_R2","CRIS_W05_OB_HS_R3","CRIS_W07_OB_HS_R1","CRIS_W07_OB_HS_R2","CRIS_W07_OB_HS_R3","CRIS_W09_OB_HS_R1","CRIS_W09_OB_HS_R2","CRIS_W09_OB_HS_R3","CRIS_W11_OB_HS_R1","CRIS_W11_OB_HS_R2","CRIS_W11_OB_HS_R3","CRIS_W13_OB_NS_R1","CRIS_W13_OB_NS_R2","CRIS_W13_OB_NS_R3","CRIS_W15_OB_NS_R1","CRIS_W15_OB_NS_R2","CRIS_W15_OB_NS_R3","CRIS_W17_OB_NS_R1","CRIS_W17_OB_NS_R2","CRIS_W17_OB_NS_R3","CRIS_W19_OB_NS_R1","CRIS_W19_OB_NS_R2","CRIS_W19_OB_NS_R3","CRIS_W21_OB_NS_R1","CRIS_W21_OB_NS_R2","CRIS_W21_OB_NS_R3","CRIS_W23_OB_NS_R1","CRIS_W23_OB_NS_R2","CRIS_W23_OB_NS_R3","CRIS_W25_OB_NS_R1","CRIS_W25_OB_NS_R2","CRIS_W25_OB_NS_R3","CRIS_W27_OB_NS_R1","CRIS_W27_OB_NS_R2","CRIS_W27_OB_NS_R3","CRIS_W29_OB_NS_R1","CRIS_W29_OB_NS_R2","CRIS_W29_OB_NS_R3","CRIS_W31_OB_NS_R1","CRIS_W31_OB_NS_R2","CRIS_W31_OB_NS_R3","CRIS_W33_OB_NS_R1","CRIS_W33_OB_NS_R2","CRIS_W33_OB_NS_R3","CRIS_W35_OB_NS_R1","CRIS_W35_OB_NS_R2","CRIS_W35_OB_NS_R3","CRIS_W37_OB_NS_R1","CRIS_W37_OB_NS_R2","CRIS_W37_OB_NS_R3","CRIS_W39_OB_NS_R1","CRIS_W39_OB_NS_R2","CRIS_W39_OB_NS_R3","CRIS_W41_OB_HS_R1","CRIS_W41_OB_HS_R2","CRIS_W43_OB_HS_R1","CRIS_W43_OB_HS_R2","CRIS_W43_OB_HS_R3","CRIS_W45_OB_HS_R2","CRIS_W45_OB_HS_R3","CRIS_W47_OB_HS_R1","CRIS_W47_OB_HS_R2","CRIS_W47_OB_HS_R3","CRIS_W49_OB_HS_R1","CRIS_W49_OB_HS_R2","CRIS_W49_OB_HS_R3","CRIS_W51_OB_HS_R1","CRIS_W51_OB_HS_R2","CRIS_W51_OB_HS_R3","PAST_W01_IB_HS_R1","PAST_W01_IB_HS_R2","PAST_W01_IB_HS_R3","PAST_W03_IB_HS_R1","PAST_W03_IB_HS_R2","PAST_W03_IB_HS_R3","PAST_W05_IB_HS_R1","PAST_W05_IB_HS_R2","PAST_W05_IB_HS_R3","PAST_W07_IB_HS_R1","PAST_W07_IB_HS_R2","PAST_W07_IB_HS_R3","PAST_W09_IB_HS_R1","PAST_W09_IB_HS_R2","PAST_W09_IB_HS_R3","PAST_W11_IB_HS_R1","PAST_W11_IB_HS_R2","PAST_W11_IB_HS_R3","PAST_W13_IB_NS_R1","PAST_W13_IB_NS_R2","PAST_W13_IB_NS_R3","PAST_W15_IB_NS_R1","PAST_W15_IB_NS_R2","PAST_W15_IB_NS_R3","PAST_W17_IB_NS_R1","PAST_W17_IB_NS_R2","PAST_W17_IB_NS_R3","PAST_W19_IB_NS_R1","PAST_W19_IB_NS_R2","PAST_W19_IB_NS_R3","PAST_W21_IB_NS_R1","PAST_W21_IB_NS_R2","PAST_W21_IB_NS_R3","PAST_W23_IB_NS_R1","PAST_W23_IB_NS_R2","PAST_W23_IB_NS_R3","PAST_W25_IB_NS_R1","PAST_W25_IB_NS_R2","PAST_W25_IB_NS_R3","PAST_W27_IB_NS_R1","PAST_W27_IB_NS_R2","PAST_W29_IB_NS_R1","PAST_W29_IB_NS_R2","PAST_W29_IB_NS_R3","PAST_W31_IB_NS_R1","PAST_W31_IB_NS_R2","PAST_W31_IB_NS_R3","PAST_W33_IB_NS_R1","PAST_W33_IB_NS_R2","PAST_W33_IB_NS_R3","PAST_W35_IB_NS_R1","PAST_W35_IB_NS_R2","PAST_W35_IB_NS_R3","PAST_W37_IB_NS_R1","PAST_W37_IB_NS_R3","PAST_W39_IB_NS_R1","PAST_W39_IB_NS_R2","PAST_W39_IB_NS_R3","PAST_W41_IB_HS_R1","PAST_W41_IB_HS_R2","PAST_W41_IB_HS_R3","PAST_W43_IB_HS_R1","PAST_W43_IB_HS_R2","PAST_W43_IB_HS_R3","PAST_W45_IB_HS_R1","PAST_W45_IB_HS_R2","PAST_W45_IB_HS_R3","PAST_W47_IB_HS_R1","PAST_W47_IB_HS_R2","PAST_W47_IB_HS_R3","PAST_W49_IB_HS_R1","PAST_W49_IB_HS_R2","PAST_W49_IB_HS_R3","PAST_W51_IB_HS_R1","PAST_W51_IB_HS_R2","PAST_W51_IB_HS_R3","PUCL_W01_OB_HS_R1","PUCL_W01_OB_HS_R2","PUCL_W01_OB_HS_R3","PUCL_W03_OB_HS_R1","PUCL_W03_OB_HS_R2","PUCL_W03_OB_HS_R3","PUCL_W05_OB_HS_R1","PUCL_W05_OB_HS_R2","PUCL_W05_OB_HS_R3","PUCL_W07_OB_HS_R1","PUCL_W07_OB_HS_R2","PUCL_W07_OB_HS_R3","PUCL_W09_OB_HS_R1","PUCL_W09_OB_HS_R2","PUCL_W09_OB_HS_R3","PUCL_W11_OB_HS_R1","PUCL_W11_OB_HS_R2","PUCL_W11_OB_HS_R3","PUCL_W13_OB_NS_R1","PUCL_W13_OB_NS_R2","PUCL_W13_OB_NS_R3","PUCL_W15_OB_NS_R1","PUCL_W15_OB_NS_R2","PUCL_W15_OB_NS_R3","PUCL_W17_OB_NS_R1","PUCL_W17_OB_NS_R2","PUCL_W17_OB_NS_R3","PUCL_W19_OB_NS_R1","PUCL_W19_OB_NS_R2","PUCL_W19_OB_NS_R3","PUCL_W21_OB_NS_R1","PUCL_W21_OB_NS_R2","PUCL_W21_OB_NS_R3","PUCL_W23_OB_NS_R1","PUCL_W23_OB_NS_R2","PUCL_W23_OB_NS_R3","PUCL_W25_OB_NS_R1","PUCL_W25_OB_NS_R2","PUCL_W25_OB_NS_R3","PUCL_W27_OB_NS_R1","PUCL_W27_OB_NS_R2","PUCL_W27_OB_NS_R3","PUCL_W29_OB_NS_R1","PUCL_W29_OB_NS_R2","PUCL_W29_OB_NS_R3","PUCL_W31_OB_NS_R1","PUCL_W31_OB_NS_R2","PUCL_W31_OB_NS_R3","PUCL_W33_OB_NS_R1","PUCL_W33_OB_NS_R2","PUCL_W33_OB_NS_R3","PUCL_W35_OB_NS_R1","PUCL_W35_OB_NS_R2","PUCL_W37_OB_NS_R1","PUCL_W37_OB_NS_R2","PUCL_W37_OB_NS_R3","PUCL_W39_OB_NS_R1","PUCL_W39_OB_NS_R2","PUCL_W39_OB_NS_R3","PUCL_W41_OB_HS_R1","PUCL_W41_OB_HS_R2","PUCL_W41_OB_HS_R3","PUCL_W43_OB_HS_R1","PUCL_W43_OB_HS_R3","PUCL_W45_OB_HS_R1","PUCL_W45_OB_HS_R2","PUCL_W45_OB_HS_R3","PUCL_W47_OB_HS_R1","PUCL_W47_OB_HS_R2","PUCL_W47_OB_HS_R3","PUCL_W49_OB_HS_R1","PUCL_W49_OB_HS_R2","PUCL_W49_OB_HS_R3","PUCL_W51_OB_HS_R1","PUCL_W51_OB_HS_R2","PUCL_W51_OB_HS_R3"],["ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","ALMR","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","CRIS","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PAST","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL","PUCL"],["W01","W01","W01","W03","W03","W03","W05","W05","W05","W07","W07","W07","W09","W09","W09","W11","W11","W11","W13","W13","W13","W15","W15","W15","W17","W17","W17","W19","W19","W19","W21","W21","W21","W23","W23","W23","W25","W25","W25","W27","W27","W27","W29","W29","W31","W31","W31","W33","W33","W33","W35","W35","W35","W37","W37","W39","W39","W39","W41","W41","W41","W43","W43","W43","W45","W45","W45","W47","W47","W47","W49","W49","W49","W51","W51","W51","W01","W01","W01","W03","W03","W03","W05","W05","W05","W07","W07","W07","W09","W09","W09","W11","W11","W11","W13","W13","W13","W15","W15","W15","W17","W17","W17","W19","W19","W19","W21","W21","W21","W23","W23","W23","W25","W25","W25","W27","W27","W27","W29","W29","W29","W31","W31","W31","W33","W33","W33","W35","W35","W35","W37","W37","W37","W39","W39","W39","W41","W41","W43","W43","W43","W45","W45","W47","W47","W47","W49","W49","W49","W51","W51","W51","W01","W01","W01","W03","W03","W03","W05","W05","W05","W07","W07","W07","W09","W09","W09","W11","W11","W11","W13","W13","W13","W15","W15","W15","W17","W17","W17","W19","W19","W19","W21","W21","W21","W23","W23","W23","W25","W25","W25","W27","W27","W29","W29","W29","W31","W31","W31","W33","W33","W33","W35","W35","W35","W37","W37","W39","W39","W39","W41","W41","W41","W43","W43","W43","W45","W45","W45","W47","W47","W47","W49","W49","W49","W51","W51","W51","W01","W01","W01","W03","W03","W03","W05","W05","W05","W07","W07","W07","W09","W09","W09","W11","W11","W11","W13","W13","W13","W15","W15","W15","W17","W17","W17","W19","W19","W19","W21","W21","W21","W23","W23","W23","W25","W25","W25","W27","W27","W27","W29","W29","W29","W31","W31","W31","W33","W33","W33","W35","W35","W37","W37","W37","W39","W39","W39","W41","W41","W41","W43","W43","W45","W45","W45","W47","W47","W47","W49","W49","W49","W51","W51","W51"],["IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","IB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB","OB"],["HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","NS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS","HS"],["R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R1","R2","R3","R1","R2","R3","R1","R2","R3","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R1","R2","R3","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3","R1","R2","R3"],[2438,2478,2461,2459,2489,2475,2480,2472,2489,2409,2470,2412,2423,2442,2413,2401,2440,2437,2448,2514,2443,2476,2489,2474,2481,2438,2429,2468,2474,2521,2458,2469,2445,2483,2419,2436,2499,2470,2469,2453,2424,2455,2492,2486,2485,2475,2493,2460,2392,2431,2445,2420,2455,2470,2476,2484,2478,2511,2453,2481,2478,2485,2455,2491,2421,2470,2424,2392,2411,2392,2456,2452,2377,2470,2374,2391,2403,2456,2457,2380,2395,2457,2421,2557,2387,2452,2416,2435,2450,2406,2433,2435,2463,2454,2414,2415,2466,2458,2363,2424,2448,2490,2433,2368,2452,2398,2415,2456,2429,2391,2470,2433,2458,2489,2463,2447,2431,2439,2450,2452,2456,2432,2426,2379,2317,2393,2431,2464,2459,2451,2394,2451,2436,2400,2392,2473,2417,2425,2392,2424,2414,2424,2386,2378,2400,2366,2354,2250,2382,2406,2419,2435,2479,2451,2431,2368,2419,2380,2468,2440,2391,2426,2443,2416,2423,2448,2441,2450,2422,2412,2470,2489,2449,2396,2412,2416,2425,2452,2499,2485,2462,2435,2380,2500,2434,2429,2462,2444,2509,2454,2411,2452,2465,2390,2420,2412,2471,2424,2441,2349,2242,2263,2479,2459,2484,2439,2477,2456,2466,2452,2454,2436,2378,2419,2458,2461,2397,2387,2394,2346,2480,2358,2464,2459,2490,2436,2420,2371,2432,2454,2435,2497,2479,2470,2442,2471,2394,2447,2439,2443,2451,2475,2463,2444,2487,2491,2422,2416,2442,2433,2441,2400,2465,2430,2424,2445,2457,2422,2446,2478,2415,2428,2480,2442,2428,2431,2401,2472,2447,2468,2463,2501,2432,2469,2425,2389,2452,2367,2388,2428,2487,2532,2474,2427,2419,2442,2469,2460,2428,2430,2466,2479,2369,2392,2448,2371,2374,2362,2388,2429,2507,2352,2450,2415],[671,772,824,750,770,743,783,788,666,857,783,819,820,826,762,738,747,741,822,681,732,789,736,672,765,804,816,633,690,686,726,710,717,662,810,676,714,765,711,800,720,738,717,747,653,684,718,771,829,806,811,793,809,651,721,734,813,821,806,750,715,555,739,745,786,768,811,757,824,847,853,792,801,812,773,799,790,816,785,798,826,733,756,486,811,769,831,790,746,839,758,751,728,685,765,820,641,709,842,650,734,747,753,819,721,721,810,528,669,614,804,733,678,724,715,737,728,669,719,666,732,735,721,685,854,801,806,740,761,792,753,657,788,738,724,701,634,705,724,729,730,683,814,780,788,833,798,572,801,802,731,785,830,773,789,718,754,814,835,806,829,815,832,772,829,830,811,766,680,728,779,696,757,683,708,692,816,805,727,712,720,742,818,820,732,706,756,690,792,808,864,690,711,689,742,722,795,704,763,658,778,775,815,782,788,774,754,781,726,652,774,531,766,781,716,655,881,873,834,821,715,793,860,772,761,776,843,670,833,775,888,861,929,882,875,917,831,841,818,813,740,738,854,807,806,753,712,872,879,835,820,812,852,698,808,781,848,812,811,745,803,740,777,734,855,828,841,739,823,611,785,742,760,636,882,927,893,924,958,868,905,703,798,839,697,594,795,880,855,767,801,743,948,963,867,943,966,838,912,978,819,816,783,896]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>SamName<\/th>\n      <th>SITE<\/th>\n      <th>WEEK<\/th>\n      <th>REGION<\/th>\n      <th>SEASON<\/th>\n      <th>REP<\/th>\n      <th>total_reads_p<\/th>\n      <th>total_otus_p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"dom":"Blfrtip","buttons":["copy","csv","excel"],"pageLength":5,"lengthMenu":[5,15],"columnDefs":[{"className":"dt-right","targets":[6,7]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<p>And here is how the data sets changed through the PIME filtering process.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><br/></p>
<div class="layout-chunk" data-layout="l-body-outset">
<table>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
object name
</th>
<th style="text-align:right;">
no. samples
</th>
<th style="text-align:right;">
total reads
</th>
<th style="text-align:right;">
total otus
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FULL data set
</td>
<td style="text-align:left;">
ssu_ps_work_otu
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
6457054
</td>
<td style="text-align:right;">
21906
</td>
</tr>
<tr>
<td style="text-align:left;">
Rarefied data
</td>
<td style="text-align:left;">
ssu_pime_ds
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
820496
</td>
<td style="text-align:right;">
13450
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME filtered data
</td>
<td style="text-align:left;">
ssu_ps_pime
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
740985
</td>
<td style="text-align:right;">
3371
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (ALMR)
</td>
<td style="text-align:left;">
ssu_ps_pime_ALMR
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
186549
</td>
<td style="text-align:right;">
1719
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (CRIS)
</td>
<td style="text-align:left;">
ssu_ps_pime_CRIS
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
184235
</td>
<td style="text-align:right;">
1713
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PAST)
</td>
<td style="text-align:left;">
ssu_ps_pime_PAST
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
184770
</td>
<td style="text-align:right;">
1771
</td>
</tr>
<tr>
<td style="text-align:left;">
PIME (PUCL)
</td>
<td style="text-align:left;">
ssu_ps_pime_PUCL
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
185431
</td>
<td style="text-align:right;">
1893
</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/tropical-repo/web/blob/master/trepo-pime.Rmd">clicking this link</a>. Please note, that in order to process the data <em>and</em> build the website, we needed to run the workflow and get the results. Then hard code the results and turn off the individual commands. So the raw file for this page is a bit messy—you have been warned.</p>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/tropical-repo/web/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/tropical-repo/web/">https://github.com/tropical-repo/web/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>© Copyright 2021 The TROPICAL-REPO Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
